{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "JELLYFISH = \"/home/analytics/distr/jellyfish-linux\"\n",
    "FASTQC_PATH = \"/home/analytics/distr/FastQC/fastqc\"\n",
    "TRIMMOMATIC_JAR = \"/home/analytics/bin/trimmomatic.jar\"\n",
    "SPADES = \"/home/analytics/distr/SPAdes-3.12.0-Linux/bin/spades.py\"\n",
    "QUAST = \"/home/analytics/distr/quast-5.0.0/quast.py\"\n",
    "PROKKA = \"/home/analytics/anaconda3/bin/prokka\"\n",
    "BARRNAP = \"/home/analytics/anaconda3/bin/barrnap\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!wget -P ../data/week3/ https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292678sub_S1_L001_R1_001.fastq.gz 2> /dev/null\n",
    "!wget -P ../data/week3/ https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292678sub_S1_L001_R2_001.fastq.gz 2> /dev/null\n",
    "    \n",
    "!wget -P ../data/week3/ https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292862_S2_L001_R1_001.fastq.gz 2> /dev/null\n",
    "!wget -P ../data/week3/ https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292862_S2_L001_R2_001.fastq.gz 2> /dev/null\n",
    "    \n",
    "!wget -P ../data/week3/ https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292770_S1_L001_R1_001.fastq.gz 2> /dev/null\n",
    "!wget -P ../data/week3/ https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292770_S1_L001_R2_001.fastq.gz 2> /dev/null"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!gunzip ../data/week3/SRR292678sub_S1_L001_R1_001.fastq.gz\n",
    "!gunzip ../data/week3/SRR292678sub_S1_L001_R2_001.fastq.gz\n",
    "!gunzip ../data/week3/SRR292862_S2_L001_R1_001.fastq.gz\n",
    "!gunzip ../data/week3/SRR292862_S2_L001_R2_001.fastq.gz\n",
    "!gunzip ../data/week3/SRR292770_S1_L001_R1_001.fastq.gz\n",
    "!gunzip ../data/week3/SRR292770_S1_L001_R2_001.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR292678sub_S1_L001_R1_001.fastq\n",
      "Analysis complete for SRR292678sub_S1_L001_R2_001.fastq\n",
      "Analysis complete for SRR292862_S2_L001_R1_001.fastq\n",
      "Analysis complete for SRR292862_S2_L001_R2_001.fastq\n",
      "Analysis complete for SRR292770_S1_L001_R1_001.fastq\n",
      "Analysis complete for SRR292770_S1_L001_R2_001.fastq\n"
     ]
    }
   ],
   "source": [
    "!$FASTQC_PATH ../data/week3/SRR292678sub_S1_L001_R1_001.fastq 2>/dev/null\n",
    "!$FASTQC_PATH ../data/week3/SRR292678sub_S1_L001_R2_001.fastq 2>/dev/null\n",
    "\n",
    "!$FASTQC_PATH ../data/week3/SRR292862_S2_L001_R1_001.fastq 2> /dev/null\n",
    "!$FASTQC_PATH ../data/week3/SRR292862_S2_L001_R2_001.fastq 2> /dev/null\n",
    "\n",
    "!$FASTQC_PATH ../data/week3/SRR292770_S1_L001_R1_001.fastq 2> /dev/null\n",
    "!$FASTQC_PATH ../data/week3/SRR292770_S1_L001_R2_001.fastq 2> /dev/null"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "from IPython.display import IFrame\n",
    "from boto.s3 import connect_to_region\n",
    "from boto import connect_s3\n",
    "from boto.s3.key import Key\n",
    "AWS_S3_ROOT = \"http://bioinf-workshop.s3-website.eu-central-1.amazonaws.com/\"\n",
    "\n",
    "def perform_file_upload(file_, file_name):\n",
    "\n",
    "    conn = connect_to_region(\"eu-central-1\", \n",
    "                             aws_access_key_id=open(\"../.access_key\", \"r\").read().strip(),\n",
    "                             aws_secret_access_key=open(\"../.secret_key\", \"r\").read().strip())\n",
    "\n",
    "    bucket = conn.get_bucket(\"bioinf-workshop\")\n",
    "    key = Key(bucket)\n",
    "    key.key = file_name\n",
    "    key.set_contents_from_string(file_, headers={\"Content-Type\": \"text/html\"})\n",
    "    key.set_acl('public-read')\n",
    "    return AWS_S3_ROOT + file_name # public url to access the uploaded file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <iframe\n",
       "            width=\"1000\"\n",
       "            height=\"600\"\n",
       "            src=\"http://bioinf-workshop.s3-website.eu-central-1.amazonaws.com/week_3/fastqc_1.html\"\n",
       "            frameborder=\"0\"\n",
       "            allowfullscreen\n",
       "        ></iframe>\n",
       "        "
      ],
      "text/plain": [
       "<IPython.lib.display.IFrame at 0x7f9a0b5f7978>"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = perform_file_upload(open(\"../data/week3/SRR292678sub_S1_L001_R1_001_fastqc.html\", \"r\").read(), \"week_3/fastqc_1.html\")\n",
    "IFrame(src=f, width=1000, height=600)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <iframe\n",
       "            width=\"1000\"\n",
       "            height=\"600\"\n",
       "            src=\"http://bioinf-workshop.s3-website.eu-central-1.amazonaws.com/week_3/fastqc_2.html\"\n",
       "            frameborder=\"0\"\n",
       "            allowfullscreen\n",
       "        ></iframe>\n",
       "        "
      ],
      "text/plain": [
       "<IPython.lib.display.IFrame at 0x7f9a0b44d2e8>"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = perform_file_upload(open(\"../data/week3/SRR292678sub_S1_L001_R2_001_fastqc.html\", \"r\").read(), \"week_3/fastqc_2.html\")\n",
    "IFrame(src=f, width=1000, height=600)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <iframe\n",
       "            width=\"1000\"\n",
       "            height=\"600\"\n",
       "            src=\"http://bioinf-workshop.s3-website.eu-central-1.amazonaws.com/week_3/fastqc_2.html\"\n",
       "            frameborder=\"0\"\n",
       "            allowfullscreen\n",
       "        ></iframe>\n",
       "        "
      ],
      "text/plain": [
       "<IPython.lib.display.IFrame at 0x7f9a0ac1a898>"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = perform_file_upload(open(\"../data/week3/SRR292862_S2_L001_R1_001_fastqc.html\", \"r\").read(), \"week_3/fastqc_2.html\")\n",
    "IFrame(src=f, width=1000, height=600)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <iframe\n",
       "            width=\"1000\"\n",
       "            height=\"600\"\n",
       "            src=\"http://bioinf-workshop.s3-website.eu-central-1.amazonaws.com/week_3/fastqc_2.html\"\n",
       "            frameborder=\"0\"\n",
       "            allowfullscreen\n",
       "        ></iframe>\n",
       "        "
      ],
      "text/plain": [
       "<IPython.lib.display.IFrame at 0x7f9a0b484d68>"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = perform_file_upload(open(\"../data/week3/SRR292862_S2_L001_R2_001_fastqc.html\", \"r\").read(), \"week_3/fastqc_2.html\")\n",
    "IFrame(src=f, width=1000, height=600)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <iframe\n",
       "            width=\"1000\"\n",
       "            height=\"600\"\n",
       "            src=\"http://bioinf-workshop.s3-website.eu-central-1.amazonaws.com/week_3/fastqc_2.html\"\n",
       "            frameborder=\"0\"\n",
       "            allowfullscreen\n",
       "        ></iframe>\n",
       "        "
      ],
      "text/plain": [
       "<IPython.lib.display.IFrame at 0x7f9a0ac1acc0>"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = perform_file_upload(open(\"../data/week3/SRR292770_S1_L001_R1_001_fastqc.html\", \"r\").read(), \"week_3/fastqc_2.html\")\n",
    "IFrame(src=f, width=1000, height=600)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <iframe\n",
       "            width=\"1000\"\n",
       "            height=\"600\"\n",
       "            src=\"http://bioinf-workshop.s3-website.eu-central-1.amazonaws.com/week_3/fastqc_2.html\"\n",
       "            frameborder=\"0\"\n",
       "            allowfullscreen\n",
       "        ></iframe>\n",
       "        "
      ],
      "text/plain": [
       "<IPython.lib.display.IFrame at 0x7f9a0b09fda0>"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = perform_file_upload(open(\"../data/week3/SRR292770_S1_L001_R2_001_fastqc.html\", \"r\").read(), \"week_3/fastqc_2.html\")\n",
    "IFrame(src=f, width=1000, height=600)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Command line: /home/analytics/distr/SPAdes-3.12.0-Linux/bin/spades.py\t-t\t50\t-m\t500\t-o\t/home/analytics/projects/bioinf_workshop/data/week3/assembly\t-1\t/home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R1_001.fastq\t-2\t/home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R2_001.fastq\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.12.0\n",
      "  Python version: 3.5.4\n",
      "  OS: Linux-3.10.0-862.3.3.el7.x86_64-x86_64-with-centos-7.5.1804-Core\n",
      "\n",
      "Output dir: /home/analytics/projects/bioinf_workshop/data/week3/assembly\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R1_001.fastq']\n",
      "      right reads: ['/home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R2_001.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: automatic selection based on read length\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/analytics/projects/bioinf_workshop/data/week3/assembly/tmp\n",
      "  Threads: 50\n",
      "  Memory limit (in Gb): 500\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/analytics/projects/bioinf_workshop/data/week3/assembly/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/analytics/distr/SPAdes-3.12.0-Linux/bin/spades-hammer /home/analytics/projects/bioinf_workshop/data/week3/assembly/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/analytics/projects/bioinf_workshop/data/week3/assembly/corrected/configs/config.info\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 50\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 500 Gb\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.002     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 800 files using 50 threads. This might take a while.\n",
      "  0:00:00.003     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.003     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 3.33331 Gb\n",
      "  0:00:00.003     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 83886\n",
      "  0:00:03.793    27G / 27G   INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R1_001.fastq\n",
      "  0:00:18.673    27G / 28G   INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 5499346 reads\n",
      "  0:00:18.674    27G / 28G   INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R2_001.fastq\n",
      "  0:00:33.491    27G / 28G   INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 10998692 reads\n",
      "  0:00:33.491    27G / 28G   INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 10998692 reads processed\n",
      "  0:00:35.822   200M / 28G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:36.348   200M / 28G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 35869396 kmers in total.\n",
      "  0:00:36.349   200M / 28G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:36.613   200M / 28G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:37.880   204M / 28G   INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:38.203   204M / 28G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 16640728 bytes occupied (3.7114 bits per kmer).\n",
      "  0:00:38.207   200M / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:39.195   752M / 28G   INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:47.133   752M / 28G   INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:54.869   752M / 28G   INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 13059930\n",
      "  0:00:54.870   476M / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:55.160     1G / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R1_001.fastq\n",
      "  0:01:10.009     1G / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R2_001.fastq\n",
      "  0:01:24.744     1G / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:01:24.884     1G / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 35869396 kmers in total. Among them 23232864 (64.7707%) are singletons.\n",
      "  0:01:24.885     1G / 28G   INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:01:28.562     1G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 7 non-read kmers were generated.\n",
      "  0:01:28.562     1G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:01:28.562     1G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 4366674. Among them 2314470 (53.003%) are good\n",
      "  0:01:28.562     1G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 40749. Among them 40317 (98.9399%) are good\n",
      "  0:01:28.562     1G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 8817994. Among them 7328745 (83.1112%) are good\n",
      "  0:01:28.562     1G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 3.57255 kmers\n",
      "  0:01:28.562     1G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.01904\n",
      "  0:01:28.562     1G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 9683532\n",
      "  0:01:28.562     1G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.92945,0.0468015,0.0182831,0.00546531),(0.0176801,0.97103,0.00425595,0.00703388),(0.00700894,0.00424543,0.971047,0.0176981),(0.00545963,0.0182562,0.0467842,0.9295))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:28.600     1G / 28G   INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:01:28.600     1G / 28G   INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 50 threads.\n",
      "  0:01:54.709     1G / 28G   INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 53046 new k-mers.\n",
      "  0:02:20.938     1G / 28G   INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 1043 new k-mers.\n",
      "  0:02:47.319     1G / 28G   INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 30 new k-mers.\n",
      "  0:03:13.665     1G / 28G   INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 3 produced 0 new k-mers.\n",
      "  0:03:13.665     1G / 28G   INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:03:13.665     1G / 28G   INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 50 threads.\n",
      "  0:03:13.665     1G / 28G   INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R1_001.fastq and /home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R2_001.fastq\n",
      "  0:03:28.573     4G / 28G   INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 5000000 reads.\n",
      "  0:03:38.414     5G / 28G   INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:03:45.473     5G / 28G   INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:03:46.760     5G / 28G   INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 1 of 499346 reads.\n",
      "  0:03:47.745     5G / 28G   INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 1\n",
      "  0:03:48.373     5G / 28G   INFO    General                 (hammer_tools.cpp          : 185)   Written batch 1\n",
      "  0:03:51.496     1G / 28G   INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 935573 bases in 788580 reads.\n",
      "  0:03:51.496     1G / 28G   INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 989881557.\n",
      "  0:03:51.502   200M / 28G   INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/analytics/projects/bioinf_workshop/data/week3/assembly/corrected/corrected.yaml\n",
      "  0:03:51.503   200M / 28G   INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/analytics/projects/bioinf_workshop/data/week3/assembly/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K21\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/analytics/projects/bioinf_workshop/data/week3/assembly/K21/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 500 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/analytics/projects/bioinf_workshop/data/week3/assembly/dataset.info) with K=21\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 50\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.001     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.307    80M / 80M   INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.511    88M / 88M   INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:00:00.848   112M / 112M  INFO    General                 (binary_converter.hpp      :  93)   65536 reads processed\n",
      "  0:00:01.427   156M / 156M  INFO    General                 (binary_converter.hpp      :  93)   131072 reads processed\n",
      "  0:00:02.442   244M / 244M  INFO    General                 (binary_converter.hpp      :  93)   262144 reads processed\n",
      "  0:00:04.419   424M / 424M  INFO    General                 (binary_converter.hpp      :  93)   524288 reads processed\n",
      "  0:00:10.673   704M / 704M  INFO    General                 (binary_converter.hpp      :  93)   1048576 reads processed\n",
      "  0:00:20.974   704M / 704M  INFO    General                 (binary_converter.hpp      :  93)   2097152 reads processed\n",
      "  0:00:41.957   704M / 704M  INFO    General                 (binary_converter.hpp      :  93)   4194304 reads processed\n",
      "  0:00:56.171   640M / 704M  INFO    General                 (binary_converter.hpp      : 117)   5498906 reads written\n",
      "  0:00:57.334     4M / 704M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:57.435   132M / 704M  INFO    General                 (binary_converter.hpp      : 117)   415 reads written\n",
      "  0:00:57.449     4M / 704M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:57.546   132M / 704M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:57.675     4M / 704M  INFO    General                 (construction.cpp          : 111)   Max read length 90\n",
      "  0:00:57.675     4M / 704M  INFO    General                 (construction.cpp          : 117)   Average read length 89.9999\n",
      "  0:00:57.675     4M / 704M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:57.676     4M / 704M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 2500 files using 50 threads. This might take a while.\n",
      "  0:00:57.678     4M / 704M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:00:57.678     4M / 704M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 3.33331 Gb\n",
      "  0:00:57.678     4M / 704M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 26843\n",
      "  0:01:05.585    27G / 27G   INFO    General                 (kmer_splitters.hpp        : 289)   Processed 21996454 reads\n",
      "  0:01:05.586    27G / 27G   INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:01:07.843   208M / 27G   INFO    General                 (kmer_splitters.hpp        : 308)   Used 21996454 reads\n",
      "  0:01:07.843   208M / 27G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:01:14.709   208M / 27G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6272563 kmers in total.\n",
      "  0:01:14.709   208M / 27G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:01:14.937   200M / 27G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:01:14.937   200M / 27G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:01:14.937   200M / 27G   INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 800 files using 50 threads. This might take a while.\n",
      "  0:01:14.938   200M / 27G   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:01:14.938   200M / 27G   INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 3.33203 Gb\n",
      "  0:01:14.938   200M / 27G   INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 83886\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:19.407    27G / 27G   INFO    General                 (kmer_splitters.hpp        : 380)   Processed 6272563 kmers\n",
      "  0:01:19.407    27G / 27G   INFO    General                 (kmer_splitters.hpp        : 385)   Used 6272563 kmers.\n",
      "  0:01:21.955   204M / 27G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:01:23.659   204M / 27G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6272518 kmers in total.\n",
      "  0:01:23.660   204M / 27G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:01:23.786   204M / 27G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:01:23.978   204M / 27G   INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:01:24.049   204M / 27G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2916896 bytes occupied (3.72022 bits per kmer).\n",
      "  0:01:24.054   212M / 27G   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:01:24.185   212M / 27G   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:01:24.196   208M / 27G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
      "  0:01:24.196   208M / 27G   INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
      "  0:01:24.196   208M / 27G   INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:01:24.412   208M / 27G   INFO   Early tip clipping       (early_simplification.hpp  : 184)   310446 22-mers were removed by early tip clipper\n",
      "  0:01:24.412   208M / 27G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:01:24.416   208M / 27G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:01:24.535   212M / 27G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 32960 sequences extracted\n",
      "  0:01:24.619   212M / 27G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:01:24.656   212M / 27G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 5 loops collected\n",
      "  0:01:24.680   212M / 27G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:01:24.680   212M / 27G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:01:24.680   212M / 27G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:01:24.773   212M / 27G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2934144 bytes occupied (3.74219 bits per kmer).\n",
      "  0:01:24.781   236M / 27G   INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:01:29.000   236M / 27G   INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:01:29.087   236M / 27G   INFO    General                 (construction.cpp          : 464)   Processed 65891 edges\n",
      "  0:01:29.126   200M / 27G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:01:29.126   200M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 3\n",
      "  0:01:29.126   200M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 142\n",
      "  0:01:29.126   200M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 142. Coverage mad: 51.891\n",
      "  0:01:29.126   200M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:01:29.265   200M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:01:29.619   200M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:01:30.859   200M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:01:33.606   200M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:01:34.191   200M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 138.203. Fitted coverage std. dev: 48.8732\n",
      "  0:01:34.196   200M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.855544\n",
      "  0:01:34.196   200M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 35\n",
      "  0:01:34.196   200M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 35\n",
      "  0:01:34.196   200M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 5167637\n",
      "  0:01:34.196   200M / 27G   INFO    General                 (genomic_info_filler.cpp   : 112)   Mean coverage was calculated as 138.203\n",
      "  0:01:34.196   200M / 27G   INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 35\n",
      "  0:01:34.196   200M / 27G   INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:01:34.196   200M / 27G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:01:34.196   200M / 27G   INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:01:34.196   200M / 27G   INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:01:34.196   200M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:01:34.201   200M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:01:34.201   200M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
      "  0:01:34.214   200M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 212 times\n",
      "  0:01:34.214   200M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
      "  0:01:34.342   200M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 4114 times\n",
      "  0:01:34.342   200M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
      "  0:01:34.367   200M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 7842 times\n",
      "  0:01:34.367   200M / 27G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:01:34.367   200M / 27G   INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:01:34.367   200M / 27G   INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:01:34.367   200M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:01:34.367   200M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.369   200M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 20 times\n",
      "  0:01:34.369   200M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:34.481   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2687 times\n",
      "  0:01:34.481   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:34.486   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 88 times\n",
      "  0:01:34.486   204M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:01:34.486   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.486   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:01:34.486   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:34.489   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 40 times\n",
      "  0:01:34.489   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:34.491   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 79 times\n",
      "  0:01:34.491   204M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:01:34.491   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.491   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:34.491   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:34.494   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 56 times\n",
      "  0:01:34.494   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:34.495   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 26 times\n",
      "  0:01:34.495   204M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:01:34.495   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.495   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:34.495   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:34.496   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 24 times\n",
      "  0:01:34.496   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:34.496   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 20 times\n",
      "  0:01:34.496   204M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:01:34.496   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.496   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:34.496   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:34.496   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 9 times\n",
      "  0:01:34.496   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 8 times\n",
      "  0:01:34.497   204M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 6 times\n",
      "  0:01:34.497   204M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:01:34.497   204M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 6 times\n",
      "  0:01:34.497   204M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 6 times\n",
      "  0:01:34.497   204M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
      "  0:01:34.497   204M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:01:34.497   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.498   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:34.498   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:34.504   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 6 times\n",
      "  0:01:34.504   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:34.505   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:01:34.505   204M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:01:34.505   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.505   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:34.505   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:34.505   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:01:34.505   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:34.505   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:01:34.505   204M / 27G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:01:34.505   204M / 27G   INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:01:34.505   204M / 27G   INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:01:34.505   204M / 27G   INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:01:34.505   204M / 27G   INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:01:34.505   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.505   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:34.505   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:34.511   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:01:34.511   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:34.511   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:34.511   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:34.517   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:01:34.517   204M / 27G   INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:01:34.519   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:01:34.520   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 311 times\n",
      "  0:01:34.520   204M / 27G   INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:01:34.521   204M / 27G   INFO    General                 (simplification.cpp        : 476)   Average coverage = 145.521\n",
      "  0:01:34.521   204M / 27G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:01:34.521   204M / 27G   INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K21/assembly_graph_with_scaffolds.gfa\n",
      "  0:01:34.568   204M / 27G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K21/before_rr.fasta\n",
      "  0:01:34.624   204M / 27G   INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K21/assembly_graph.fastg\n",
      "  0:01:34.824   204M / 27G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K21/simplified_contigs.fasta\n",
      "  0:01:34.881   204M / 27G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K21/final_contigs.fasta\n",
      "  0:01:34.938   204M / 27G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:01:34.938   204M / 27G   INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K21/assembly_graph_with_scaffolds.gfa\n",
      "  0:01:34.984   204M / 27G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K21/before_rr.fasta\n",
      "  0:01:35.099   204M / 27G   INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K21/assembly_graph.fastg\n",
      "  0:01:35.334   204M / 27G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K21/simplified_contigs.fasta\n",
      "  0:01:35.391   204M / 27G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K21/final_contigs.fasta\n",
      "  0:01:35.450   204M / 27G   INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:01:35.458   200M / 27G   INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 1 minutes 35 seconds\n",
      "Max read length detected as 90\n",
      "\n",
      "== Running assembler: K33\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/analytics/projects/bioinf_workshop/data/week3/assembly/K33/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 500 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/analytics/projects/bioinf_workshop/data/week3/assembly/dataset.info) with K=33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 50\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
      "  0:00:00.003     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 90\n",
      "  0:00:00.003     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 89.9999\n",
      "  0:00:00.003     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.003     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 2500 files using 50 threads. This might take a while.\n",
      "  0:00:00.005     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:00:00.005     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 3.33331 Gb\n",
      "  0:00:00.005     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 16384\n",
      "  0:00:09.303    34G / 34G   INFO    General                 (kmer_splitters.hpp        : 289)   Processed 21996454 reads\n",
      "  0:00:09.303    34G / 34G   INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:13.790   212M / 34G   INFO    General                 (kmer_splitters.hpp        : 308)   Used 21996454 reads\n",
      "  0:00:13.790   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:18.460   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6242403 kmers in total.\n",
      "  0:00:18.461   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:18.875   200M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:18.875   200M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:18.875   200M / 34G   INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 800 files using 50 threads. This might take a while.\n",
      "  0:00:18.875   200M / 34G   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:00:18.875   200M / 34G   INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 3.33203 Gb\n",
      "  0:00:18.875   200M / 34G   INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 41943\n",
      "  0:00:23.844    27G / 34G   INFO    General                 (kmer_splitters.hpp        : 380)   Processed 6242403 kmers\n",
      "  0:00:23.845    27G / 34G   INFO    General                 (kmer_splitters.hpp        : 385)   Used 6242403 kmers.\n",
      "  0:00:26.442   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:27.242   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6246463 kmers in total.\n",
      "  0:00:27.243   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:27.408   204M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:27.565   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:27.675   204M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2904720 bytes occupied (3.72015 bits per kmer).\n",
      "  0:00:27.680   212M / 34G   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:27.815   212M / 34G   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:27.820   208M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
      "  0:00:27.820   208M / 34G   INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:27.820   208M / 34G   INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:00:28.032   208M / 34G   INFO   Early tip clipping       (early_simplification.hpp  : 184)   366647 34-mers were removed by early tip clipper\n",
      "  0:00:28.032   208M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:28.035   208M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:28.153   208M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 22195 sequences extracted\n",
      "  0:00:28.240   208M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:28.276   208M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 3 loops collected\n",
      "  0:00:28.295   212M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:28.295   212M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:28.295   212M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:28.394   212M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2920064 bytes occupied (3.74223 bits per kmer).\n",
      "  0:00:28.402   236M / 34G   INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:32.051   232M / 34G   INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:32.136   236M / 34G   INFO    General                 (construction.cpp          : 464)   Processed 44386 edges\n",
      "  0:00:32.180   200M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:32.180   200M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 3\n",
      "  0:00:32.180   200M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 116\n",
      "  0:00:32.180   200M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 116. Coverage mad: 44.478\n",
      "  0:00:32.180   200M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:00:32.349   200M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:00:32.692   200M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:00:33.757   200M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:00:36.171   200M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:00:37.191   200M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 114.16. Fitted coverage std. dev: 41.5006\n",
      "  0:00:37.195   200M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.810054\n",
      "  0:00:37.195   200M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 30\n",
      "  0:00:37.195   200M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 30\n",
      "  0:00:37.195   200M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 5169113\n",
      "  0:00:37.195   200M / 34G   INFO    General                 (genomic_info_filler.cpp   : 112)   Mean coverage was calculated as 114.16\n",
      "  0:00:37.195   200M / 34G   INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 30\n",
      "  0:00:37.195   200M / 34G   INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:37.195   200M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:37.195   200M / 34G   INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:37.195   200M / 34G   INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:37.196   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:37.197   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:37.198   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
      "  0:00:37.208   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 252 times\n",
      "  0:00:37.208   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
      "  0:00:37.293   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 1903 times\n",
      "  0:00:37.293   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
      "  0:00:37.317   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 8240 times\n",
      "  0:00:37.317   200M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:37.318   200M / 34G   INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:37.318   200M / 34G   INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:37.318   200M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:37.318   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:37.319   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 17 times\n",
      "  0:00:37.319   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:37.424   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1671 times\n",
      "  0:00:37.424   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:37.429   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 51 times\n",
      "  0:00:37.429   200M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:37.429   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:37.429   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:37.429   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:37.432   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 41 times\r\n",
      "  0:00:37.432   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:37.434   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 58 times\r\n",
      "  0:00:37.434   200M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\r\n",
      "  0:00:37.434   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:37.435   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\r\n",
      "  0:00:37.435   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:37.439   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 39 times\r\n",
      "  0:00:37.439   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:37.440   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 28 times\r\n",
      "  0:00:37.440   200M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\r\n",
      "  0:00:37.440   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:37.440   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\r\n",
      "  0:00:37.440   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:37.441   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 24 times\r\n",
      "  0:00:37.441   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:37.441   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\r\n",
      "  0:00:37.441   200M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\r\n",
      "  0:00:37.441   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:37.441   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:37.441   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:37.441   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2 times\r\n",
      "  0:00:37.441   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\r\n",
      "  0:00:37.442   200M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\r\n",
      "  0:00:37.442   200M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\r\n",
      "  0:00:37.442   200M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\r\n",
      "  0:00:37.442   200M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\r\n",
      "  0:00:37.442   200M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:37.442   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:37.443   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\r\n",
      "  0:00:37.443   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:37.443   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\r\n",
      "  0:00:37.443   200M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\r\n",
      "  0:00:37.443   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:37.445   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:37.445   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:37.451   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2 times\r\n",
      "  0:00:37.451   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:37.452   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\r\n",
      "  0:00:37.452   200M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\r\n",
      "  0:00:37.452   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:37.452   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:37.452   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:37.452   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\r\n",
      "  0:00:37.452   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:37.452   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\r\n",
      "  0:00:37.452   200M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\r\n",
      "  0:00:37.452   200M / 34G   INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\r\n",
      "  0:00:37.452   200M / 34G   INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\r\n",
      "  0:00:37.452   200M / 34G   INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\r\n",
      "  0:00:37.452   200M / 34G   INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\r\n",
      "  0:00:37.452   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:37.452   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:37.452   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:37.454   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\r\n",
      "  0:00:37.454   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:37.455   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:37.455   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:37.457   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\r\n",
      "  0:00:37.457   200M / 34G   INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\r\n",
      "  0:00:37.458   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\r\n",
      "  0:00:37.459   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 160 times\r\n",
      "  0:00:37.459   200M / 34G   INFO    General                 (simplification.cpp        : 470)   Counting average coverage\r\n",
      "  0:00:37.459   200M / 34G   INFO    General                 (simplification.cpp        : 476)   Average coverage = 119.293\r\n",
      "  0:00:37.459   200M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\r\n",
      "  0:00:37.459   200M / 34G   INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K33/assembly_graph_with_scaffolds.gfa\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:37.504   200M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K33/before_rr.fasta\n",
      "  0:00:37.557   200M / 34G   INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K33/assembly_graph.fastg\n",
      "  0:00:37.750   204M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K33/simplified_contigs.fasta\n",
      "  0:00:37.803   200M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K33/final_contigs.fasta\n",
      "  0:00:37.856   200M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:37.856   200M / 34G   INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K33/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:37.900   200M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K33/before_rr.fasta\n",
      "  0:00:38.005   200M / 34G   INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K33/assembly_graph.fastg\n",
      "  0:00:38.373   204M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K33/simplified_contigs.fasta\n",
      "  0:00:38.494   200M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K33/final_contigs.fasta\n",
      "  0:00:38.579   200M / 34G   INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:38.586   200M / 34G   INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 38 seconds\n",
      "\n",
      "== Running assembler: K55\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/analytics/projects/bioinf_workshop/data/week3/assembly/K55/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 500 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/analytics/projects/bioinf_workshop/data/week3/assembly/dataset.info) with K=55\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 50\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.004     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
      "  0:00:00.007     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 90\n",
      "  0:00:00.007     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 89.9999\n",
      "  0:00:00.007     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.007     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 2500 files using 50 threads. This might take a while.\n",
      "  0:00:00.008     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:00:00.008     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 3.33331 Gb\n",
      "  0:00:00.008     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 16384\n",
      "  0:00:07.851    34G / 34G   INFO    General                 (kmer_splitters.hpp        : 289)   Processed 21996454 reads\n",
      "  0:00:07.851    34G / 34G   INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:11.885   212M / 34G   INFO    General                 (kmer_splitters.hpp        : 308)   Used 21996454 reads\n",
      "  0:00:11.885   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:14.659   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6099280 kmers in total.\n",
      "  0:00:14.660   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:14.921   200M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:14.921   200M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:14.921   200M / 34G   INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 800 files using 50 threads. This might take a while.\n",
      "  0:00:14.922   200M / 34G   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:00:14.922   200M / 34G   INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 3.33203 Gb\n",
      "  0:00:14.922   200M / 34G   INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 41943\n",
      "  0:00:19.644    27G / 34G   INFO    General                 (kmer_splitters.hpp        : 380)   Processed 6099280 kmers\n",
      "  0:00:19.645    27G / 34G   INFO    General                 (kmer_splitters.hpp        : 385)   Used 6099280 kmers.\n",
      "  0:00:22.175   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:22.837   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6107735 kmers in total.\n",
      "  0:00:22.838   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:23.012   204M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:23.166   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:23.253   204M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2840440 bytes occupied (3.72045 bits per kmer).\n",
      "  0:00:23.257   212M / 34G   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:23.390   212M / 34G   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:23.395   208M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:23.588   208M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:23.749   216M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 148420 sequences extracted\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:23.847   216M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:23.882   216M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 1 loops collected\n",
      "  0:00:23.956   216M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:23.956   216M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:23.956   216M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:24.086   216M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2854304 bytes occupied (3.74379 bits per kmer).\n",
      "  0:00:24.096   240M / 34G   INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:26.628   240M / 34G   INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:26.718   240M / 34G   INFO    General                 (construction.cpp          : 464)   Processed 296841 edges\n",
      "  0:00:26.755   204M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:26.755   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 8\n",
      "  0:00:26.755   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 70\n",
      "  0:00:26.755   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 70. Coverage mad: 29.652\n",
      "  0:00:26.755   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:00:26.844   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:00:27.100   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:00:27.845   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:00:29.103   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:00:29.888   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 70.0199. Fitted coverage std. dev: 26.7788\n",
      "  0:00:29.890   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.270451\n",
      "  0:00:29.890   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 20\n",
      "  0:00:29.890   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 20\n",
      "  0:00:29.890   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 5138731\n",
      "  0:00:29.890   204M / 34G   INFO    General                 (genomic_info_filler.cpp   : 112)   Mean coverage was calculated as 70.0199\n",
      "  0:00:29.890   204M / 34G   INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 20\n",
      "  0:00:29.890   204M / 34G   INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:29.890   204M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:29.890   204M / 34G   INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:29.891   204M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:29.891   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 800 files using 50 threads. This might take a while.\n",
      "  0:00:29.892   204M / 34G   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:00:29.892   204M / 34G   INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 3.33201 Gb\n",
      "  0:00:29.892   204M / 34G   INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 41943\n",
      "  0:00:34.153    27G / 34G   INFO    General                 (edge_index_builders.hpp   :  77)   Processed 296841 edges\n",
      "  0:00:34.154    27G / 34G   INFO    General                 (edge_index_builders.hpp   :  82)   Used 296841 sequences.\n",
      "  0:00:36.574   208M / 34G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:37.429   208M / 34G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6099280 kmers in total.\n",
      "  0:00:37.430   208M / 34G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:37.574   208M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:37.720   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:37.817   212M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2838030 bytes occupied (3.72245 bits per kmer).\n",
      "  0:00:37.873   352M / 34G   INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:37.968   348M / 34G   INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:37.969   348M / 34G   INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:38.123   364M / 34G   INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:39.159   364M / 34G   INFO    General                 (gap_closer.cpp            : 138)   Used 5498906 paired reads\n",
      "  0:00:39.159   364M / 34G   INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:39.197   352M / 34G   INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:39.373   352M / 34G   INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 8 gaps after checking 249 candidates\n",
      "  0:00:39.394   348M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:39.395   204M / 34G   INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:39.395   204M / 34G   INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:39.395   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:39.404   208M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:39.404   208M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
      "  0:00:40.252   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 66988 times\n",
      "  0:00:40.252   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
      "  0:00:40.275   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 168 times\n",
      "  0:00:40.275   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
      "  0:00:40.339   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 9944 times\n",
      "  0:00:40.340   212M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:40.340   212M / 34G   INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:40.340   212M / 34G   INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:40.340   212M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:40.340   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:40.344   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 15 times\n",
      "  0:00:40.344   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:40.624   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 957 times\n",
      "  0:00:40.624   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:40.629   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:40.629   212M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:40.629   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:40.630   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:40.630   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:40.630   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:40.630   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:40.636   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 40 times\n",
      "  0:00:40.636   212M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:40.636   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:40.636   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 5 times\n",
      "  0:00:40.636   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:40.648   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 34 times\n",
      "  0:00:40.648   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:40.651   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 12 times\n",
      "  0:00:40.651   212M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:40.651   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:40.651   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:40.651   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:40.654   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 4 times\n",
      "  0:00:40.654   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:40.654   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
      "  0:00:40.654   212M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:40.654   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:40.654   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:40.654   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:40.654   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:40.654   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:40.655   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
      "  0:00:40.655   212M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:40.655   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:40.655   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:40.655   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:40.656   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2 times\n",
      "  0:00:40.656   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:40.656   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
      "  0:00:40.656   212M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:40.656   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:40.656   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:40.656   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:40.656   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:40.656   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
      "  0:00:40.657   212M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:40.657   212M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:40.657   212M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:40.657   212M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:40.657   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:40.663   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:40.663   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:40.669   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 5 times\n",
      "  0:00:40.669   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:40.670   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:40.670   212M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:40.670   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:40.670   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:40.670   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:40.670   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:40.670   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:40.670   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:40.670   212M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:40.670   212M / 34G   INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:40.671   212M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:40.671   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 800 files using 50 threads. This might take a while.\n",
      "  0:00:40.672   212M / 34G   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:00:40.672   212M / 34G   INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 3.33195 Gb\n",
      "  0:00:40.672   212M / 34G   INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 41943\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:44.897    27G / 34G   INFO    General                 (edge_index_builders.hpp   :  77)   Processed 2419 edges\n",
      "  0:00:44.897    27G / 34G   INFO    General                 (edge_index_builders.hpp   :  82)   Used 2419 sequences.\n",
      "  0:00:47.264   216M / 34G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:49.196   216M / 34G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 5284628 kmers in total.\n",
      "  0:00:49.197   216M / 34G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:49.360   216M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:49.496   216M / 34G   INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:49.604   216M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2458784 bytes occupied (3.72217 bits per kmer).\n",
      "  0:00:49.679   340M / 34G   INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:49.960   336M / 34G   INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:49.961   336M / 34G   INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:49.963   336M / 34G   INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:50.936   336M / 34G   INFO    General                 (gap_closer.cpp            : 138)   Used 5498906 paired reads\n",
      "  0:00:50.936   336M / 34G   INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:50.938   336M / 34G   INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:50.940   336M / 34G   INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 9 gaps after checking 106 candidates\n",
      "  0:00:51.296   336M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:51.296   336M / 34G   INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:51.296   336M / 34G   INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:51.296   336M / 34G   INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:51.296   336M / 34G   INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:51.296   336M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:51.297   336M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:51.297   336M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:51.299   336M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:51.299   336M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:51.299   336M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:51.299   336M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:51.301   336M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:51.301   336M / 34G   INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:51.302   336M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:51.305   336M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 33 times\n",
      "  0:00:51.305   336M / 34G   INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:51.305   336M / 34G   INFO    General                 (simplification.cpp        : 476)   Average coverage = 72.738\n",
      "  0:00:51.305   336M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:51.305   336M / 34G   INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 132740 kmers to process\n",
      "  0:00:51.352   336M / 34G   INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:54.125   360M / 34G   INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:54.251   340M / 34G   INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:54.251   340M / 34G   INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 4 nucleotides\n",
      "  0:00:54.253   336M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:54.253   336M / 34G   INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K55/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:54.300   336M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K55/before_rr.fasta\n",
      "  0:00:54.375   336M / 34G   INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K55/assembly_graph.fastg\n",
      "  0:00:54.601   336M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K55/simplified_contigs.fasta\n",
      "  0:00:54.653   336M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K55/intermediate_contigs.fasta\n",
      "  0:00:54.704   336M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:54.707   336M / 34G   INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 132922 kmers to process\n",
      "  0:00:54.753   336M / 34G   INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:54.754   336M / 34G   INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 58171\n",
      "  0:00:54.754   336M / 34G   INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:54.754   336M / 34G   INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:55.007     1G / 34G   INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:58.409     1G / 34G   INFO    General                 (sequence_mapper_notifier.h:  98)   Total 5498906 reads processed\n",
      "  0:00:59.690     1G / 34G   INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:59.690     1G / 34G   INFO    General                 (pair_info_count.cpp       : 213)   2636627 paired reads (47.9482% of all) aligned to long edges\n",
      "  0:00:59.694   336M / 34G   INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 469.025, deviation = 14.6939, left quantile = 450, right quantile = 486, read length = 90\n",
      "  0:00:59.746   528M / 34G   INFO    General                 (pair_info_count.cpp       : 371)   Filtering data for library #0\n",
      "  0:00:59.747   528M / 34G   INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:01:02.092   528M / 34G   INFO    General                 (sequence_mapper_notifier.h:  98)   Total 5498906 reads processed\n",
      "  0:01:02.093   528M / 34G   INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:01:02.093   528M / 34G   INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:01:02.093   528M / 34G   INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 450, right insert size quantile 486, filtering threshold 2, rounding threshold 0\n",
      "  0:01:02.104   540M / 34G   INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:04.397   540M / 34G   INFO    General                 (sequence_mapper_notifier.h:  98)   Total 5498906 reads processed\n",
      "  0:01:04.404   336M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:01:04.404   336M / 34G   INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:01:04.404   336M / 34G   INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:01:04.404   336M / 34G   INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:01:04.413   336M / 34G   INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:01:04.413   336M / 34G   INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 10863\n",
      "  0:01:04.417   336M / 34G   INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:01:04.417   336M / 34G   INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:01:04.419   336M / 34G   INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:01:04.419   336M / 34G   INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:01:04.432   336M / 34G   INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 1692; contradictional = 204\n",
      "  0:01:04.445   336M / 34G   INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 82; contradictional = 42\n",
      "  0:01:04.445   336M / 34G   INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:01:04.445   336M / 34G   INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:01:04.450   336M / 34G   INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:01:04.450   336M / 34G   INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 2511\n",
      "  0:01:04.451   336M / 34G   INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:01:04.451   336M / 34G   INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:01:04.453   336M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:01:04.453   336M / 34G   INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:01:04.453   336M / 34G   INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:01:04.457   340M / 34G   INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:01:04.457   340M / 34G   INFO    General                 (extenders_logic.cpp       : 275)   Estimated coverage of library #0 is 72.738\n",
      "  0:01:04.458   340M / 34G   INFO    General                 (extenders_logic.cpp       : 275)   Estimated coverage of library #0 is 72.738\n",
      "  0:01:04.458   340M / 34G   INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:01:04.458   340M / 34G   INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:01:04.458   340M / 34G   INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:01:04.458   340M / 34G   INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:01:04.458   340M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 1126 (0%)\n",
      "  0:01:04.547   344M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 113 paths from 1126 (10%)\n",
      "  0:01:04.553   344M / 34G   INFO    General                 (path_extender.hpp         : 883)   Processed 128 paths from 1126 (11%)\n",
      "  0:01:04.598   348M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 226 paths from 1126 (20%)\n",
      "  0:01:04.602   348M / 34G   INFO    General                 (path_extender.hpp         : 883)   Processed 256 paths from 1126 (22%)\n",
      "  0:01:04.627   348M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 339 paths from 1126 (30%)\n",
      "  0:01:04.639   348M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 452 paths from 1126 (40%)\n",
      "  0:01:04.644   348M / 34G   INFO    General                 (path_extender.hpp         : 883)   Processed 512 paths from 1126 (45%)\n",
      "  0:01:04.645   348M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 565 paths from 1126 (50%)\n",
      "  0:01:04.649   348M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 678 paths from 1126 (60%)\n",
      "  0:01:04.652   348M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 791 paths from 1126 (70%)\n",
      "  0:01:04.660   348M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 904 paths from 1126 (80%)\n",
      "  0:01:04.661   348M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 1017 paths from 1126 (90%)\n",
      "  0:01:04.661   348M / 34G   INFO    General                 (path_extender.hpp         : 883)   Processed 1024 paths from 1126 (90%)\n",
      "  0:01:04.661   348M / 34G   INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:01:04.661   348M / 34G   INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:01:04.663   348M / 34G   INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:01:04.663   348M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:01:04.663   348M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:01:04.663   348M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:01:04.663   348M / 34G   INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:01:04.667   348M / 34G   INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:01:04.670   348M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:01:04.673   348M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:01:04.674   348M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:01:04.675   348M / 34G   INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:01:04.675   348M / 34G   INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:01:04.676   348M / 34G   INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:01:04.677   348M / 34G   INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:01:04.681   348M / 34G   INFO    General                 (launcher.cpp              : 296)   Traversed 1 loops\n",
      "  0:01:04.681   348M / 34G   INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:01:04.681   348M / 34G   INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:01:04.681   348M / 34G   INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:01:04.681   348M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:01:04.681   348M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:01:04.681   348M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:01:04.681   348M / 34G   INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:01:04.682   348M / 34G   INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:01:04.682   348M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:01:04.683   348M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:01:04.683   348M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:01:04.683   348M / 34G   INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:01:04.685   352M / 34G   INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:01:04.693   340M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:01:04.693   340M / 34G   INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K55/assembly_graph_with_scaffolds.gfa\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:04.739   340M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K55/before_rr.fasta\n",
      "  0:01:04.943   340M / 34G   INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K55/assembly_graph.fastg\n",
      "  0:01:05.202   340M / 34G   INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K55/final_contigs.paths\n",
      "  0:01:05.332   340M / 34G   INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/analytics/projects/bioinf_workshop/data/week3/assembly//K55/scaffolds.paths\n",
      "  0:01:05.531   340M / 34G   INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:01:05.607   204M / 34G   INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 1 minutes 5 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 21, 33, 55 \n",
      "\n",
      " * Corrected reads are in /home/analytics/projects/bioinf_workshop/data/week3/assembly/corrected/\n",
      " * Assembled contigs are in /home/analytics/projects/bioinf_workshop/data/week3/assembly/contigs.fasta\n",
      " * Assembled scaffolds are in /home/analytics/projects/bioinf_workshop/data/week3/assembly/scaffolds.fasta\n",
      " * Assembly graph is in /home/analytics/projects/bioinf_workshop/data/week3/assembly/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/analytics/projects/bioinf_workshop/data/week3/assembly/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/analytics/projects/bioinf_workshop/data/week3/assembly/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/analytics/projects/bioinf_workshop/data/week3/assembly/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/analytics/projects/bioinf_workshop/data/week3/assembly/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n"
     ]
    }
   ],
   "source": [
    "!mkdir -p ../data/week3/assembly\n",
    "!$SPADES -t 50 -m 500 -o ../data/week3/assembly -1 ../data/week3/SRR292678sub_S1_L001_R1_001.fastq -2 ../data/week3/SRR292678sub_S1_L001_R2_001.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/analytics/distr/quast-5.0.0/quast.py -o ../data/week3/quast ../data/week3/assembly/contigs.fasta ../data/week3/assembly/scaffolds.fasta\n",
      "\n",
      "Version: 5.0.0, de6973bb\n",
      "\n",
      "System information:\n",
      "  OS: Linux-3.10.0-862.3.3.el7.x86_64-x86_64-with-centos-7.5.1804-Core (linux_64)\n",
      "  Python version: 3.5.4\n",
      "  CPUs number: 56\n",
      "\n",
      "Started: 2018-10-12 11:47:07\n",
      "\n",
      "Logging to /home/analytics/projects/bioinf_workshop/data/week3/quast/quast.log\n",
      "NOTICE: Maximum number of threads is set to 14 (use --threads option to set it manually)\n",
      "\n",
      "CWD: /home/analytics/projects/bioinf_workshop/notebooks\n",
      "Main parameters: \n",
      "  MODE: default, threads: 14, minimum contig length: 500, minimum alignment length: 65, \\\n",
      "  ambiguity: one, threshold for extensive misassembly size: 1000\n",
      "\n",
      "Contigs:\n",
      "  Pre-processing...\n",
      "  1  ../data/week3/assembly/contigs.fasta ==> contigs\n",
      "  2  ../data/week3/assembly/scaffolds.fasta ==> scaffolds\n",
      "\n",
      "2018-10-12 11:47:08\n",
      "Running Basic statistics processor...\n",
      "  Contig files: \n",
      "    1  contigs\n",
      "    2  scaffolds\n",
      "  Calculating N50 and L50...\n",
      "    1  contigs, N50 = 105346, L50 = 15, Total length = 5243761, GC % = 50.53, # N's per 100 kbp =  0.00\n",
      "    2  scaffolds, N50 = 105346, L50 = 15, Total length = 5252701, GC % = 50.51, # N's per 100 kbp =  32.23\n",
      "  Drawing Nx plot...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/basic_stats/Nx_plot.pdf\n",
      "  Drawing cumulative plot...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/basic_stats/cumulative_plot.pdf\n",
      "  Drawing GC content plot...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/basic_stats/GC_content_plot.pdf\n",
      "  Drawing contigs GC content plot...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/basic_stats/contigs_GC_content_plot.pdf\n",
      "  Drawing scaffolds GC content plot...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/basic_stats/scaffolds_GC_content_plot.pdf\n",
      "  Drawing Coverage histogram (bin size: 1x)...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/basic_stats/coverage_histogram.pdf\n",
      "  Drawing contigs coverage histogram (bin size: 1x)...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/basic_stats/contigs_coverage_histogram.pdf\n",
      "  Drawing scaffolds coverage histogram (bin size: 1x)...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/basic_stats/scaffolds_coverage_histogram.pdf\n",
      "Done.\n",
      "\n",
      "NOTICE: Genes are not predicted by default. Use --gene-finding or --glimmer option to enable it.\n",
      "\n",
      "2018-10-12 11:47:10\n",
      "Creating large visual summaries...\n",
      "This may take a while: press Ctrl-C to skip this step..\n",
      "  1 of 2: Creating Icarus viewers...\n",
      "  2 of 2: Creating PDF with all tables and plots...\n",
      "Done\n",
      "\n",
      "2018-10-12 11:47:11\n",
      "RESULTS:\n",
      "  Text versions of total report are saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/report.txt, report.tsv, and report.tex\n",
      "  Text versions of transposed total report are saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/transposed_report.txt, transposed_report.tsv, and transposed_report.tex\n",
      "  HTML version (interactive tables and plots) is saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/report.html\n",
      "  PDF version (tables and plots) is saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/report.pdf\n",
      "  Icarus (contig browser) is saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/icarus.html\n",
      "  Log is saved to /home/analytics/projects/bioinf_workshop/data/week3/quast/quast.log\n",
      "\n",
      "Finished: 2018-10-12 11:47:11\n",
      "Elapsed time: 0:00:04.491883\n",
      "NOTICEs: 2; WARNINGs: 0; non-fatal ERRORs: 0\n",
      "\n",
      "Thank you for using QUAST!\n"
     ]
    }
   ],
   "source": [
    "!mkdir -p ../data/week3/quast\n",
    "!$QUAST -o ../data/week3/quast ../data/week3/assembly/contigs.fasta ../data/week3/assembly/scaffolds.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "basic_stats\tquast.log    report.tex  transposed_report.tex\r\n",
      "icarus.html\treport.html  report.tsv  transposed_report.tsv\r\n",
      "icarus_viewers\treport.pdf   report.txt  transposed_report.txt\r\n"
     ]
    }
   ],
   "source": [
    "!ls "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <iframe\n",
       "            width=\"1000\"\n",
       "            height=\"600\"\n",
       "            src=\"http://bioinf-workshop.s3-website.eu-central-1.amazonaws.com/week_3/quast_1.html\"\n",
       "            frameborder=\"0\"\n",
       "            allowfullscreen\n",
       "        ></iframe>\n",
       "        "
      ],
      "text/plain": [
       "<IPython.lib.display.IFrame at 0x7f9a0ac2ec18>"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = perform_file_upload(open(\"../data/week3/quast/report.html\", \"r\").read(), \"week_3/quast_1.html\")\n",
    "IFrame(src=f, width=1000, height=600)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Command line: /home/analytics/distr/SPAdes-3.12.0-Linux/bin/spades.py\t-t\t50\t-m\t700\t-o\t/home/analytics/projects/bioinf_workshop/data/week3/assembly_all\t--pe1-1\t/home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R1_001.fastq\t--pe1-2\t/home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R2_001.fastq\t--mp1-1\t/home/analytics/projects/bioinf_workshop/data/week3/SRR292862_S2_L001_R1_001.fastq\t--mp1-2\t/home/analytics/projects/bioinf_workshop/data/week3/SRR292862_S2_L001_R2_001.fastq\t--mp2-1\t/home/analytics/projects/bioinf_workshop/data/week3/SRR292770_S1_L001_R1_001.fastq\t--mp2-2\t/home/analytics/projects/bioinf_workshop/data/week3/SRR292770_S1_L001_R2_001.fastq\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.12.0\n",
      "  Python version: 3.5.4\n",
      "  OS: Linux-3.10.0-862.3.3.el7.x86_64-x86_64-with-centos-7.5.1804-Core\n",
      "\n",
      "Output dir: /home/analytics/projects/bioinf_workshop/data/week3/assembly_all\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)\n",
      "  Reads:\n",
      "    Library number: 1, library type: mate-pairs\n",
      "      orientation: rf\n",
      "      left reads: ['/home/analytics/projects/bioinf_workshop/data/week3/SRR292862_S2_L001_R1_001.fastq']\n",
      "      right reads: ['/home/analytics/projects/bioinf_workshop/data/week3/SRR292862_S2_L001_R2_001.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "    Library number: 2, library type: mate-pairs\n",
      "      orientation: rf\n",
      "      left reads: ['/home/analytics/projects/bioinf_workshop/data/week3/SRR292770_S1_L001_R1_001.fastq']\n",
      "      right reads: ['/home/analytics/projects/bioinf_workshop/data/week3/SRR292770_S1_L001_R2_001.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "    Library number: 3, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R1_001.fastq']\n",
      "      right reads: ['/home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R2_001.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: automatic selection based on read length\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/tmp\n",
      "  Threads: 50\n",
      "  Memory limit (in Gb): 700\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/analytics/distr/SPAdes-3.12.0-Linux/bin/spades-hammer /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/corrected/configs/config.info\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 50\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 700 Gb\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.002     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 800 files using 50 threads. This might take a while.\n",
      "  0:00:00.003     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.003     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 4.66664 Gb\n",
      "  0:00:00.003     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 83886\n",
      "  0:00:03.941    27G / 27G   INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292862_S2_L001_R1_001.fastq\n",
      "  0:00:14.602    27G / 27G   INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 5102041 reads\n",
      "  0:00:14.602    27G / 27G   INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292862_S2_L001_R2_001.fastq\n",
      "  0:00:24.816    27G / 27G   INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 10204082 reads\n",
      "  0:00:24.817    27G / 27G   INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292770_S1_L001_R1_001.fastq\n",
      "  0:00:35.100    27G / 27G   INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 15306123 reads\n",
      "  0:00:35.100    27G / 27G   INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292770_S1_L001_R2_001.fastq\n",
      "  0:00:45.466    27G / 27G   INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 20408164 reads\n",
      "  0:00:45.466    27G / 27G   INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R1_001.fastq\n",
      "  0:01:00.087    27G / 28G   INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 25907510 reads\n",
      "  0:01:00.087    27G / 28G   INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R2_001.fastq\n",
      "  0:01:14.635    27G / 28G   INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 31406856 reads\n",
      "  0:01:14.635    27G / 28G   INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 31406856 reads processed\n",
      "  0:01:16.997   200M / 28G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:01:17.957   200M / 28G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 169592000 kmers in total.\n",
      "  0:01:17.957   200M / 28G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:01:19.200   200M / 28G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:01:23.422   272M / 28G   INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:01:24.756   272M / 28G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 78648912 bytes occupied (3.71003 bits per kmer).\n",
      "  0:01:24.760   268M / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:01:29.585     2G / 28G   INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:02:22.486     2G / 28G   INFO    General                 (main.cpp                  : 155)   Extracting clusters\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:03:04.034     2G / 28G   INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 112762733\n",
      "  0:03:04.038     1G / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:03:05.360     5G / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292862_S2_L001_R1_001.fastq\n",
      "  0:03:15.238     5G / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292862_S2_L001_R2_001.fastq\n",
      "  0:03:24.981     5G / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292770_S1_L001_R1_001.fastq\n",
      "  0:03:34.885     5G / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292770_S1_L001_R2_001.fastq\n",
      "  0:03:44.692     5G / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R1_001.fastq\n",
      "  0:04:00.207     5G / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R2_001.fastq\n",
      "  0:04:15.563     5G / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:04:16.207     5G / 28G   INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 169592000 kmers in total. Among them 148258856 (87.4209%) are singletons.\n",
      "  0:04:16.207     5G / 28G   INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:04:30.144     5G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 208 non-read kmers were generated.\n",
      "  0:04:30.144     5G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:04:30.144     5G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 100545802. Among them 58761900 (58.4429%) are good\n",
      "  0:04:30.144     5G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 20579. Among them 20294 (98.6151%) are good\n",
      "  0:04:30.144     5G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 12464199. Among them 11339972 (90.9804%) are good\n",
      "  0:04:30.144     5G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 5.53958 kmers\n",
      "  0:04:30.144     5G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.02192\n",
      "  0:04:30.144     5G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 70122166\n",
      "  0:04:30.144     5G / 28G   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.935523,0.0317479,0.0205961,0.0121332),(0.0179321,0.955346,0.0118945,0.0148278),(0.0138628,0.0109861,0.956187,0.0189645),(0.0111441,0.0196266,0.0307645,0.938465))\n",
      "  0:04:30.337     5G / 28G   INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:04:30.337     5G / 28G   INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 50 threads.\n",
      "  0:05:33.752     5G / 28G   INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 11858697 new k-mers.\n",
      "  0:06:36.641     5G / 28G   INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 175449 new k-mers.\n",
      "  0:07:39.043     5G / 28G   INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 878 new k-mers.\n",
      "  0:08:41.739     5G / 28G   INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 3 produced 0 new k-mers.\n",
      "  0:08:41.740     5G / 28G   INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:08:41.740     5G / 28G   INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 50 threads.\n",
      "  0:08:41.740     5G / 28G   INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/analytics/projects/bioinf_workshop/data/week3/SRR292862_S2_L001_R1_001.fastq and /home/analytics/projects/bioinf_workshop/data/week3/SRR292862_S2_L001_R2_001.fastq\n",
      "  0:08:51.301     7G / 28G   INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 5000000 reads.\n",
      "  0:08:57.587     9G / 28G   INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:09:03.239     9G / 28G   INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:09:03.413     9G / 28G   INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 1 of 102041 reads.\n",
      "  0:09:03.559     9G / 28G   INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 1\n",
      "  0:09:03.663     9G / 28G   INFO    General                 (hammer_tools.cpp          : 185)   Written batch 1\n",
      "  0:09:06.098     5G / 28G   INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/analytics/projects/bioinf_workshop/data/week3/SRR292770_S1_L001_R1_001.fastq and /home/analytics/projects/bioinf_workshop/data/week3/SRR292770_S1_L001_R2_001.fastq\n",
      "  0:09:16.016     7G / 28G   INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 5000000 reads.\n",
      "  0:09:22.289     9G / 28G   INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:09:27.721     9G / 28G   INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:09:27.879     9G / 28G   INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 1 of 102041 reads.\n",
      "  0:09:28.049     9G / 28G   INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 1\n",
      "  0:09:28.152     9G / 28G   INFO    General                 (hammer_tools.cpp          : 185)   Written batch 1\n",
      "  0:09:30.864     5G / 28G   INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R1_001.fastq and /home/analytics/projects/bioinf_workshop/data/week3/SRR292678sub_S1_L001_R2_001.fastq\n",
      "  0:09:46.054     8G / 28G   INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 5000000 reads.\n",
      "  0:09:57.299     9G / 28G   INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:10:04.408     9G / 28G   INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:10:05.888     9G / 28G   INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 1 of 499346 reads.\n",
      "  0:10:07.026     9G / 28G   INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 1\n",
      "  0:10:07.655     9G / 28G   INFO    General                 (hammer_tools.cpp          : 185)   Written batch 1\n",
      "  0:10:11.077     5G / 28G   INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 1341560 bases in 1180772 reads.\n",
      "  0:10:11.077     5G / 28G   INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 1989880233.\n",
      "  0:10:11.103   200M / 28G   INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/corrected/corrected.yaml\n",
      "  0:10:11.104   200M / 28G   INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "== Dataset description file was created: /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K21\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/K21/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 700 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/analytics/projects/bioinf_workshop/data/week3/assembly_all/dataset.info) with K=21\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 50\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.001     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #2 (takes a while)\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.236    80M / 80M   INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.403    88M / 88M   INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:00:00.723   112M / 112M  INFO    General                 (binary_converter.hpp      :  93)   65536 reads processed\n",
      "  0:00:01.216   156M / 156M  INFO    General                 (binary_converter.hpp      :  93)   131072 reads processed\n",
      "  0:00:02.200   244M / 244M  INFO    General                 (binary_converter.hpp      :  93)   262144 reads processed\n",
      "  0:00:04.165   424M / 424M  INFO    General                 (binary_converter.hpp      :  93)   524288 reads processed\n",
      "  0:00:10.269   704M / 704M  INFO    General                 (binary_converter.hpp      :  93)   1048576 reads processed\n",
      "  0:00:20.291   704M / 704M  INFO    General                 (binary_converter.hpp      :  93)   2097152 reads processed\n",
      "  0:00:40.357   704M / 704M  INFO    General                 (binary_converter.hpp      :  93)   4194304 reads processed\n",
      "  0:00:54.108   640M / 704M  INFO    General                 (binary_converter.hpp      : 117)   5498917 reads written\n",
      "  0:00:55.214     4M / 704M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:55.314   132M / 704M  INFO    General                 (binary_converter.hpp      : 117)   406 reads written\n",
      "  0:00:55.329     4M / 704M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:55.426   132M / 704M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:55.504     4M / 704M  INFO    General                 (construction.cpp          : 111)   Max read length 90\n",
      "  0:00:55.504     4M / 704M  INFO    General                 (construction.cpp          : 117)   Average read length 89.9999\n",
      "  0:00:55.504     4M / 704M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:55.504     4M / 704M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 2500 files using 50 threads. This might take a while.\n",
      "  0:00:55.506     4M / 704M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:00:55.506     4M / 704M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 4.66664 Gb\n",
      "  0:00:55.506     4M / 704M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 26843\n",
      "  0:01:03.666    27G / 27G   INFO    General                 (kmer_splitters.hpp        : 289)   Processed 21996480 reads\n",
      "  0:01:03.666    27G / 27G   INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:01:05.863   208M / 27G   INFO    General                 (kmer_splitters.hpp        : 308)   Used 21996480 reads\n",
      "  0:01:05.863   208M / 27G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:01:08.523   208M / 27G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 7657488 kmers in total.\n",
      "  0:01:08.523   208M / 27G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:01:08.729   200M / 27G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:01:08.729   200M / 27G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:01:08.729   200M / 27G   INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 800 files using 50 threads. This might take a while.\n",
      "  0:01:08.730   200M / 27G   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:01:08.730   200M / 27G   INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 4.66536 Gb\n",
      "  0:01:08.730   200M / 27G   INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 83886\n",
      "  0:01:13.548    27G / 27G   INFO    General                 (kmer_splitters.hpp        : 380)   Processed 7657488 kmers\n",
      "  0:01:13.548    27G / 27G   INFO    General                 (kmer_splitters.hpp        : 385)   Used 7657488 kmers.\n",
      "  0:01:16.057   204M / 27G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:01:16.822   204M / 27G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 7614760 kmers in total.\n",
      "  0:01:16.823   204M / 27G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:01:16.955   204M / 27G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:01:17.176   204M / 27G   INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:01:17.239   204M / 27G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 3539280 bytes occupied (3.71834 bits per kmer).\n",
      "  0:01:17.245   212M / 27G   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:01:17.406   212M / 27G   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:01:17.413   208M / 27G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
      "  0:01:17.413   208M / 27G   INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
      "  0:01:17.413   208M / 27G   INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:01:17.933   208M / 27G   INFO   Early tip clipping       (early_simplification.hpp  : 184)   743625 22-mers were removed by early tip clipper\n",
      "  0:01:17.933   208M / 27G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:01:17.938   208M / 27G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:18.117   216M / 27G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 159211 sequences extracted\n",
      "  0:01:18.232   216M / 27G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:01:18.276   216M / 27G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 5 loops collected\n",
      "  0:01:18.354   212M / 27G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:01:18.354   212M / 27G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:01:18.354   212M / 27G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:01:18.495   212M / 27G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 3576624 bytes occupied (3.7366 bits per kmer).\n",
      "  0:01:18.506   244M / 27G   INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:01:22.831   244M / 27G   INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:01:22.941   244M / 27G   INFO    General                 (construction.cpp          : 464)   Processed 318390 edges\n",
      "  0:01:22.972   204M / 27G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:01:22.972   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 16\n",
      "  0:01:22.972   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 142\n",
      "  0:01:22.972   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 142. Coverage mad: 51.891\n",
      "  0:01:22.972   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:01:23.128   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:01:23.542   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:01:24.812   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:01:28.004   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:01:33.396   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 32\n",
      "  0:01:45.707   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 64\n",
      "  0:01:51.672   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 137.461. Fitted coverage std. dev: 48.8921\n",
      "  0:01:51.677   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.202337\n",
      "  0:01:51.677   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 28\n",
      "  0:01:51.677   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 28\n",
      "  0:01:51.677   204M / 27G   INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 5200488\n",
      "  0:01:51.677   204M / 27G   INFO    General                 (genomic_info_filler.cpp   : 112)   Mean coverage was calculated as 137.461\n",
      "  0:01:51.677   204M / 27G   INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 28\n",
      "  0:01:51.677   204M / 27G   INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:01:51.677   204M / 27G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:01:51.677   204M / 27G   INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:01:51.677   204M / 27G   INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:01:51.677   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:01:51.688   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:01:51.688   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
      "  0:01:51.707   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 619 times\n",
      "  0:01:51.707   204M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
      "  0:01:52.583   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 42824 times\n",
      "  0:01:52.584   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
      "  0:01:52.612   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 7844 times\n",
      "  0:01:52.615   208M / 27G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:01:52.615   208M / 27G   INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:01:52.615   208M / 27G   INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:01:52.615   208M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:01:52.615   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.620   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 70 times\n",
      "  0:01:52.620   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:52.906   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 6480 times\n",
      "  0:01:52.906   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:52.916   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 66 times\n",
      "  0:01:52.916   208M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:01:52.917   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.917   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:01:52.917   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:52.920   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 35 times\n",
      "  0:01:52.920   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:52.925   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 183 times\n",
      "  0:01:52.925   208M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:01:52.925   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.925   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:52.925   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:52.931   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 111 times\n",
      "  0:01:52.931   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:52.932   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 36 times\n",
      "  0:01:52.932   208M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:01:52.932   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.932   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:52.932   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:52.934   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 38 times\n",
      "  0:01:52.934   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:52.935   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 18 times\n",
      "  0:01:52.935   208M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:01:52.935   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.935   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:52.935   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:52.935   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 19 times\n",
      "  0:01:52.935   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:52.935   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 11 times\n",
      "  0:01:52.935   208M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:01:52.935   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.935   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:52.935   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:52.935   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2 times\n",
      "  0:01:52.935   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 7 times\n",
      "  0:01:52.936   208M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
      "  0:01:52.936   208M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
      "  0:01:52.936   208M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:01:52.936   208M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:01:52.936   208M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:01:52.936   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.940   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:52.940   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:52.959   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 7 times\n",
      "  0:01:52.959   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:52.959   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:01:52.960   208M / 27G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:01:52.960   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.960   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:52.960   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:52.960   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:01:52.960   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:01:52.960   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:01:52.960   208M / 27G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:01:52.960   208M / 27G   INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:01:52.960   208M / 27G   INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:01:52.960   208M / 27G   INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:01:52.960   208M / 27G   INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:01:52.960   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.961   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:52.961   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:52.973   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:01:52.973   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:01:52.973   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:01:52.973   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:01:52.985   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:01:52.985   208M / 27G   INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:01:52.988   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:01:52.993   208M / 27G   INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 309 times\n",
      "  0:01:52.993   208M / 27G   INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:01:52.994   208M / 27G   INFO    General                 (simplification.cpp        : 476)   Average coverage = 145.245\n",
      "  0:01:52.994   208M / 27G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:01:52.994   208M / 27G   INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K21/assembly_graph_with_scaffolds.gfa\n",
      "  0:01:53.067   208M / 27G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K21/before_rr.fasta\n",
      "  0:01:53.145   208M / 27G   INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K21/assembly_graph.fastg\n",
      "  0:01:53.418   208M / 27G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K21/simplified_contigs.fasta\n",
      "  0:01:53.496   208M / 27G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K21/final_contigs.fasta\n",
      "  0:01:53.573   208M / 27G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:01:53.573   208M / 27G   INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K21/assembly_graph_with_scaffolds.gfa\n",
      "  0:01:53.637   208M / 27G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K21/before_rr.fasta\n",
      "  0:01:53.749   208M / 27G   INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K21/assembly_graph.fastg\n",
      "  0:01:54.057   208M / 27G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K21/simplified_contigs.fasta\n",
      "  0:01:54.137   208M / 27G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K21/final_contigs.fasta\n",
      "  0:01:54.219   208M / 27G   INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:01:54.242   204M / 27G   INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 1 minutes 54 seconds\n",
      "Max read length detected as 90\n",
      "\n",
      "== Running assembler: K33\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/K33/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 700 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/analytics/projects/bioinf_workshop/data/week3/assembly_all/dataset.info) with K=33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 50\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.001     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
      "  0:00:00.004     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 90\n",
      "  0:00:00.004     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 89.9999\n",
      "  0:00:00.004     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.004     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 2500 files using 50 threads. This might take a while.\n",
      "  0:00:00.006     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:00:00.006     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 4.66664 Gb\n",
      "  0:00:00.006     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 16384\n",
      "  0:00:09.594    34G / 34G   INFO    General                 (kmer_splitters.hpp        : 289)   Processed 21996480 reads\n",
      "  0:00:09.595    34G / 34G   INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:13.897   212M / 34G   INFO    General                 (kmer_splitters.hpp        : 308)   Used 21996480 reads\n",
      "  0:00:13.897   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:17.512   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 7980632 kmers in total.\n",
      "  0:00:17.512   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:17.829   200M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:17.829   200M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:17.829   200M / 34G   INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 800 files using 50 threads. This might take a while.\n",
      "  0:00:17.830   200M / 34G   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:00:17.830   200M / 34G   INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 4.66536 Gb\n",
      "  0:00:17.830   200M / 34G   INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 41943\n",
      "  0:00:22.601    27G / 34G   INFO    General                 (kmer_splitters.hpp        : 380)   Processed 7980632 kmers\n",
      "  0:00:22.601    27G / 34G   INFO    General                 (kmer_splitters.hpp        : 385)   Used 7980632 kmers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:25.389   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:26.098   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 7965125 kmers in total.\n",
      "  0:00:26.098   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:26.288   204M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:26.533   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:26.699   204M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 3701504 bytes occupied (3.71771 bits per kmer).\n",
      "  0:00:26.705   212M / 34G   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:26.873   212M / 34G   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:26.881   208M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
      "  0:00:26.881   208M / 34G   INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:26.881   208M / 34G   INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:00:27.174   208M / 34G   INFO   Early tip clipping       (early_simplification.hpp  : 184)   1411627 34-mers were removed by early tip clipper\n",
      "  0:00:27.174   208M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:27.177   208M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:27.321   212M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 82415 sequences extracted\n",
      "  0:00:27.417   212M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:27.464   212M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 3 loops collected\n",
      "  0:00:27.501   212M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:27.501   212M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:27.501   212M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:27.615   216M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 3734324 bytes occupied (3.74339 bits per kmer).\n",
      "  0:00:27.626   248M / 34G   INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:31.329   244M / 34G   INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:31.443   244M / 34G   INFO    General                 (construction.cpp          : 464)   Processed 164826 edges\n",
      "  0:00:31.489   204M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:31.489   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 13\n",
      "  0:00:31.489   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 116\n",
      "  0:00:31.489   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 116. Coverage mad: 44.478\n",
      "  0:00:31.489   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:00:31.626   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:00:31.991   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:00:33.280   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:00:35.394   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:00:38.870   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 114.251. Fitted coverage std. dev: 40.9346\n",
      "  0:00:38.874   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.516745\n",
      "  0:00:38.874   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 39\n",
      "  0:00:38.874   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 39\n",
      "  0:00:38.874   204M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 5085581\n",
      "  0:00:38.874   204M / 34G   INFO    General                 (genomic_info_filler.cpp   : 112)   Mean coverage was calculated as 114.251\n",
      "  0:00:38.874   204M / 34G   INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 39\n",
      "  0:00:38.874   204M / 34G   INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:38.874   204M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:38.874   204M / 34G   INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:38.874   204M / 34G   INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:38.874   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:38.885   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:38.885   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
      "  0:00:38.904   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 901 times\n",
      "  0:00:38.904   200M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
      "  0:00:39.321   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 19325 times\n",
      "  0:00:39.321   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
      "  0:00:39.352   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 8402 times\n",
      "  0:00:39.353   204M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:39.353   204M / 34G   INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:39.353   204M / 34G   INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:39.353   204M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:39.353   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.356   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 158 times\n",
      "  0:00:39.356   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.576   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 3902 times\n",
      "  0:00:39.576   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:39.582   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 111 times\n",
      "  0:00:39.582   204M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:39.582   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.583   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 12 times\n",
      "  0:00:39.583   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.591   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 93 times\n",
      "  0:00:39.591   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:39.593   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 52 times\n",
      "  0:00:39.593   204M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:39.593   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.593   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:39.593   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.597   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 39 times\n",
      "  0:00:39.597   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:39.598   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 17 times\n",
      "  0:00:39.598   204M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:39.598   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.598   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:39.598   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.598   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 10 times\n",
      "  0:00:39.598   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:39.598   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
      "  0:00:39.598   204M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:39.598   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.598   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:39.598   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.598   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:39.598   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
      "  0:00:39.599   204M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:39.599   204M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
      "  0:00:39.599   204M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
      "  0:00:39.599   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
      "  0:00:39.600   204M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
      "  0:00:39.600   204M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:39.600   204M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:39.600   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.601   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:39.601   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.602   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 5 times\n",
      "  0:00:39.602   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:39.603   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:39.603   204M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:39.603   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.603   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:39.603   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.603   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:39.603   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:39.603   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:39.603   204M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:39.603   204M / 34G   INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:39.603   204M / 34G   INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:39.603   204M / 34G   INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:39.603   204M / 34G   INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:39.603   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.603   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:39.603   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.604   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:39.604   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:39.605   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:39.605   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:39.606   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:39.606   204M / 34G   INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:39.607   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:39.611   204M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 161 times\n",
      "  0:00:39.611   204M / 34G   INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:39.612   204M / 34G   INFO    General                 (simplification.cpp        : 476)   Average coverage = 118.963\n",
      "  0:00:39.612   204M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:39.612   204M / 34G   INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K33/assembly_graph_with_scaffolds.gfa\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:39.681   204M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K33/before_rr.fasta\n",
      "  0:00:39.754   204M / 34G   INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K33/assembly_graph.fastg\n",
      "  0:00:40.014   204M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K33/simplified_contigs.fasta\n",
      "  0:00:40.086   204M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K33/final_contigs.fasta\n",
      "  0:00:40.157   204M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:40.157   204M / 34G   INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K33/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:40.216   204M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K33/before_rr.fasta\n",
      "  0:00:40.333   204M / 34G   INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K33/assembly_graph.fastg\n",
      "  0:00:40.632   204M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K33/simplified_contigs.fasta\n",
      "  0:00:40.707   204M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K33/final_contigs.fasta\n",
      "  0:00:40.791   204M / 34G   INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:40.811   204M / 34G   INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 40 seconds\n",
      "\n",
      "== Running assembler: K55\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/K55/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 700 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/analytics/projects/bioinf_workshop/data/week3/assembly_all/dataset.info) with K=55\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 50\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.004     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
      "  0:00:00.006     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 90\n",
      "  0:00:00.006     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 89.9999\n",
      "  0:00:00.006     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.006     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 2500 files using 50 threads. This might take a while.\n",
      "  0:00:00.008     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:00:00.008     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 4.66664 Gb\n",
      "  0:00:00.008     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 16384\n",
      "  0:00:07.985    34G / 34G   INFO    General                 (kmer_splitters.hpp        : 289)   Processed 21996480 reads\n",
      "  0:00:07.985    34G / 34G   INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:12.038   212M / 34G   INFO    General                 (kmer_splitters.hpp        : 308)   Used 21996480 reads\n",
      "  0:00:12.038   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:15.254   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 7887810 kmers in total.\n",
      "  0:00:15.255   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:15.559   200M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:15.559   200M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:15.559   200M / 34G   INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 800 files using 50 threads. This might take a while.\n",
      "  0:00:15.560   200M / 34G   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:00:15.560   200M / 34G   INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 4.66536 Gb\n",
      "  0:00:15.560   200M / 34G   INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 41943\n",
      "  0:00:19.807    27G / 34G   INFO    General                 (kmer_splitters.hpp        : 380)   Processed 7887810 kmers\n",
      "  0:00:19.807    27G / 34G   INFO    General                 (kmer_splitters.hpp        : 385)   Used 7887810 kmers.\n",
      "  0:00:22.383   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:23.169   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 7909518 kmers in total.\n",
      "  0:00:23.169   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:23.348   204M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:23.554   204M / 34G   INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:23.681   204M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 3675928 bytes occupied (3.71798 bits per kmer).\n",
      "  0:00:23.687   212M / 34G   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:23.864   212M / 34G   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:23.872   208M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:23.955   208M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:24.145   220M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 290058 sequences extracted\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:24.301   220M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:24.354   220M / 34G   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 1 loops collected\n",
      "  0:00:24.499   224M / 34G   INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:24.499   224M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:24.499   224M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:24.627   224M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 3683896 bytes occupied (3.73629 bits per kmer).\n",
      "  0:00:24.638   256M / 34G   INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:27.187   256M / 34G   INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:27.374   256M / 34G   INFO    General                 (construction.cpp          : 464)   Processed 580117 edges\n",
      "  0:00:27.418   212M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:27.418   212M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 8\n",
      "  0:00:27.418   212M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 70\n",
      "  0:00:27.418   212M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 70. Coverage mad: 29.652\n",
      "  0:00:27.418   212M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:00:27.501   212M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:00:27.710   212M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:00:28.468   212M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:00:30.114   212M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:00:31.080   212M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 70.2285. Fitted coverage std. dev: 26.3528\n",
      "  0:00:31.083   212M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.570453\n",
      "  0:00:31.083   212M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 27\n",
      "  0:00:31.083   212M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 27\n",
      "  0:00:31.083   212M / 34G   INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 5001377\n",
      "  0:00:31.083   212M / 34G   INFO    General                 (genomic_info_filler.cpp   : 112)   Mean coverage was calculated as 70.2285\n",
      "  0:00:31.083   212M / 34G   INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 27\n",
      "  0:00:31.083   212M / 34G   INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:31.083   212M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:31.083   212M / 34G   INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:31.083   212M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:31.083   212M / 34G   INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 800 files using 50 threads. This might take a while.\n",
      "  0:00:31.084   212M / 34G   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\n",
      "  0:00:31.084   212M / 34G   INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 4.66529 Gb\n",
      "  0:00:31.084   212M / 34G   INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 41943\n",
      "  0:00:35.483    27G / 34G   INFO    General                 (edge_index_builders.hpp   :  77)   Processed 580117 edges\n",
      "  0:00:35.483    27G / 34G   INFO    General                 (edge_index_builders.hpp   :  82)   Used 580117 sequences.\n",
      "  0:00:38.073   216M / 34G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:38.895   216M / 34G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 7887810 kmers in total.\n",
      "  0:00:38.895   216M / 34G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:39.106   216M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:39.326   216M / 34G   INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:39.466   216M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 3665984 bytes occupied (3.71813 bits per kmer).\n",
      "  0:00:39.542   400M / 34G   INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:39.672   396M / 34G   INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:39.673   396M / 34G   INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:39.988   424M / 34G   INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:41.180   424M / 34G   INFO    General                 (gap_closer.cpp            : 138)   Used 5498917 paired reads\n",
      "  0:00:41.180   424M / 34G   INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:41.276   400M / 34G   INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:41.702   400M / 34G   INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 5 gaps after checking 313 candidates\n",
      "  0:00:41.743   396M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:41.746   212M / 34G   INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:41.746   212M / 34G   INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:41.746   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:41.768   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:41.768   212M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
      "  0:00:43.515   220M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 129897 times\n",
      "  0:00:43.515   220M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
      "  0:00:43.556   220M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 698 times\n",
      "  0:00:43.556   220M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
      "  0:00:43.683   220M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 24259 times\n",
      "  0:00:43.685   216M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:43.685   216M / 34G   INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:43.685   216M / 34G   INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:43.686   216M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:43.686   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:43.696   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 140 times\n",
      "  0:00:43.696   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:44.038   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1479 times\r\n",
      "  0:00:44.038   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:44.046   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 22 times\r\n",
      "  0:00:44.046   216M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\r\n",
      "  0:00:44.046   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:44.046   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 5 times\r\n",
      "  0:00:44.046   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:44.049   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 14 times\r\n",
      "  0:00:44.049   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:44.053   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 40 times\r\n",
      "  0:00:44.053   216M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\r\n",
      "  0:00:44.053   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:44.055   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 7 times\r\n",
      "  0:00:44.055   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:44.059   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 20 times\r\n",
      "  0:00:44.059   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:44.059   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\r\n",
      "  0:00:44.059   216M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\r\n",
      "  0:00:44.059   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:44.059   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:44.059   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:44.059   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\r\n",
      "  0:00:44.059   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:44.060   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\r\n",
      "  0:00:44.060   216M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\r\n",
      "  0:00:44.060   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:44.060   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:44.060   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:44.060   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2 times\r\n",
      "  0:00:44.060   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\r\n",
      "  0:00:44.061   216M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\r\n",
      "  0:00:44.061   216M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\r\n",
      "  0:00:44.061   216M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\r\n",
      "  0:00:44.061   216M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\r\n",
      "  0:00:44.061   216M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:44.061   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\r\n",
      "  0:00:44.062   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:44.062   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\r\n",
      "  0:00:44.062   216M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\r\n",
      "  0:00:44.062   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:44.065   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:44.066   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:44.073   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 4 times\r\n",
      "  0:00:44.073   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:44.074   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\r\n",
      "  0:00:44.074   216M / 34G   INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\r\n",
      "  0:00:44.074   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\r\n",
      "  0:00:44.075   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\r\n",
      "  0:00:44.075   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\r\n",
      "  0:00:44.075   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\r\n",
      "  0:00:44.075   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\r\n",
      "  0:00:44.075   216M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\r\n",
      "  0:00:44.075   216M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\r\n",
      "  0:00:44.075   216M / 34G   INFO    General                 (graph_pack.hpp            : 101)   Index refill\r\n",
      "  0:00:44.076   216M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\r\n",
      "  0:00:44.076   216M / 34G   INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 800 files using 50 threads. This might take a while.\r\n",
      "  0:00:44.078   216M / 34G   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2600\r\n",
      "  0:00:44.078   216M / 34G   INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 4.66526 Gb\r\n",
      "  0:00:44.078   216M / 34G   INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 41943\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:48.215    27G / 34G   INFO    General                 (edge_index_builders.hpp   :  77)   Processed 2389 edges\n",
      "  0:00:48.215    27G / 34G   INFO    General                 (edge_index_builders.hpp   :  82)   Used 2389 sequences.\n",
      "  0:00:50.565   220M / 34G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:51.452   220M / 34G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 5283497 kmers in total.\n",
      "  0:00:51.452   220M / 34G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:51.605   220M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:51.742   220M / 34G   INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:51.842   220M / 34G   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2458280 bytes occupied (3.7222 bits per kmer).\n",
      "  0:00:51.921   344M / 34G   INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:52.209   340M / 34G   INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:52.210   340M / 34G   INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:52.211   340M / 34G   INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:53.191   340M / 34G   INFO    General                 (gap_closer.cpp            : 138)   Used 5498917 paired reads\n",
      "  0:00:53.191   340M / 34G   INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:53.194   340M / 34G   INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:53.198   340M / 34G   INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 11 gaps after checking 108 candidates\n",
      "  0:00:53.647   340M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:53.647   340M / 34G   INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:53.647   340M / 34G   INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:53.648   340M / 34G   INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:53.648   340M / 34G   INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:53.648   340M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:53.648   340M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:53.648   340M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:53.649   340M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:53.649   340M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:53.649   340M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:53.649   340M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:53.651   340M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:53.651   340M / 34G   INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:53.652   340M / 34G   INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:53.657   340M / 34G   INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 33 times\n",
      "  0:00:53.657   340M / 34G   INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:53.657   340M / 34G   INFO    General                 (simplification.cpp        : 476)   Average coverage = 72.3729\n",
      "  0:00:53.657   340M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:53.657   340M / 34G   INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 191006 kmers to process\n",
      "  0:00:53.731   340M / 34G   INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:56.716   368M / 34G   INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:56.844   344M / 34G   INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:56.845   344M / 34G   INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 3 nucleotides\n",
      "  0:00:56.846   340M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:56.846   340M / 34G   INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K55/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:56.890   340M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K55/before_rr.fasta\n",
      "  0:00:56.942   340M / 34G   INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K55/assembly_graph.fastg\n",
      "  0:00:57.130   340M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K55/simplified_contigs.fasta\n",
      "  0:00:57.181   340M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K55/intermediate_contigs.fasta\n",
      "  0:00:57.232   340M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:57.240   340M / 34G   INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 191122 kmers to process\n",
      "  0:00:57.310   340M / 34G   INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:57.311   340M / 34G   INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 58171\n",
      "  0:00:57.311   340M / 34G   INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:57.311   340M / 34G   INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:57.557     1G / 34G   INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:57.557     1G / 34G   INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:57.675     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:57.757     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:00:57.920     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   65536 reads processed\n",
      "  0:00:58.248     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   131072 reads processed\n",
      "  0:00:58.972     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   262144 reads processed\n",
      "  0:01:00.471     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   524288 reads processed\n",
      "  0:01:04.814     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   1048576 reads processed\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:11.879     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   2097152 reads processed\n",
      "  0:01:26.409     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   4194304 reads processed\n",
      "  0:01:34.265     1G / 34G   INFO    General                 (binary_converter.hpp      : 117)   5097264 reads written\n",
      "  0:01:34.823     1G / 34G   INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:01:34.933     1G / 34G   INFO    General                 (binary_converter.hpp      : 117)   4730 reads written\n",
      "  0:01:34.950     1G / 34G   INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:01:35.043     1G / 34G   INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:01:35.115     1G / 34G   INFO    General                 (pair_info_count.cpp       :  30)   Mapping mate-pairs using BWA-mem mapper\n",
      "  0:01:44.679     1G / 34G   INFO    General                 (sequence_mapper_notifier.h:  98)   Total 5097264 reads processed\n",
      "  0:01:45.960     1G / 34G   INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:01:45.960     1G / 34G   INFO    General                 (pair_info_count.cpp       : 213)   1977309 paired reads (38.7916% of all) aligned to long edges\n",
      "  0:01:45.967   340M / 34G   INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 2574.34, deviation = 159.949, left quantile = 2370, right quantile = 2780, read length = 49\n",
      "  0:01:45.967   340M / 34G   INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:01:45.967   340M / 34G   INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:01:45.967   340M / 34G   INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 2370, right insert size quantile 2780, filtering threshold 2, rounding threshold 0\n",
      "  0:01:45.973   356M / 34G   INFO    General                 (pair_info_count.cpp       :  30)   Mapping mate-pairs using BWA-mem mapper\n",
      "  0:01:54.666   368M / 34G   INFO    General                 (sequence_mapper_notifier.h:  98)   Total 5097264 reads processed\n",
      "  0:01:54.682   340M / 34G   INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #1\n",
      "  0:01:54.682   340M / 34G   INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:01:54.950     1G / 34G   INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #1 (takes a while)\n",
      "  0:01:54.950     1G / 34G   INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:01:55.070     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:01:55.152     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:01:55.315     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   65536 reads processed\n",
      "  0:01:55.643     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   131072 reads processed\n",
      "  0:01:56.294     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   262144 reads processed\n",
      "  0:01:57.608     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   524288 reads processed\n",
      "  0:02:01.952     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   1048576 reads processed\n",
      "  0:02:09.140     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   2097152 reads processed\n",
      "  0:02:23.046     1G / 34G   INFO    General                 (binary_converter.hpp      :  93)   4194304 reads processed\n",
      "  0:02:30.161     1G / 34G   INFO    General                 (binary_converter.hpp      : 117)   5097081 reads written\n",
      "  0:02:30.710     1G / 34G   INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:02:30.823     1G / 34G   INFO    General                 (binary_converter.hpp      : 117)   4917 reads written\n",
      "  0:02:30.842     1G / 34G   INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:02:30.938     1G / 34G   INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:02:30.955     1G / 34G   INFO    General                 (pair_info_count.cpp       :  30)   Mapping mate-pairs using BWA-mem mapper\n",
      "  0:02:40.403     1G / 34G   INFO    General                 (sequence_mapper_notifier.h:  98)   Total 5097081 reads processed\n",
      "  0:02:41.710     1G / 34G   INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:02:41.710     1G / 34G   INFO    General                 (pair_info_count.cpp       : 213)   1946465 paired reads (38.1878% of all) aligned to long edges\n",
      "  0:02:41.725   340M / 34G   INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 6219.1, deviation = 263.053, left quantile = 5890, right quantile = 6553, read length = 49\n",
      "  0:02:41.725   340M / 34G   INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #1\n",
      "  0:02:41.725   340M / 34G   INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:02:41.725   340M / 34G   INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 5890, right insert size quantile 6553, filtering threshold 2, rounding threshold 0\n",
      "  0:02:41.732   356M / 34G   INFO    General                 (pair_info_count.cpp       :  30)   Mapping mate-pairs using BWA-mem mapper\n",
      "  0:02:50.343   368M / 34G   INFO    General                 (sequence_mapper_notifier.h:  98)   Total 5097081 reads processed\n",
      "  0:02:50.355   340M / 34G   INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #2\n",
      "  0:02:50.355   340M / 34G   INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:02:50.609     1G / 34G   INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:02:54.165     1G / 34G   INFO    General                 (sequence_mapper_notifier.h:  98)   Total 5498917 reads processed\n",
      "  0:02:55.378     1G / 34G   INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:02:55.378     1G / 34G   INFO    General                 (pair_info_count.cpp       : 213)   2628013 paired reads (47.7915% of all) aligned to long edges\n",
      "  0:02:55.381   340M / 34G   INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 469.023, deviation = 14.6942, left quantile = 450, right quantile = 486, read length = 90\n",
      "  0:02:55.434   532M / 34G   INFO    General                 (pair_info_count.cpp       : 371)   Filtering data for library #2\n",
      "  0:02:55.435   536M / 34G   INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:02:57.872   536M / 34G   INFO    General                 (sequence_mapper_notifier.h:  98)   Total 5498917 reads processed\n",
      "  0:02:57.874   532M / 34G   INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #2\n",
      "  0:02:57.874   532M / 34G   INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:02:57.874   532M / 34G   INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 450, right insert size quantile 486, filtering threshold 2, rounding threshold 0\n",
      "  0:02:57.880   548M / 34G   INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:03:00.322   548M / 34G   INFO    General                 (sequence_mapper_notifier.h:  98)   Total 5498917 reads processed\n",
      "  0:03:00.328   340M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:03:00.328   340M / 34G   INFO    General                 (distance_estimation.cpp   : 173)   Processing library #2\n",
      "  0:03:00.328   340M / 34G   INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:03:00.328   340M / 34G   INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:03:00.335   340M / 34G   INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:03:00.335   340M / 34G   INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 10629\n",
      "  0:03:00.338   344M / 34G   INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:03:00.338   340M / 34G   INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:03:00.340   340M / 34G   INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:03:00.340   340M / 34G   INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:03:00.352   340M / 34G   INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 1675; contradictional = 136\n",
      "  0:03:00.363   340M / 34G   INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 75; contradictional = 26\n",
      "  0:03:00.363   340M / 34G   INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:03:00.363   340M / 34G   INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:03:00.368   340M / 34G   INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:03:00.368   340M / 34G   INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 2463\n",
      "  0:03:00.369   344M / 34G   INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:03:00.369   340M / 34G   INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:03:00.371   340M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\r\n",
      "  0:03:00.371   340M / 34G   INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\r\n",
      "  0:03:00.371   340M / 34G   INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\r\n",
      "  0:03:00.372   340M / 34G   INFO    General                 (launcher.cpp              : 170)   Autodetecting unique edge set parameters...\r\n",
      "  0:03:00.372   340M / 34G   INFO    General                 (launcher.cpp              : 172)   Minimal unique edge length set to the smallest MP library IS: 6219\r\n",
      "  0:03:00.372   340M / 34G   INFO    General                 (coverage_uniformity_analyz:  31)   genomic coverage is 68.8291 calculated of length 4813988\r\n",
      "  0:03:00.372   340M / 34G   INFO    General                 (launcher.cpp              : 178)   median coverage for edges longer than 6219 is 68.8291 uniformity 99%\r\n",
      "  0:03:00.373   340M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 6219, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.373   340M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 216, unique edges length 9595552, total edges length 10558933\r\n",
      "  0:03:00.377   348M / 34G   INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 6219\r\n",
      "  0:03:00.377   348M / 34G   INFO    General                 (extenders_logic.cpp       : 275)   Estimated coverage of library #2 is 72.3729\r\n",
      "  0:03:00.377   348M / 34G   INFO    General                 (extenders_logic.cpp       : 275)   Estimated coverage of library #2 is 72.3729\r\n",
      "  0:03:00.377   348M / 34G   INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\r\n",
      "  0:03:00.377   348M / 34G   INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\r\n",
      "  0:03:00.377   348M / 34G   INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\r\n",
      "  0:03:00.377   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 5919\r\n",
      "  0:03:00.378   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 5919, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.378   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 216, unique edges length 9595552, total edges length 10558933\r\n",
      "  0:03:00.378   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 5619\r\n",
      "  0:03:00.378   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 5619, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.378   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 220, unique edges length 9618710, total edges length 10558933\r\n",
      "  0:03:00.378   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 5319\r\n",
      "  0:03:00.379   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 5319, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.379   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 220, unique edges length 9618710, total edges length 10558933\r\n",
      "  0:03:00.379   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 5019\r\n",
      "  0:03:00.379   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 5019, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.379   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 224, unique edges length 9639114, total edges length 10558933\r\n",
      "  0:03:00.379   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 4719\r\n",
      "  0:03:00.380   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 4719, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.380   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 238, unique edges length 9706994, total edges length 10558933\r\n",
      "  0:03:00.380   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 4419\r\n",
      "  0:03:00.380   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 4419, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.380   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 242, unique edges length 9725264, total edges length 10558933\r\n",
      "  0:03:00.380   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 4119\r\n",
      "  0:03:00.381   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 4119, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.381   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 254, unique edges length 9776638, total edges length 10558933\r\n",
      "  0:03:00.381   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 3819\r\n",
      "  0:03:00.381   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 3819, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.381   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 256, unique edges length 9784858, total edges length 10558933\r\n",
      "  0:03:00.381   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 3519\r\n",
      "  0:03:00.381   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 3519, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.381   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 264, unique edges length 9813774, total edges length 10558933\r\n",
      "  0:03:00.381   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 3219\r\n",
      "  0:03:00.382   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 3219, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.382   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 270, unique edges length 9833786, total edges length 10558933\r\n",
      "  0:03:00.382   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 2919\r\n",
      "  0:03:00.382   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 2919, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.382   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 274, unique edges length 9846122, total edges length 10558933\r\n",
      "  0:03:00.382   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 2619\r\n",
      "  0:03:00.383   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 2619, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.383   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 290, unique edges length 9890166, total edges length 10558933\r\n",
      "  0:03:00.383   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 2319\r\n",
      "  0:03:00.383   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 2319, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.383   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 302, unique edges length 9919262, total edges length 10558933\r\n",
      "  0:03:00.383   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 2019\r\n",
      "  0:03:00.384   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 2019, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.384   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 316, unique edges length 9949150, total edges length 10558933\r\n",
      "  0:03:00.384   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 1719\r\n",
      "  0:03:00.384   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 1719, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.384   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 336, unique edges length 9987644, total edges length 10558933\r\n",
      "  0:03:00.384   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 1419\r\n",
      "  0:03:00.385   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 1419, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.385   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 346, unique edges length 10002724, total edges length 10558933\r\n",
      "  0:03:00.385   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 1119\r\n",
      "  0:03:00.385   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 1119, median long edge coverage: 68.8746, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.385   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 372, unique edges length 10036608, total edges length 10558933\r\n",
      "  0:03:00.385   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 819\r\n",
      "  0:03:00.386   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 819, median long edge coverage: 68.9992, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.386   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 416, unique edges length 10079446, total edges length 10558933\r\n",
      "  0:03:00.386   348M / 34G   INFO    General                 (launcher.cpp              : 313)   Will add extenders for length 519\r\n",
      "  0:03:00.386   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 519, median long edge coverage: 68.9992, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.386   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 482, unique edges length 10124468, total edges length 10558933\r\n",
      "  0:03:00.386   348M / 34G   INFO    General                 (launcher.cpp              : 318)   Will add final extenders for length 500\r\n",
      "  0:03:00.387   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  56)   With length cutoff: 500, median long edge coverage: 68.9992, and maximal unique coverage: 0.5\r\n",
      "  0:03:00.387   348M / 34G   INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  57)   Unique edges quantity: 490, unique edges length 10128504, total edges length 10558933\r\n",
      "  0:03:00.387   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.387   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.387   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.387   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.387   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.388   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.388   348M / 34G   INFO    General                 (extenders_logic.cpp       : 330)   Using 2 mate-pair libraries\r\n",
      "  0:03:00.388   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.388   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.388   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.389   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.389   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.389   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.389   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.389   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.389   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.390   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.390   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.390   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.390   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.390   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.390   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.391   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.391   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.391   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.391   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.391   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.392   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.392   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.392   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.392   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.392   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.392   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.393   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.393   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.393   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.393   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.393   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.393   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.394   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.394   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.394   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.394   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.394   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.394   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.395   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.395   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.395   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.395   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.395   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.395   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.396   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.396   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.396   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.396   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.396   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.396   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.397   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.397   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.397   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.397   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.397   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.398   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.398   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.398   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.398   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.398   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.399   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.399   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.399   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.399   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.399   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.399   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.400   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.400   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.400   348M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.400   348M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.400   348M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.400   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.401   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.401   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.401   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.401   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.401   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.401   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.402   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.402   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.402   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.402   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.402   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.402   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.403   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.403   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.403   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.403   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.403   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.404   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.404   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.404   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.404   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.404   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.404   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.405   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.405   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.405   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.405   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.405   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.405   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.406   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.406   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.406   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.406   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.407   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.407   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.407   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.407   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.407   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.407   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.408   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.408   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.408   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.408   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #0\r\n",
      "  0:03:00.408   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.408   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.409   352M / 34G   INFO    General                 (extenders_logic.cpp       : 152)   Creating Scaffolding 2015 extender for lib #1\r\n",
      "  0:03:00.409   352M / 34G   INFO    General                 (extenders_logic.cpp       : 156)   Paired unclustered indices not empty, using them\r\n",
      "  0:03:00.409   352M / 34G   INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  51)   ExtensionChooser2015 created\r\n",
      "  0:03:00.409   352M / 34G   INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 45\r\n",
      "  0:03:00.409   352M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 1111 (0%)\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:03:02.173   368M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 112 paths from 1111 (10%)\n",
      "  0:03:02.206   368M / 34G   INFO    General                 (path_extender.hpp         : 883)   Processed 128 paths from 1111 (11%)\n",
      "  0:03:02.683   372M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 224 paths from 1111 (20%)\n",
      "  0:03:02.705   372M / 34G   INFO    General                 (path_extender.hpp         : 883)   Processed 256 paths from 1111 (23%)\n",
      "  0:03:02.852   372M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 336 paths from 1111 (30%)\n",
      "  0:03:03.189   372M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 448 paths from 1111 (40%)\n",
      "  0:03:03.232   372M / 34G   INFO    General                 (path_extender.hpp         : 883)   Processed 512 paths from 1111 (46%)\n",
      "  0:03:03.238   372M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 560 paths from 1111 (50%)\n",
      "  0:03:03.258   372M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 672 paths from 1111 (60%)\n",
      "  0:03:03.277   372M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 784 paths from 1111 (70%)\n",
      "  0:03:03.314   376M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 896 paths from 1111 (80%)\n",
      "  0:03:03.315   376M / 34G   INFO    General                 (path_extender.hpp         : 885)   Processed 1008 paths from 1111 (90%)\n",
      "  0:03:03.315   376M / 34G   INFO    General                 (path_extender.hpp         : 883)   Processed 1024 paths from 1111 (92%)\n",
      "  0:03:03.318   376M / 34G   INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:03:03.318   376M / 34G   INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:03:03.323   372M / 34G   INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:03:03.323   372M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:03:03.323   372M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:03:03.324   372M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:03:03.324   372M / 34G   INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:03:03.328   372M / 34G   INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:03:03.333   372M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:03:03.334   372M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:03:03.335   372M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:03:03.337   376M / 34G   INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:03:03.337   376M / 34G   INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:03:03.403   376M / 34G   INFO   PathPolisher             (path_polisher.cpp         :  60)   Unexpected cycle while polishing path, stopping polishing\n",
      "  0:03:03.514   376M / 34G   INFO   PathPolisher             (path_polisher.cpp         :  60)   Unexpected cycle while polishing path, stopping polishing\n",
      "  0:03:03.645   376M / 34G   INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:03:03.647   376M / 34G   INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:03:03.650   376M / 34G   INFO    General                 (launcher.cpp              : 296)   Traversed 1 loops\n",
      "  0:03:03.650   376M / 34G   INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:03:03.650   376M / 34G   INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:03:03.651   376M / 34G   INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:03:03.651   376M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:03:03.651   376M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:03:03.651   376M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:03:03.651   376M / 34G   INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:03:03.652   376M / 34G   INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:03:03.654   376M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:03:03.654   376M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:03:03.654   376M / 34G   INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:03:03.654   376M / 34G   INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:03:03.656   376M / 34G   INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:03:03.703   348M / 34G   INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:03:03.703   348M / 34G   INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K55/assembly_graph_with_scaffolds.gfa\n",
      "  0:03:03.751   344M / 34G   INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K55/before_rr.fasta\n",
      "  0:03:04.265   344M / 34G   INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K55/assembly_graph.fastg\n",
      "  0:03:04.570   348M / 34G   INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K55/final_contigs.paths\n",
      "  0:03:04.689   344M / 34G   INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/analytics/projects/bioinf_workshop/data/week3/assembly_all//K55/scaffolds.paths\n",
      "  0:03:04.777   344M / 34G   INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:03:04.904   208M / 34G   INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 3 minutes 4 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 21, 33, 55 \n",
      "\n",
      " * Corrected reads are in /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/corrected/\n",
      " * Assembled contigs are in /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/contigs.fasta\n",
      " * Assembled scaffolds are in /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/scaffolds.fasta\n",
      " * Assembly graph is in /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/analytics/projects/bioinf_workshop/data/week3/assembly_all/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n"
     ]
    }
   ],
   "source": [
    "!mkdir -p ../data/week3/assembly_all\n",
    "!$SPADES -t 50 -m 700 -o ../data/week3/assembly_all \\\n",
    "    --pe1-1 ../data/week3/SRR292678sub_S1_L001_R1_001.fastq \\\n",
    "    --pe1-2 ../data/week3/SRR292678sub_S1_L001_R2_001.fastq \\\n",
    "    --mp1-1 ../data/week3/SRR292862_S2_L001_R1_001.fastq \\\n",
    "    --mp1-2 ../data/week3/SRR292862_S2_L001_R2_001.fastq \\\n",
    "    --mp2-1 ../data/week3/SRR292770_S1_L001_R1_001.fastq \\\n",
    "    --mp2-2 ../data/week3/SRR292770_S1_L001_R2_001.fastq "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/analytics/distr/quast-5.0.0/quast.py -o ../data/week3/quast_full ../data/week3/assembly_all/contigs.fasta ../data/week3/assembly_all/scaffolds.fasta\n",
      "\n",
      "Version: 5.0.0, de6973bb\n",
      "\n",
      "System information:\n",
      "  OS: Linux-3.10.0-862.3.3.el7.x86_64-x86_64-with-centos-7.5.1804-Core (linux_64)\n",
      "  Python version: 3.5.4\n",
      "  CPUs number: 56\n",
      "\n",
      "Started: 2018-10-12 13:56:00\n",
      "\n",
      "Logging to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/quast.log\n",
      "NOTICE: Maximum number of threads is set to 14 (use --threads option to set it manually)\n",
      "\n",
      "CWD: /home/analytics/projects/bioinf_workshop/notebooks\n",
      "Main parameters: \n",
      "  MODE: default, threads: 14, minimum contig length: 500, minimum alignment length: 65, \\\n",
      "  ambiguity: one, threshold for extensive misassembly size: 1000\n",
      "\n",
      "Contigs:\n",
      "  Pre-processing...\n",
      "  1  ../data/week3/assembly_all/contigs.fasta ==> contigs\n",
      "  2  ../data/week3/assembly_all/scaffolds.fasta ==> scaffolds\n",
      "\n",
      "2018-10-12 13:56:01\n",
      "Running Basic statistics processor...\n",
      "  Contig files: \n",
      "    1  contigs\n",
      "    2  scaffolds\n",
      "  Calculating N50 and L50...\n",
      "    1  contigs, N50 = 155269, L50 = 11, Total length = 5301146, GC % = 50.58, # N's per 100 kbp =  0.00\n",
      "    2  scaffolds, N50 = 1046850, L50 = 2, Total length = 5376279, GC % = 50.56, # N's per 100 kbp =  1078.27\n",
      "  Drawing Nx plot...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/basic_stats/Nx_plot.pdf\n",
      "  Drawing cumulative plot...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/basic_stats/cumulative_plot.pdf\n",
      "  Drawing GC content plot...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/basic_stats/GC_content_plot.pdf\n",
      "  Drawing contigs GC content plot...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/basic_stats/contigs_GC_content_plot.pdf\n",
      "  Drawing scaffolds GC content plot...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/basic_stats/scaffolds_GC_content_plot.pdf\n",
      "  Drawing Coverage histogram (bin size: 1x)...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/basic_stats/coverage_histogram.pdf\n",
      "  Drawing contigs coverage histogram (bin size: 2x)...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/basic_stats/contigs_coverage_histogram.pdf\n",
      "  Drawing scaffolds coverage histogram (bin size: 1x)...\n",
      "    saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/basic_stats/scaffolds_coverage_histogram.pdf\n",
      "Done.\n",
      "\n",
      "NOTICE: Genes are not predicted by default. Use --gene-finding or --glimmer option to enable it.\n",
      "\n",
      "2018-10-12 13:56:03\n",
      "Creating large visual summaries...\n",
      "This may take a while: press Ctrl-C to skip this step..\n",
      "  1 of 2: Creating Icarus viewers...\n",
      "  2 of 2: Creating PDF with all tables and plots...\n",
      "Done\n",
      "\n",
      "2018-10-12 13:56:04\n",
      "RESULTS:\n",
      "  Text versions of total report are saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/report.txt, report.tsv, and report.tex\n",
      "  Text versions of transposed total report are saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/transposed_report.txt, transposed_report.tsv, and transposed_report.tex\n",
      "  HTML version (interactive tables and plots) is saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/report.html\n",
      "  PDF version (tables and plots) is saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/report.pdf\n",
      "  Icarus (contig browser) is saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/icarus.html\n",
      "  Log is saved to /home/analytics/projects/bioinf_workshop/data/week3/quast_full/quast.log\n",
      "\n",
      "Finished: 2018-10-12 13:56:05\n",
      "Elapsed time: 0:00:04.426114\n",
      "NOTICEs: 2; WARNINGs: 0; non-fatal ERRORs: 0\n",
      "\n",
      "Thank you for using QUAST!\n"
     ]
    }
   ],
   "source": [
    "!mkdir -p ../data/week3/quast_full\n",
    "!$QUAST -o ../data/week3/quast_full ../data/week3/assembly_all/contigs.fasta ../data/week3/assembly_all/scaffolds.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <iframe\n",
       "            width=\"1000\"\n",
       "            height=\"600\"\n",
       "            src=\"http://bioinf-workshop.s3-website.eu-central-1.amazonaws.com/week_3/quast_2.html\"\n",
       "            frameborder=\"0\"\n",
       "            allowfullscreen\n",
       "        ></iframe>\n",
       "        "
      ],
      "text/plain": [
       "<IPython.lib.display.IFrame at 0x7f9a0ac2ee80>"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = perform_file_upload(open(\"../data/week3/quast_full/report.html\", \"r\").read(), \"week_3/quast_2.html\")\n",
    "IFrame(src=f, width=1000, height=600)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[18:07:54] This is prokka 1.13.3\n",
      "[18:07:54] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
      "[18:07:54] Homepage is https://github.com/tseemann/prokka\n",
      "[18:07:54] Local time is Sun Oct 14 18:07:54 2018\n",
      "[18:07:54] You are analytics\n",
      "[18:07:54] Operating system is linux\n",
      "[18:07:54] You have BioPerl 1.007002\n",
      "[18:07:54] System has 56 cores.\n",
      "[18:07:54] Will use maximum of 8 cores.\n",
      "[18:07:54] Annotating as >>> Bacteria <<<\n",
      "[18:07:54] Generating locus_tag from '../data/week3/assembly_all/contigs.fasta' contents.\n",
      "[18:07:54] Setting --locustag DDOCCBLM from MD5 dd8ccb56ff3c0879095371c7d2dda040\n",
      "[18:07:54] Creating new output folder: PROKKA_10142018\n",
      "[18:07:54] Running: mkdir -p PROKKA_10142018\n",
      "[18:07:54] Using filename prefix: PROKKA_10142018.XXX\n",
      "[18:07:54] Setting HMMER_NCPU=1\n",
      "[18:07:54] Writing log to: PROKKA_10142018/PROKKA_10142018.log\n",
      "[18:07:54] Command: /home/analytics/anaconda3/bin/prokka ../data/week3/assembly_all/contigs.fasta\n",
      "[18:07:54] Appending to PATH: /home/analytics/anaconda3/bin\n",
      "[18:07:54] Looking for 'aragorn' - found /home/analytics/anaconda3/bin/aragorn\n",
      "[18:07:54] Determined aragorn version is 1.2\n",
      "[18:07:54] Looking for 'barrnap' - found /home/analytics/anaconda3/bin/barrnap\n",
      "[18:07:55] Determined barrnap version is 0.9\n",
      "[18:07:55] Looking for 'blastp' - found /home/analytics/anaconda3/bin/blastp\n",
      "[18:07:55] Determined blastp version is 2.7\n",
      "[18:07:55] Looking for 'cmpress' - found /home/analytics/anaconda3/bin/cmpress\n",
      "[18:07:55] Determined cmpress version is 1.1\n",
      "[18:07:55] Looking for 'cmscan' - found /home/analytics/anaconda3/bin/cmscan\n",
      "[18:07:55] Determined cmscan version is 1.1\n",
      "[18:07:55] Looking for 'egrep' - found /usr/bin/egrep\n",
      "[18:07:55] Looking for 'find' - found /usr/bin/find\n",
      "[18:07:55] Looking for 'grep' - found /usr/bin/grep\n",
      "[18:07:55] Looking for 'hmmpress' - found /home/analytics/anaconda3/bin/hmmpress\n",
      "[18:07:55] Determined hmmpress version is 3.2\n",
      "[18:07:55] Looking for 'hmmscan' - found /home/analytics/anaconda3/bin/hmmscan\n",
      "[18:07:55] Determined hmmscan version is 3.2\n",
      "[18:07:55] Looking for 'java' - found /home/analytics/anaconda3/bin/java\n",
      "[18:07:55] Looking for 'less' - found /usr/bin/less\n",
      "[18:07:55] Looking for 'makeblastdb' - found /home/analytics/anaconda3/bin/makeblastdb\n",
      "[18:07:55] Determined makeblastdb version is 2.7\n",
      "[18:07:55] Looking for 'minced' - found /home/analytics/anaconda3/bin/minced\n",
      "[18:07:55] Determined minced version is 3.0\n",
      "[18:07:55] Looking for 'parallel' - found /home/analytics/anaconda3/bin/parallel\n",
      "[18:07:55] Determined parallel version is 20180922\n",
      "[18:07:55] Looking for 'prodigal' - found /home/analytics/anaconda3/bin/prodigal\n",
      "[18:07:55] Determined prodigal version is 2.6\n",
      "[18:07:55] Looking for 'prokka-genbank_to_fasta_db' - found /home/analytics/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[18:07:55] Looking for 'sed' - found /usr/bin/sed\n",
      "[18:07:55] Looking for 'tbl2asn' - found /home/analytics/anaconda3/bin/tbl2asn\n",
      "[18:07:55] Determined tbl2asn version is 25.6\n",
      "[18:07:55] Using genetic code table 11.\n",
      "[18:07:55] Loading and checking input file: ../data/week3/assembly_all/contigs.fasta\n",
      "[18:07:55] Wrote 569 contigs totalling 5386179 bp.\n",
      "[18:07:55] Predicting tRNAs and tmRNAs\n",
      "[18:07:55] Running: aragorn -l -gc11  -w PROKKA_10142018\\/PROKKA_10142018\\.fna\n",
      "[18:07:57] 1 tRNA-Pro [204937,205013] 35 (ggg)\n",
      "[18:07:57] 2 tRNA-Arg [369799,369873] 34 (cct)\n",
      "[18:07:57] 1 tRNA-Arg c[8527,8603] 35 (tct)\n",
      "[18:07:57] 2 tRNA-Ser c[226120,226200] 38 (gga)\n",
      "[18:07:57] 3 tRNA-Thr c[313685,313760] 34 (cgt)\n",
      "[18:07:57] 4 tRNA-Asp c[368314,368390] 35 (gtc)\n",
      "[18:07:57] 1 tRNA-Val [104489,104565] 35 (gac)\n",
      "[18:07:57] 2 tRNA-Val [104570,104646] 35 (gac)\n",
      "[18:07:57] 1 tRNA-Leu [259770,259856] 35 (cag)\n",
      "[18:07:57] 2 tRNA-Leu [259885,259971] 35 (cag)\n",
      "[18:07:57] 3 tRNA-Leu [260006,260092] 35 (cag)\n",
      "[18:07:57] 1 tRNA-Ser [35165,35252] 35 (tga)\n",
      "[18:07:57] 2 tRNA-Ser [142423,142510] 35 (gga)\n",
      "[18:07:57] 1 tRNA-Val [99,174] 34 (tac)\n",
      "[18:07:57] 2 tRNA-Lys [179,254] 34 (ttt)\n",
      "[18:07:57] 1 tRNA-Glu c[147228,147303] 35 (ttc)\n",
      "[18:07:57] 1 tRNA-Gly c[58342,58415] 33 (ccc)\n",
      "[18:07:57] 2 tRNA-Phe [173780,173855] 34 (gaa)\n",
      "[18:07:57] 1 tRNA-Leu [96477,96563] 35 (gag)\n",
      "[18:07:57] 2 tRNA-Met [100344,100420] 35 (cat)\n",
      "[18:07:57] 1 tRNA-Met c[23543,23618] 34 (cat)\n",
      "[18:07:57] 1 tRNA-Glu c[3258,3333] 35 (ttc)\n",
      "[18:07:57] 1 tRNA-Ala c[3248,3323] 34 (tgc)\n",
      "[18:07:57] 2 tRNA-Ile c[3366,3442] 35 (gat)\n",
      "[18:07:57] 3 tRNA-Pro c[61381,61457] 35 (tgg)\n",
      "[18:07:57] 4 tRNA-Leu c[61500,61586] 35 (cag)\n",
      "[18:07:57] 5 tRNA-His c[61607,61682] 34 (gtg)\n",
      "[18:07:57] 6 tRNA-Arg c[61741,61817] 35 (ccg)\n",
      "[18:07:57] 7 tRNA-Trp c[97161,97236] 34 (cca)\n",
      "[18:07:57] 8 tRNA-Asp c[97245,97321] 35 (gtc)\n",
      "[18:07:57] 1 tRNA-Met [69,145] 35 (cat)\n",
      "[18:07:57] 2 tRNA-Gln [194,268] 33 (ctg)\n",
      "[18:07:57] 3 tRNA-Gln [306,380] 33 (ctg)\n",
      "[18:07:57] 1 tRNA-Gly [18077,18152] 34 (gcc)\n",
      "[18:07:57] 2 tRNA-Cys [18207,18280] 33 (gca)\n",
      "[18:07:57] 3 tRNA-Leu [18293,18379] 35 (taa)\n",
      "[18:07:57] 4 tRNA-Met [76430,76505] 34 (cat)\n",
      "[18:07:57] 5 tRNA-Arg [76515,76591] 35 (tcg)\n",
      "[18:07:57] 6 tRNA-Arg [76605,76681] 35 (tct)\n",
      "[18:07:57] 1 tmRNA [25912,26274] 90,125 ANDENYALAA**\n",
      "[18:07:57] 2 tRNA-Met c[43559,43634] 34 (cat)\n",
      "[18:07:57] 3 tRNA-Arg c[75661,75737] 35 (acg)\n",
      "[18:07:57] 1 tRNA-Leu c[6238,6322] 35 (caa)\n",
      "[18:07:57] 1 tRNA-SeC c[1643,1737] 35 (tca)\n",
      "[18:07:57] 1 tRNA-Gly c[36233,36308] 34 (gcc)\n",
      "[18:07:57] 2 tRNA-Gly c[36345,36420] 34 (gcc)\n",
      "[18:07:57] 3 tRNA-Gly c[36457,36532] 34 (gcc)\n",
      "[18:07:57] 4 tRNA-Phe [66260,66335] 34 (gaa)\n",
      "[18:07:57] 1 tRNA-Asn c[12354,12429] 34 (gtt)\n",
      "[18:07:57] 2 tRNA-Asn c[14763,14838] 34 (gtt)\n",
      "[18:07:57] 3 tRNA-Asn [16587,16662] 34 (gtt)\n",
      "[18:07:58] 4 tRNA-Asn c[56478,56553] 34 (gtt)\n",
      "[18:07:58] 5 tRNA-Ser [57545,57634] 35 (cga)\n",
      "[18:07:58] 1 tRNA-Met [60565,60641] 35 (cat)\n",
      "[18:07:58] 1 tRNA-Lys c[57410,57485] 34 (ttt)\n",
      "[18:07:58] 2 tRNA-Lys c[57519,57594] 34 (ttt)\n",
      "[18:07:58] 3 tRNA-Lys c[57741,57816] 34 (ttt)\n",
      "[18:07:58] 1 tRNA-Tyr [35725,35809] 35 (gta)\n",
      "[18:07:58] 1 tRNA-Glu c[330,405] 35 (ttc)\n",
      "[18:07:58] 1 tRNA-Pro [36728,36804] 35 (cgg)\n",
      "[18:07:58] 1 tRNA-Thr c[305,380] 34 (ggt)\n",
      "[18:07:58] 2 tRNA-Gly c[387,461] 34 (tcc)\n",
      "[18:07:58] 3 tRNA-Tyr c[578,662] 35 (gta)\n",
      "[18:07:58] 4 tRNA-Thr c[671,746] 34 (tgt)\n",
      "[18:07:58] 1 tRNA-Ser c[58,150] 35 (gct)\n",
      "[18:07:58] 1 tRNA-Arg c[1927,2003] 35 (tct)\n",
      "[18:07:58] 2 tRNA-Arg c[2017,2093] 35 (tcg)\n",
      "[18:07:58] 1 tRNA-Thr c[105,180] 34 (ggt)\n",
      "[18:07:58] 1 tRNA-Leu c[79,163] 35 (tag)\n",
      "[18:07:58] 1 tRNA-Arg [62,138] 35 (tcg)\n",
      "[18:07:58] 1 tRNA-Ala [1,76] 34 (ggc)\n",
      "[18:07:58] 1 tRNA-Gln [1,75] 33 (ttg)\n",
      "[18:07:58] Found 72 tRNAs\n",
      "[18:07:58] Predicting Ribosomal RNAs\n",
      "[18:07:58] Running Barrnap with 8 threads\n",
      "[18:08:03] 1 NODE_12_length_151014_cov_79.944660 6 23S ribosomal RNA\n",
      "[18:08:03] 2 NODE_12_length_151014_cov_79.944660 2860 5S ribosomal RNA\n",
      "[18:08:03] 3 NODE_14_length_117508_cov_91.829149 162 23S ribosomal RNA\n",
      "[18:08:03] 4 NODE_14_length_117508_cov_91.829149 3420 16S ribosomal RNA\n",
      "[18:08:03] 5 NODE_16_length_97481_cov_91.125480 162 23S ribosomal RNA\n",
      "[18:08:03] 6 NODE_16_length_97481_cov_91.125480 3512 16S ribosomal RNA\n",
      "[18:08:03] 7 NODE_16_length_97481_cov_91.125480 97379 5S ribosomal RNA\n",
      "[18:08:03] 8 NODE_192_length_440_cov_1.363636 34 16S ribosomal RNA (partial)\n",
      "[18:08:03] 9 NODE_19_length_84802_cov_71.598074 84700 5S ribosomal RNA\n",
      "[18:08:03] 10 NODE_21_length_75978_cov_83.375907 46 16S ribosomal RNA\n",
      "[18:08:03] 11 NODE_31_length_49097_cov_85.510664 492 16S ribosomal RNA\n",
      "[18:08:03] 12 NODE_364_length_223_cov_0.720238 19 5S ribosomal RNA\n",
      "[18:08:03] 13 NODE_40_length_27934_cov_115.763155 26352 16S ribosomal RNA\n",
      "[18:08:03] 14 NODE_4_length_311953_cov_73.800710 46 16S ribosomal RNA\n",
      "[18:08:03] 15 NODE_74_length_4485_cov_92.833634 4383 5S ribosomal RNA\n",
      "[18:08:03] 16 NODE_7_length_200177_cov_70.432206 197208 5S ribosomal RNA\n",
      "[18:08:03] 17 NODE_7_length_200177_cov_70.432206 197417 23S ribosomal RNA\n",
      "[18:08:03] 18 NODE_8_length_180977_cov_80.739927 143923 5S ribosomal RNA\n",
      "[18:08:03] 19 NODE_8_length_180977_cov_80.739927 144132 23S ribosomal RNA\n",
      "[18:08:03] 20 NODE_8_length_180977_cov_80.739927 147390 16S ribosomal RNA\n",
      "[18:08:03] Found 20 rRNAs\n",
      "[18:08:03] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[18:08:03] Total of 91 tRNA + rRNA features\n",
      "[18:08:03] Searching for CRISPR repeats\n",
      "[18:08:04] CRISPR1 NODE_27_length_60643_cov_69.781079 16926 with 14 spacers\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[18:08:04] Found 1 CRISPRs\n",
      "[18:08:04] Predicting coding sequences\n",
      "[18:08:04] Contigs total 5386179 bp, so using single mode\n",
      "[18:08:04] Running: prodigal -i PROKKA_10142018\\/PROKKA_10142018\\.fna -c -m -g 11 -p single -f sco -q\n",
      "[18:08:09] Excluding CDS which overlaps existing RNA (tRNA) at NODE_2_length_376590_cov_67.773500:225885..226130 on - strand\n",
      "[18:08:09] Excluding CDS which overlaps existing RNA (rRNA) at NODE_4_length_311953_cov_73.800710:145..399 on - strand\n",
      "[18:08:10] Excluding CDS which overlaps existing RNA (rRNA) at NODE_8_length_180977_cov_80.739927:147489..147743 on - strand\n",
      "[18:08:10] Excluding CDS which overlaps existing RNA (tRNA) at NODE_9_length_176064_cov_73.410274:173768..173989 on + strand\n",
      "[18:08:10] Excluding CDS which overlaps existing RNA (rRNA) at NODE_14_length_117508_cov_91.829149:3519..3773 on - strand\n",
      "[18:08:10] Excluding CDS which overlaps existing RNA (rRNA) at NODE_16_length_97481_cov_91.125480:3611..3865 on - strand\n",
      "[18:08:11] Excluding CDS which overlaps existing RNA (tRNA) at NODE_20_length_76937_cov_74.426641:76611..76790 on - strand\n",
      "[18:08:11] Excluding CDS which overlaps existing RNA (rRNA) at NODE_21_length_75978_cov_83.375907:145..399 on - strand\n",
      "[18:08:11] Excluding CDS which overlaps existing RNA (tRNA) at NODE_29_length_52209_cov_74.801818:35716..35988 on - strand\n",
      "[18:08:11] Excluding CDS which overlaps existing RNA (rRNA) at NODE_31_length_49097_cov_85.510664:591..845 on - strand\n",
      "[18:08:11] Excluding CDS which overlaps existing RNA (rRNA) at NODE_40_length_27934_cov_115.763155:27536..27790 on + strand\n",
      "[18:08:11] Excluding CDS which overlaps existing RNA (rRNA) at NODE_192_length_440_cov_1.363636:98..364 on + strand\n",
      "[18:08:12] Found 4998 CDS\n",
      "[18:08:12] Connecting features back to sequences\n",
      "[18:08:12] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[18:08:12] Annotating CDS, please be patient.\n",
      "[18:08:12] Will use 8 CPUs for similarity searching.\n",
      "[18:08:13] There are still 4998 unannotated CDS left (started with 4998)\n",
      "[18:08:13] Will use blast to search against /home/analytics/anaconda3/db/kingdom/Bacteria/IS with 8 CPUs\n",
      "[18:08:13] Running: cat PROKKA_10142018\\/IS\\.faa | parallel --gnu --plain -j 8 --block 97320 --recstart '>' --pipe blastp -query - -db /home/analytics/anaconda3/db/kingdom/Bacteria/IS -evalue 1e-30 -qcov_hsp_perc 90 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > PROKKA_10142018\\/IS\\.blast 2> /dev/null\n",
      "[18:08:29] Deleting unwanted file: PROKKA_10142018/IS.faa\n",
      "[18:08:29] Deleting unwanted file: PROKKA_10142018/IS.blast\n",
      "[18:08:31] There are still 4930 unannotated CDS left (started with 4998)\n",
      "[18:08:31] Will use blast to search against /home/analytics/anaconda3/db/kingdom/Bacteria/AMR with 8 CPUs\n",
      "[18:08:31] Running: cat PROKKA_10142018\\/AMR\\.faa | parallel --gnu --plain -j 8 --block 96203 --recstart '>' --pipe blastp -query - -db /home/analytics/anaconda3/db/kingdom/Bacteria/AMR -evalue 9.99999999999999e-301 -qcov_hsp_perc 90 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > PROKKA_10142018\\/AMR\\.blast 2> /dev/null\n",
      "[18:08:42] Deleting unwanted file: PROKKA_10142018/AMR.faa\n",
      "[18:08:42] Deleting unwanted file: PROKKA_10142018/AMR.blast\n",
      "[18:08:43] There are still 4930 unannotated CDS left (started with 4998)\n",
      "[18:08:43] Will use blast to search against /home/analytics/anaconda3/db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[18:08:43] Running: cat PROKKA_10142018\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 96203 --recstart '>' --pipe blastp -query - -db /home/analytics/anaconda3/db/kingdom/Bacteria/sprot -evalue 1e-09 -qcov_hsp_perc 80 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > PROKKA_10142018\\/sprot\\.blast 2> /dev/null\n",
      "[18:09:32] Modify product: Uncharacterized lipoprotein GfcB => putative lipoprotein GfcB\n",
      "[18:09:32] Modify product: Uncharacterized N-acetyltransferase YjaB => putative N-acetyltransferase YjaB\n",
      "[18:09:32] Modify product: Uncharacterized protein HI_0431 => putative protein\n",
      "[18:09:32] Modify product: UPF0053 inner membrane protein YgdQ => hypothetical protein\n",
      "[18:09:32] Modify product: Uncharacterized lipoprotein YgdR => putative lipoprotein YgdR\n",
      "[18:09:32] Modify product: Uncharacterized protein YqfG => putative protein YqfG\n",
      "[18:09:32] Modify product: UPF0073 inner membrane protein YqfA => hypothetical protein\n",
      "[18:09:32] Modify product: UPF0267 protein YqfB => hypothetical protein\n",
      "[18:09:32] Modify product: Uncharacterized oxidoreductase YgfF => putative oxidoreductase YgfF\n",
      "[18:09:32] Modify product: UPF0149 protein HI_0817 => hypothetical protein\n",
      "[18:09:33] Modify product: UPF0301 protein YqgE => hypothetical protein\n",
      "[18:09:33] Modify product: Uncharacterized protein YggR => putative protein YggR\n",
      "[18:09:33] Modify product: UPF0235 protein YggU => hypothetical protein\n",
      "[18:09:33] Modify product: Uncharacterized protein HI_0341 => putative protein\n",
      "[18:09:33] Modify product: Probable Fe(2+)-trafficking protein => putative Fe(2+)-trafficking protein\n",
      "[18:09:33] Modify product: Uncharacterized protein YcgM => putative protein YcgM\n",
      "[18:09:33] Modify product: Probable autotransporter ROD_p1121 => putative autotransporter\n",
      "[18:09:33] Modify product: Uncharacterized protein YmgD => putative protein YmgD\n",
      "[18:09:33] Modify product: Probable cyclic di-GMP phosphodiesterase PdeG => putative cyclic di-GMP phosphodiesterase PdeG\n",
      "[18:09:33] Modify product: Uncharacterized protein YmgC => putative protein YmgC\n",
      "[18:09:33] Modify product: Probable two-component-system connector protein AriR => putative two-component-system connector protein AriR\n",
      "[18:09:33] Modify product: Probable two-component-system connector protein YcgZ => putative two-component-system connector protein YcgZ\n",
      "[18:09:33] Modify product: Uncharacterized protein YdfO => putative protein YdfO\n",
      "[18:09:33] Modify product: Probable pyridine nucleotide-disulfide oxidoreductase RclA => putative pyridine nucleotide-disulfide oxidoreductase RclA\n",
      "[18:09:33] Modify product: Uncharacterized membrane protein YkgR => putative membrane protein YkgR\n",
      "[18:09:33] Modify product: Probable fimbrial chaperone EcpB => putative fimbrial chaperone EcpB\n",
      "[18:09:33] Modify product: Probable outer membrane usher protein EcpC => putative outer membrane usher protein EcpC\n",
      "[18:09:33] Modify product: Probable fimbrial chaperone EcpE => putative fimbrial chaperone EcpE\n",
      "[18:09:33] Modify product: UPF0255 protein VV1_0328 => hypothetical protein\n",
      "[18:09:33] Modify product: Uncharacterized N-acetyltransferase YafP => putative N-acetyltransferase YafP\n",
      "[18:09:33] Modify product: Uncharacterized methyltransferase YcgJ => putative methyltransferase YcgJ\n",
      "[18:09:33] Modify product: Uncharacterized oxidoreductase YdgJ => putative oxidoreductase YdgJ\n",
      "[18:09:33] Modify product: Uncharacterized protein YnhF => putative protein YnhF\n",
      "[18:09:33] Modify product: Uncharacterized HTH-type transcriptional regulator YahB => putative HTH-type transcriptional regulator YahB\n",
      "[18:09:33] Modify product: Probable autotransporter ROD_p1121 => putative autotransporter\n",
      "[18:09:33] Modify product: Uncharacterized protein YdhT => putative protein YdhT\n",
      "[18:09:33] Modify product: Uncharacterized ferredoxin-like protein YdhX => putative ferredoxin-like protein YdhX\n",
      "[18:09:33] Modify product: Uncharacterized protein YdhW => putative protein YdhW\n",
      "[18:09:33] Modify product: Uncharacterized oxidoreductase YdhV => putative oxidoreductase YdhV\n",
      "[18:09:33] Modify product: Uncharacterized ferredoxin-like protein YdhY => putative ferredoxin-like protein YdhY\n",
      "[18:09:33] Modify product: Probable L,D-transpeptidase YnhG => putative L,D-transpeptidase YnhG\n",
      "[18:09:33] Modify product: Probable ATP-dependent transporter SufC => putative ATP-dependent transporter SufC\n",
      "[18:09:33] Modify product: Uncharacterized protein HI_1163 => putative protein\n",
      "[18:09:33] Modify product: Uncharacterized protein YdiE => putative protein YdiE\n",
      "[18:09:33] Modify product: Prophage tail fiber assembly protein homolog TfaE => Prophage tail fiber assembly protein TfaE\n",
      "[18:09:33] Modify product: Protein SlyX homolog => Protein SlyX \n",
      "[18:09:33] Modify product: Probable deferrochelatase/peroxidase YfeX => putative deferrochelatase/peroxidase YfeX\n",
      "[18:09:33] Modify product: Uncharacterized protein YfeY => putative protein YfeY\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[18:09:33] Modify product: Uncharacterized protein Rv2575 => putative protein\n",
      "[18:09:33] Modify product: Probable formate transporter 2 => putative formate transporter 2\n",
      "[18:09:33] Modify product: Uncharacterized protein YfgD => putative protein YfgD\n",
      "[18:09:33] Modify product: Uncharacterized protein YfgJ => putative protein YfgJ\n",
      "[18:09:33] Modify product: UPF0070 protein HI_0370 => hypothetical protein\n",
      "[18:09:33] Modify product: Uncharacterized lipoprotein YfhM => putative lipoprotein YfhM\n",
      "[18:09:33] Modify product: Uncharacterized protein Rv2307c => putative protein\n",
      "[18:09:33] Modify product: Probable 3-phenylpropionic acid transporter => putative 3-phenylpropionic acid transporter\n",
      "[18:09:33] Modify product: Uncharacterized protein YphB => putative protein YphB\n",
      "[18:09:33] Modify product: Uncharacterized tRNA/rRNA methyltransferase YfiF => putative tRNA/rRNA methyltransferase YfiF\n",
      "[18:09:33] Modify product: Probable cadaverine/lysine antiporter => putative cadaverine/lysine antiporter\n",
      "[18:09:33] Modify product: Uncharacterized protein YjdJ => putative protein YjdJ\n",
      "[18:09:33] Modify product: Probable transcriptional regulator PhnF => putative transcriptional regulator PhnF\n",
      "[18:09:33] Modify product: Uncharacterized ferredoxin-like protein YdhX => putative ferredoxin-like protein YdhX\n",
      "[18:09:33] Modify product: Probable cyclic di-GMP phosphodiesterase PdeC => putative cyclic di-GMP phosphodiesterase PdeC\n",
      "[18:09:33] Modify product: Uncharacterized protein YjbR => putative protein YjbR\n",
      "[18:09:33] Modify product: UPF0337 protein YjbJ => hypothetical protein\n",
      "[18:09:33] Modify product: Uncharacterized protein YniD => putative protein YniD\n",
      "[18:09:33] Modify product: Uncharacterized HTH-type transcriptional regulator YdjF => putative HTH-type transcriptional regulator YdjF\n",
      "[18:09:33] Modify product: Uncharacterized sugar kinase YdjH => putative sugar kinase YdjH\n",
      "[18:09:33] Modify product: Probable fructose-bisphosphate aldolase => putative fructose-bisphosphate aldolase\n",
      "[18:09:33] Modify product: Uncharacterized zinc-type alcohol dehydrogenase-like protein YdjJ => putative zinc-type alcohol dehydrogenase-like protein YdjJ\n",
      "[18:09:33] Modify product: Stress response UPF0229 protein YhbH => hypothetical protein\n",
      "[18:09:33] Modify product: Probable diguanylate cyclase CdgI => putative diguanylate cyclase CdgI\n",
      "[18:09:33] Modify product: Probable diguanylate cyclase DgcJ => putative diguanylate cyclase DgcJ\n",
      "[18:09:33] Modify product: Uncharacterized protein YeaR => putative protein YeaR\n",
      "[18:09:33] Modify product: Probable ATP-dependent helicase DinG homolog => putative ATP-dependent helicase DinG \n",
      "[18:09:33] Modify product: Probable cyclic di-GMP phosphodiesterase PdeD => putative cyclic di-GMP phosphodiesterase PdeD\n",
      "[18:09:33] Modify product: UPF0053 inner membrane protein YoaE => hypothetical protein\n",
      "[18:09:33] Modify product: UPF0266 membrane protein YobD => hypothetical protein\n",
      "[18:09:33] Modify product: Probable manganese efflux pump MntP => putative manganese efflux pump MntP\n",
      "[18:09:33] Modify product: Uncharacterized protein HI_1672 => putative protein\n",
      "[18:09:33] Modify product: Uncharacterized protein YebV => putative protein YebV\n",
      "[18:09:33] Modify product: Probable transcriptional regulatory protein YebC => putative transcriptional regulatory protein YebC\n",
      "[18:09:33] Modify product: Uncharacterized protein YaeQ => putative protein YaeQ\n",
      "[18:09:33] Modify product: UPF0253 protein YaeP => hypothetical protein\n",
      "[18:09:33] Modify product: UPF0126 inner membrane protein YadS => hypothetical protein\n",
      "[18:09:33] Modify product: Uncharacterized fimbrial-like protein YadN => putative fimbrial-like protein YadN\n",
      "[18:09:33] Modify product: Probable fimbrial chaperone YadV => putative fimbrial chaperone YadV\n",
      "[18:09:33] Modify product: Uncharacterized fimbrial-like protein YadM => putative fimbrial-like protein YadM\n",
      "[18:09:33] Modify product: Uncharacterized fimbrial-like protein YadL => putative fimbrial-like protein YadL\n",
      "[18:09:33] Modify product: Uncharacterized fimbrial-like protein YadK => putative fimbrial-like protein YadK\n",
      "[18:09:33] Modify product: Uncharacterized fimbrial-like protein YadC => putative fimbrial-like protein YadC\n",
      "[18:09:33] Modify product: Uncharacterized ABC transporter ATP-binding protein YadG => putative ABC transporter ATP-binding protein YadG\n",
      "[18:09:34] Modify product: UPF0231 protein YacL => hypothetical protein\n",
      "[18:09:34] Modify product: Probable peptidoglycan glycosyltransferase FtsW => putative peptidoglycan glycosyltransferase FtsW\n",
      "[18:09:34] Modify product: UPF0394 inner membrane protein YeeE => hypothetical protein\n",
      "[18:09:34] Modify product: UPF0053 inner membrane protein YoaE => hypothetical protein\n",
      "[18:09:34] Modify product: Probable diguanylate cyclase DgcE => putative diguanylate cyclase DgcE\n",
      "[18:09:34] Modify product: UPF0339 protein YegP => hypothetical protein\n",
      "[18:09:34] Modify product: Uncharacterized protease YdcP => putative protease YdcP\n",
      "[18:09:34] Modify product: Uncharacterized fimbrial-like protein YehA => putative fimbrial-like protein YehA\n",
      "[18:09:34] Modify product: Probable fimbrial chaperone YehC => putative fimbrial chaperone YehC\n",
      "[18:09:34] Modify product: Uncharacterized fimbrial-like protein YehD => putative fimbrial-like protein YehD\n",
      "[18:09:34] Modify product: Uncharacterized protein YehL => putative protein YehL\n",
      "[18:09:34] Modify product: Uncharacterized lipoprotein YehR => putative lipoprotein YehR\n",
      "[18:09:34] Modify product: UPF0387 membrane protein YohO => hypothetical protein\n",
      "[18:09:34] Modify product: UPF0299 membrane protein YohJ => hypothetical protein\n",
      "[18:09:34] Modify product: UPF0324 inner membrane protein YeiH => hypothetical protein\n",
      "[18:09:34] Modify product: UPF0153 protein YeiW => hypothetical protein\n",
      "[18:09:34] Modify product: Probable cyclic di-GMP phosphodiesterase PdeN => putative cyclic di-GMP phosphodiesterase PdeN\n",
      "[18:09:34] Modify product: Uncharacterized ABC transporter ATP-binding protein YejF => putative ABC transporter ATP-binding protein YejF\n",
      "[18:09:34] Modify product: Uncharacterized protein YejG => putative protein YejG\n",
      "[18:09:34] Modify product: UPF0352 protein YejL => hypothetical protein\n",
      "[18:09:34] Modify product: Probable autotransporter ROD_p1121 => putative autotransporter\n",
      "[18:09:34] Modify product: Probable autotransporter YfaL => putative autotransporter YfaL\n",
      "[18:09:34] Modify product: Probable 4-deoxy-4-formamido-L-arabinose-phosphoundecaprenol deformylase ArnD => putative 4-deoxy-4-formamido-L-arabinose-phosphoundecaprenol deformylase ArnD\n",
      "[18:09:34] Modify product: Probable 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnE => putative 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnE\n",
      "[18:09:34] Modify product: Probable 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnF => putative 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnF\n",
      "[18:09:34] Modify product: UPF0039 protein Rv2851c => hypothetical protein\n",
      "[18:09:34] Modify product: UPF0304 protein YfbU => hypothetical protein\n",
      "[18:09:34] Modify product: UPF0208 membrane protein YfbV => hypothetical protein\n",
      "[18:09:34] Modify product: UPF0246 protein YaaA => hypothetical protein\n",
      "[18:09:34] Modify product: Probable inner membrane protein Smp => putative inner membrane protein Smp\n",
      "[18:09:34] Modify product: Uncharacterized protein YjjI => putative protein YjjI\n",
      "[18:09:34] Modify product: Uncharacterized metal-dependent hydrolase YjjV => putative metal-dependent hydrolase YjjV\n",
      "[18:09:34] Modify product: UPF0442 protein YjjB => hypothetical protein\n",
      "[18:09:34] Modify product: Probable HTH-type transcriptional regulator LgoR => putative HTH-type transcriptional regulator LgoR\n",
      "[18:09:34] Modify product: Probable L-galactonate transporter => putative L-galactonate transporter\n",
      "[18:09:34] Modify product: Uncharacterized GTP-binding protein YjiA => putative GTP-binding protein YjiA\n",
      "[18:09:34] Modify product: Probable type I restriction enzyme BthVORF4518P M protein => putative type I restriction enzymeP M protein\n",
      "[18:09:34] Modify product: Uncharacterized protein YjiK => putative protein YjiK\n",
      "[18:09:34] Modify product: Uncharacterized protein YccJ => putative protein YccJ\n",
      "[18:09:34] Modify product: Uncharacterized lipoprotein GfcB => putative lipoprotein GfcB\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[18:09:34] Modify product: Probable Ni/Fe-hydrogenase 1 B-type cytochrome subunit => putative Ni/Fe-hydrogenase 1 B-type cytochrome subunit\n",
      "[18:09:34] Modify product: Uncharacterized protein YccU => putative protein YccU\n",
      "[18:09:34] Modify product: UPF0319 protein HI_1681 => hypothetical protein\n",
      "[18:09:34] Modify product: Uncharacterized protein YcbX => putative protein YcbX\n",
      "[18:09:34] Modify product: Uncharacterized fimbrial chaperone YcbF => putative fimbrial chaperone YcbF\n",
      "[18:09:34] Modify product: Uncharacterized fimbrial-like protein YcbV => putative fimbrial-like protein YcbV\n",
      "[18:09:34] Modify product: Uncharacterized fimbrial-like protein YcbU => putative fimbrial-like protein YcbU\n",
      "[18:09:34] Modify product: Uncharacterized fimbrial-like protein ElfG => putative fimbrial-like protein ElfG\n",
      "[18:09:34] Modify product: Probable outer membrane usher protein ElfC => putative outer membrane usher protein ElfC\n",
      "[18:09:34] Modify product: Probable fimbrial chaperone protein ElfD => putative fimbrial chaperone protein ElfD\n",
      "[18:09:34] Modify product: UPF0434 protein YcaR => hypothetical protein\n",
      "[18:09:34] Modify product: Uncharacterized metalloprotease YcaL => putative metalloprotease YcaL\n",
      "[18:09:34] Modify product: Probable formate transporter 1 => putative formate transporter 1\n",
      "[18:09:34] Modify product: Uncharacterized MFS-type transporter YcaD => putative MFS-type transporter YcaD\n",
      "[18:09:34] Modify product: Probable hydrolase YcaC => putative hydrolase YcaC\n",
      "[18:09:34] Modify product: UPF0145 protein YbjQ => hypothetical protein\n",
      "[18:09:34] Modify product: Uncharacterized lipoprotein YbjP => putative lipoprotein YbjP\n",
      "[18:09:34] Modify product: Uncharacterized Nudix hydrolase YfcD => putative Nudix hydrolase YfcD\n",
      "[18:09:34] Modify product: Epimerase family protein SA0724 => Epimerase family protein\n",
      "[18:09:34] Modify product: Uncharacterized MFS-type transporter YfcJ => putative MFS-type transporter YfcJ\n",
      "[18:09:34] Modify product: Probable membrane transporter protein YfcA => putative membrane transporter protein YfcA\n",
      "[18:09:34] Modify product: Uncharacterized protein YfcO => putative protein YfcO\n",
      "[18:09:34] Modify product: Uncharacterized fimbrial-like protein YfcP => putative fimbrial-like protein YfcP\n",
      "[18:09:34] Modify product: Uncharacterized fimbrial-like protein YfcQ => putative fimbrial-like protein YfcQ\n",
      "[18:09:34] Modify product: Uncharacterized fimbrial-like protein YfcR => putative fimbrial-like protein YfcR\n",
      "[18:09:34] Modify product: Probable fimbrial chaperone YfcS => putative fimbrial chaperone YfcS\n",
      "[18:09:34] Modify product: Uncharacterized fimbrial-like protein YfcV => putative fimbrial-like protein YfcV\n",
      "[18:09:34] Modify product: UPF0381 protein YiiS => hypothetical protein\n",
      "[18:09:34] Modify product: Probable phospholipid-binding lipoprotein MlaA => putative phospholipid-binding lipoprotein MlaA\n",
      "[18:09:34] Modify product: Uncharacterized protein HI_1410 => putative protein\n",
      "[18:09:34] Modify product: Uncharacterized fimbrial-like protein SfmF => putative fimbrial-like protein SfmF\n",
      "[18:09:34] Modify product: Uncharacterized fimbrial-like protein SfmH => putative fimbrial-like protein SfmH\n",
      "[18:09:34] Modify product: Probable fimbrial chaperone SfmC => putative fimbrial chaperone SfmC\n",
      "[18:09:34] Modify product: Uncharacterized fimbrial-like protein SfmA => putative fimbrial-like protein SfmA\n",
      "[18:09:34] Modify product: Bifunctional protein FolD => Bifunctional protein FolD protein\n",
      "[18:09:34] Modify product: Uncharacterized protein YbcJ => putative protein YbcJ\n",
      "[18:09:34] Modify product: Uncharacterized protein YbbC => putative protein YbbC\n",
      "[18:09:34] Modify product: Uncharacterized protein YbbN => putative protein YbbN\n",
      "[18:09:34] Modify product: Probable iron export permease protein FetB => putative iron export permease protein FetB\n",
      "[18:09:34] Modify product: Probable iron export ATP-binding protein FetA => putative iron export ATP-binding protein FetA\n",
      "[18:09:34] Modify product: Uncharacterized HTH-type transcriptional regulator YbaQ => putative HTH-type transcriptional regulator YbaQ\n",
      "[18:09:34] Modify product: Probable cyclic di-GMP phosphodiesterase PdeB => putative cyclic di-GMP phosphodiesterase PdeB\n",
      "[18:09:34] Modify product: Uncharacterized protein YbaA => putative protein YbaA\n",
      "[18:09:34] Modify product: Uncharacterized lipoprotein YbaY => putative lipoprotein YbaY\n",
      "[18:09:34] Modify product: Probable multidrug resistance ABC transporter ATP-binding/permease protein YheI => putative multidrug resistance ABC transporter ATP-binding/permease protein YheI\n",
      "[18:09:34] Modify product: Uncharacterized protein HI_1008 => putative protein\n",
      "[18:09:34] Modify product: UPF0234 protein YajQ => hypothetical protein\n",
      "[18:09:34] Modify product: Uncharacterized lipoprotein YajI => putative lipoprotein YajI\n",
      "[18:09:34] Modify product: Uncharacterized lipoprotein YajI => putative lipoprotein YajI\n",
      "[18:09:35] Modify product: Uncharacterized protein YaiA => putative protein YaiA\n",
      "[18:09:35] Modify product: UPF0178 protein YaiI => hypothetical protein\n",
      "[18:09:35] Modify product: Probable diguanylate cyclase DgcC => putative diguanylate cyclase DgcC\n",
      "[18:09:35] Modify product: Uncharacterized transporter YycB => putative transporter YycB\n",
      "[18:09:35] Modify product: Uncharacterized HTH-type transcriptional regulator YahB => putative HTH-type transcriptional regulator YahB\n",
      "[18:09:35] Modify product: Uncharacterized protein YbjN => putative protein YbjN\n",
      "[18:09:35] Modify product: Probable cyclic di-GMP phosphodiesterase PdeI => putative cyclic di-GMP phosphodiesterase PdeI\n",
      "[18:09:35] Modify product: Uncharacterized protein YbiU => putative protein YbiU\n",
      "[18:09:35] Modify product: Uncharacterized ABC transporter ATP-binding protein YbiT => putative ABC transporter ATP-binding protein YbiT\n",
      "[18:09:35] Modify product: Probable L,D-transpeptidase YbiS => putative L,D-transpeptidase YbiS\n",
      "[18:09:35] Modify product: Uncharacterized protein McbA => putative protein McbA\n",
      "[18:09:35] Modify product: PKHD-type hydroxylase Sbal_3634 => PKHD-type hydroxylase\n",
      "[18:09:35] Modify product: Uncharacterized protein YbiI => putative protein YbiI\n",
      "[18:09:35] Modify product: Uncharacterized protein YbiB => putative protein YbiB\n",
      "[18:09:35] Modify product: Probable ATP-dependent helicase DinG => putative ATP-dependent helicase DinG\n",
      "[18:09:35] Modify product: UPF0194 membrane protein YbhG => hypothetical protein\n",
      "[18:09:35] Modify product: Probable multidrug ABC transporter ATP-binding protein YbhF => putative multidrug ABC transporter ATP-binding protein YbhF\n",
      "[18:09:35] Modify product: Probable multidrug ABC transporter permease YbhS => putative multidrug ABC transporter permease YbhS\n",
      "[18:09:35] Modify product: Probable multidrug ABC transporter permease YbhR => putative multidrug ABC transporter permease YbhR\n",
      "[18:09:35] Modify product: UPF0098 protein YbhB => hypothetical protein\n",
      "[18:09:35] Modify product: Probable diguanylate cyclase CdgI => putative diguanylate cyclase CdgI\n",
      "[18:09:35] Modify product: Prophage tail fiber assembly protein homolog TfaE => Prophage tail fiber assembly protein TfaE\n",
      "[18:09:35] Modify product: Uncharacterized protein HI_0933 => putative protein\n",
      "[18:09:35] Modify product: Probable copper-binding protein PcoE => putative copper-binding protein PcoE\n",
      "[18:09:35] Modify product: Uncharacterized MFS-type transporter YhhS => putative MFS-type transporter YhhS\n",
      "[18:09:35] Modify product: Uncharacterized protein YhhA => putative protein YhhA\n",
      "[18:09:35] Modify product: Uncharacterized oxidoreductase YhhX => putative oxidoreductase YhhX\n",
      "[18:09:35] Modify product: UPF0056 inner membrane protein YhgN => hypothetical protein\n",
      "[18:09:35] Modify product: Probable [Fe-S]-dependent transcriptional repressor FeoC => putative [Fe-S]-dependent transcriptional repressor FeoC\n",
      "[18:09:35] Modify product: Putative membrane protein IgaA homolog => Putative membrane protein IgaA \n",
      "[18:09:35] Modify product: Uncharacterized protein YhfX => putative protein YhfX\n",
      "[18:09:35] Modify product: Uncharacterized protein YhfS => putative protein YhfS\n",
      "[18:09:35] Modify product: Probable fructoselysine utilization operon transcriptional repressor => putative fructoselysine utilization operon transcriptional repressor\n",
      "[18:09:35] Modify product: Probable fructoselysine/psicoselysine transporter FrlA => putative fructoselysine/psicoselysine transporter FrlA\n",
      "[18:09:35] Modify product: Uncharacterized protein YhfG => putative protein YhfG\n",
      "[18:09:35] Modify product: Probable adenosine monophosphate-protein transferase fic => putative adenosine monophosphate-protein transferase fic\n",
      "[18:09:35] Modify product: UPF0270 protein PA3463 => hypothetical protein\n",
      "[18:09:35] Modify product: Uncharacterized ABC transporter ATP-binding protein YheS => putative ABC transporter ATP-binding protein YheS\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[18:09:35] Modify product: Uncharacterized protein YhcH => putative protein YhcH\n",
      "[18:09:35] Modify product: Probable serine transporter => putative serine transporter\n",
      "[18:09:35] Modify product: UPF0597 protein YhaM => hypothetical protein\n",
      "[18:09:35] Modify product: Uncharacterized protein YqjD => putative protein YqjD\n",
      "[18:09:35] Modify product: Uncharacterized HTH-type transcriptional regulator YbbH => putative HTH-type transcriptional regulator YbbH\n",
      "[18:09:35] Modify product: Probable RNA polymerase sigma factor FecI => putative RNA polymerase sigma factor FecI\n",
      "[18:09:35] Modify product: UPF0271 protein YbgL => hypothetical protein\n",
      "[18:09:35] Modify product: Uncharacterized protein YbgK => putative protein YbgK\n",
      "[18:09:35] Modify product: Uncharacterized protein YbgJ => putative protein YbgJ\n",
      "[18:09:35] Modify product: GTP cyclohydrolase 1 type 2 homolog => GTP cyclohydrolase 1 type 2 \n",
      "[18:09:35] Modify product: UPF0401 protein YkfF => hypothetical protein\n",
      "[18:09:35] Modify product: Uncharacterized ABC transporter ATP-binding protein YknY => putative ABC transporter ATP-binding protein YknY\n",
      "[18:09:35] Modify product: UPF0265 protein YeeX => hypothetical protein\n",
      "[18:09:35] Modify product: Uncharacterized inner membrane transporter YicL => putative inner membrane transporter YicL\n",
      "[18:09:35] Modify product: UPF0401 protein YkfF => hypothetical protein\n",
      "[18:09:35] Modify product: Uncharacterized protein YggR => putative protein YggR\n",
      "[18:09:35] Modify product: UPF0401 protein YkfF => hypothetical protein\n",
      "[18:09:35] Modify product: Probable protein kinase UbiB => putative protein kinase UbiB\n",
      "[18:09:35] Modify product: Probable transport protein YifK => putative transport protein YifK\n",
      "[18:09:35] Modify product: Probable ECA polymerase => putative ECA polymerase\n",
      "[18:09:35] Modify product: UPF0438 protein YifE => hypothetical protein\n",
      "[18:09:35] Modify product: UPF0056 inner membrane protein MarC => hypothetical protein\n",
      "[18:09:35] Modify product: Probable amino-acid metabolite efflux pump => putative amino-acid metabolite efflux pump\n",
      "[18:09:35] Modify product: Uncharacterized protein YneM => putative protein YneM\n",
      "[18:09:35] Modify product: UPF0060 membrane protein YnfA => hypothetical protein\n",
      "[18:09:35] Modify product: UPF0482 protein YnfB => hypothetical protein\n",
      "[18:09:35] Modify product: Uncharacterized protein YnfD => putative protein YnfD\n",
      "[18:09:35] Modify product: Probable dimethyl sulfoxide reductase chain YnfF => putative dimethyl sulfoxide reductase chain YnfF\n",
      "[18:09:35] Modify product: Probable glutamine ABC transporter permease protein GlnM => putative glutamine ABC transporter permease protein GlnM\n",
      "[18:09:35] Modify product: UPF0401 protein YkfF => hypothetical protein\n",
      "[18:09:35] Modify product: Probable L,D-transpeptidase YcfS => putative L,D-transpeptidase YcfS\n",
      "[18:09:35] Modify product: UPF0227 protein YcfP => hypothetical protein\n",
      "[18:09:35] Modify product: Uncharacterized metal-dependent hydrolase YcfH => putative metal-dependent hydrolase YcfH\n",
      "[18:09:35] Modify product: UPF0502 protein YceH => hypothetical protein\n",
      "[18:09:35] Modify product: Uncharacterized lipoprotein YceB => putative lipoprotein YceB\n",
      "[18:09:35] Modify product: Uncharacterized protein YceO => putative protein YceO\n",
      "[18:09:35] Modify product: Cytochrome b561 homolog 2 => Cytochrome b561 \n",
      "[18:09:35] Modify product: UPF0176 protein YceA => hypothetical protein\n",
      "[18:09:35] Modify product: Probable phosphatase YcdX => putative phosphatase YcdX\n",
      "[18:09:35] Modify product: Uncharacterized protein Rv3292 => putative protein\n",
      "[18:09:35] Modify product: Uncharacterized protein YnbD => putative protein YnbD\n",
      "[18:09:35] Modify product: Uncharacterized protein YnbC => putative protein YnbC\n",
      "[18:09:36] Modify product: Probable DNA endonuclease SmrA => putative DNA endonuclease SmrA\n",
      "[18:09:36] Modify product: Uncharacterized protein YcjY => putative protein YcjY\n",
      "[18:09:36] Modify product: UPF0283 membrane protein YcjF => hypothetical protein\n",
      "[18:09:36] Modify product: Uncharacterized protein YcjX => putative protein YcjX\n",
      "[18:09:36] Modify product: Uncharacterized protein HI_1162 => putative protein\n",
      "[18:09:36] Modify product: Uncharacterized protein YciH => putative protein YciH\n",
      "[18:09:36] Modify product: Uncharacterized protein YmiA => putative protein YmiA\n",
      "[18:09:36] Modify product: Probable protease SohB => putative protease SohB\n",
      "[18:09:36] Modify product: Uncharacterized oxidoreductase YciK => putative oxidoreductase YciK\n",
      "[18:09:36] Modify product: Uncharacterized protein YciO => putative protein YciO\n",
      "[18:09:36] Modify product: Probable tRNA-dihydrouridine synthase => putative tRNA-dihydrouridine synthase\n",
      "[18:09:36] Modify product: Uncharacterized membrane protein YhdT => putative membrane protein YhdT\n",
      "[18:09:36] Modify product: Probable acrylyl-CoA reductase AcuI => putative acrylyl-CoA reductase AcuI\n",
      "[18:09:36] Modify product: Uncharacterized protein YhcO => putative protein YhcO\n",
      "[18:09:36] Modify product: Uncharacterized protein YhcH => putative protein YhcH\n",
      "[18:09:36] Modify product: Uncharacterized protein YrbL => putative protein YrbL\n",
      "[18:09:36] Modify product: Probable phospholipid import ATP-binding protein MlaF => putative phospholipid import ATP-binding protein MlaF\n",
      "[18:09:36] Modify product: Probable phospholipid ABC transporter permease protein MlaE => putative phospholipid ABC transporter permease protein MlaE\n",
      "[18:09:36] Modify product: Probable phospholipid ABC transporter-binding protein MlaD => putative phospholipid ABC transporter-binding protein MlaD\n",
      "[18:09:36] Modify product: Probable phospholipid-binding protein MlaC => putative phospholipid-binding protein MlaC\n",
      "[18:09:36] Modify product: Probable phospholipid ABC transporter-binding protein MlaB => putative phospholipid ABC transporter-binding protein MlaB\n",
      "[18:09:36] Modify product: Uncharacterized inner membrane transporter YhbE => putative inner membrane transporter YhbE\n",
      "[18:09:36] Modify product: Uncharacterized protease YhbU => putative protease YhbU\n",
      "[18:09:36] Modify product: UPF0213 protein BH0048 => hypothetical protein\n",
      "[18:09:36] Modify product: Uncharacterized fimbrial-like protein YraK => putative fimbrial-like protein YraK\n",
      "[18:09:36] Modify product: Probable fimbrial chaperone YraI => putative fimbrial chaperone YraI\n",
      "[18:09:36] Modify product: Uncharacterized fimbrial-like protein YraH => putative fimbrial-like protein YraH\n",
      "[18:09:36] Modify product: Probable galactarate transporter => putative galactarate transporter\n",
      "[18:09:36] Modify product: Probable glycerol-3-phosphate acyltransferase => putative glycerol-3-phosphate acyltransferase\n",
      "[18:09:36] Modify product: Uncharacterized fimbrial-like protein YbgD => putative fimbrial-like protein YbgD\n",
      "[18:09:36] Modify product: Uncharacterized protein YqiC => putative protein YqiC\n",
      "[18:09:36] Modify product: Probable quinol monooxygenase YgiN => putative quinol monooxygenase YgiN\n",
      "[18:09:36] Modify product: Uncharacterized oxidoreductase YghA => putative oxidoreductase YghA\n",
      "[18:09:36] Modify product: Probable Ni/Fe-hydrogenase 2 b-type cytochrome subunit => putative Ni/Fe-hydrogenase 2 b-type cytochrome subunit\n",
      "[18:09:36] Modify product: Uncharacterized protein YtfJ => putative protein YtfJ\n",
      "[18:09:36] Modify product: Uncharacterized FAD-linked oxidoreductase Rv2280 => putative FAD-linked oxidoreductase\n",
      "[18:09:36] Modify product: UPF0758 protein YsxA => hypothetical protein\n",
      "[18:09:36] Modify product: Probable transport protein HsrA => putative transport protein HsrA\n",
      "[18:09:36] Modify product: Probable outer membrane usher protein ElfC => putative outer membrane usher protein ElfC\n",
      "[18:09:36] Modify product: Uncharacterized fimbrial-like protein ElfG => putative fimbrial-like protein ElfG\n",
      "[18:09:36] Modify product: UPF0167 protein CbrC => hypothetical protein\n",
      "[18:09:36] Modify product: Probable 2-dehydro-3-deoxygalactonokinase DgoK1 => putative 2-dehydro-3-deoxygalactonokinase DgoK1\n",
      "[18:09:36] Modify product: Uncharacterized symporter YidK => putative symporter YidK\n",
      "[18:09:36] Modify product: Uncharacterized protein YgcP => putative protein YgcP\n",
      "[18:09:36] Modify product: Uncharacterized protein YqcG => putative protein YqcG\n",
      "[18:09:36] Modify product: Probable glucarate transporter => putative glucarate transporter\n",
      "[18:09:36] Modify product: Uncharacterized protein PA3435 => putative protein\n",
      "[18:09:36] Modify product: Uncharacterized protein YqcC => putative protein YqcC\n",
      "[18:09:36] Modify product: Uncharacterized HTH-type transcriptional regulator YdjF => putative HTH-type transcriptional regulator YdjF\n",
      "[18:09:36] Modify product: UPF0382 inner membrane protein YgdD => hypothetical protein\n",
      "[18:09:36] Modify product: Uncharacterized lipoprotein YgdR => putative lipoprotein YgdR\n",
      "[18:09:36] Modify product: Uncharacterized protein YdfO => putative protein YdfO\n",
      "[18:09:36] Modify product: Uncharacterized protein YlcG => putative protein YlcG\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[18:09:36] Modify product: Uncharacterized HTH-type transcriptional regulator HI_1476 => putative HTH-type transcriptional regulator\n",
      "[18:09:36] Modify product: UPF0259 membrane protein YciC => hypothetical protein\n",
      "[18:09:36] Modify product: Probable intracellular septation protein A => putative intracellular septation protein A\n",
      "[18:09:36] Modify product: Uncharacterized acyl-CoA thioester hydrolase HI_0827 => putative acyl-CoA thioester hydrolase\n",
      "[18:09:36] Modify product: UPF0263 protein YciU => hypothetical protein\n",
      "[18:09:36] Modify product: UPF0056 inner membrane protein MarC => hypothetical protein\n",
      "[18:09:36] Modify product: Uncharacterized NTE family protein Rv1063c => putative NTE family protein\n",
      "[18:09:36] Modify product: UPF0225 protein YchJ => hypothetical protein\n",
      "[18:09:36] Modify product: Probable multidrug resistance protein EmrY => putative multidrug resistance protein EmrY\n",
      "[18:09:36] Modify product: Probable multidrug resistance protein EmrK => putative multidrug resistance protein EmrK\n",
      "[18:09:36] Modify product: Probable cyclic di-GMP phosphodiesterase PdeA => putative cyclic di-GMP phosphodiesterase PdeA\n",
      "[18:09:36] Modify product: Probable L-galactonate transporter => putative L-galactonate transporter\n",
      "[18:09:36] Modify product: Probable cystine transporter YijE => putative cystine transporter YijE\n",
      "[18:09:36] Modify product: Probable lipoprotein YiaD => putative lipoprotein YiaD\n",
      "[18:09:36] Modify product: Uncharacterized N-acetyltransferase YjaB => putative N-acetyltransferase YjaB\n",
      "[18:09:36] Modify product: Uncharacterized MFS-type transporter YhjX => putative MFS-type transporter YhjX\n",
      "[18:09:36] Modify product: Probable major fimbrial subunit LpfA => putative major fimbrial subunit LpfA\n",
      "[18:09:36] Modify product: Probable fimbrial chaperone LpfB => putative fimbrial chaperone LpfB\n",
      "[18:09:36] Modify product: Probable minor fimbrial subunit LpfD => putative minor fimbrial subunit LpfD\n",
      "[18:09:36] Modify product: Probable fimbrial subunit LpfE => putative fimbrial subunit LpfE\n",
      "[18:09:36] Modify product: Probable diguanylate cyclase DgcT => putative diguanylate cyclase DgcT\n",
      "[18:09:36] Modify product: Probable malonic semialdehyde reductase RutE => putative malonic semialdehyde reductase RutE\n",
      "[18:09:36] Modify product: Uncharacterized isomerase YddE => putative isomerase YddE\n",
      "[18:09:36] Modify product: Probable nitrate reductase molybdenum cofactor assembly chaperone NarW => putative nitrate reductase molybdenum cofactor assembly chaperone NarW\n",
      "[18:09:36] Modify product: Uncharacterized HTH-type transcriptional regulator YddM => putative HTH-type transcriptional regulator YddM\n",
      "[18:09:36] Modify product: Probable D,D-dipeptide transport ATP-binding protein DdpF => putative D,D-dipeptide transport ATP-binding protein DdpF\n",
      "[18:09:36] Modify product: Probable D,D-dipeptide transport ATP-binding protein DdpD => putative D,D-dipeptide transport ATP-binding protein DdpD\n",
      "[18:09:36] Modify product: Probable D,D-dipeptide transport system permease protein DdpC => putative D,D-dipeptide transport system permease protein DdpC\n",
      "[18:09:36] Modify product: Probable D,D-dipeptide transport system permease protein DdpB => putative D,D-dipeptide transport system permease protein DdpB\n",
      "[18:09:36] Modify product: Probable D,D-dipeptide-binding periplasmic protein DdpA => putative D,D-dipeptide-binding periplasmic protein DdpA\n",
      "[18:09:36] Modify product: Probable glutamate/gamma-aminobutyrate antiporter => putative glutamate/gamma-aminobutyrate antiporter\n",
      "[18:09:36] Modify product: Probable cyclic di-GMP phosphodiesterase PdeK => putative cyclic di-GMP phosphodiesterase PdeK\n",
      "[18:09:36] Modify product: Uncharacterized protein YbfE => putative protein YbfE\n",
      "[18:09:36] Modify product: Uncharacterized lipoprotein ChiQ => putative lipoprotein ChiQ\n",
      "[18:09:36] Modify product: Uncharacterized fimbrial-like protein YbgD => putative fimbrial-like protein YbgD\n",
      "[18:09:37] Modify product: Uncharacterized protein YedJ => putative protein YedJ\n",
      "[18:09:37] Modify product: Uncharacterized inner membrane transporter YedA => putative inner membrane transporter YedA\n",
      "[18:09:37] Modify product: Probable diguanylate cyclase DgcQ => putative diguanylate cyclase DgcQ\n",
      "[18:09:37] Modify product: Probable diguanylate cyclase DgcQ => putative diguanylate cyclase DgcQ\n",
      "[18:09:37] Modify product: UPF0098 protein YbcL => hypothetical protein\n",
      "[18:09:37] Modify product: Uncharacterized protein YncE => putative protein YncE\n",
      "[18:09:37] Modify product: Uncharacterized protein YdcY => putative protein YdcY\n",
      "[18:09:37] Modify product: Uncharacterized protease YdcP => putative protease YdcP\n",
      "[18:09:37] Modify product: Uncharacterized GST-like protein YibF => putative GST-like protein YibF\n",
      "[18:09:37] Modify product: UPF0250 protein YbeD => hypothetical protein\n",
      "[18:09:37] Modify product: Uncharacterized HTH-type transcriptional regulator YbdO => putative HTH-type transcriptional regulator YbdO\n",
      "[18:09:37] Modify product: Probable zinc-binding alcohol dehydrogenase Rv1895 => putative zinc-binding alcohol dehydrogenase\n",
      "[18:09:37] Modify product: Uncharacterized HTH-type transcriptional regulator YbdO => putative HTH-type transcriptional regulator YbdO\n",
      "[18:09:37] Modify product: Uncharacterized protein YbdM => putative protein YbdM\n",
      "[18:09:37] Modify product: UPF0381 protein YiiS => hypothetical protein\n",
      "[18:09:37] Modify product: UPF0761 membrane protein YihY => hypothetical protein\n",
      "[18:09:37] Modify product: Uncharacterized protein YphB => putative protein YphB\n",
      "[18:09:37] Modify product: Uncharacterized zinc-type alcohol dehydrogenase-like protein YjmD => putative zinc-type alcohol dehydrogenase-like protein YjmD\n",
      "[18:09:37] Modify product: Uncharacterized protein YshB => putative protein YshB\n",
      "[18:09:37] Modify product: Probable GTP-binding protein EngB => putative GTP-binding protein EngB\n",
      "[18:09:37] Modify product: Probable acyltransferase YihG => putative acyltransferase YihG\n",
      "[18:09:37] Modify product: UPF0394 inner membrane protein YedE => hypothetical protein\n",
      "[18:09:37] Modify product: Uncharacterized HTH-type transcriptional regulator HI_1476 => putative HTH-type transcriptional regulator\n",
      "[18:09:37] Modify product: UPF0053 inner membrane protein YfjD => hypothetical protein\n",
      "[18:09:37] Modify product: UPF0125 protein RatB => hypothetical protein\n",
      "[18:09:37] Modify product: Uncharacterized protein YgaU => putative protein YgaU\n",
      "[18:09:37] Modify product: Probable HTH-type transcriptional regulator YgaV => putative HTH-type transcriptional regulator YgaV\n",
      "[18:09:37] Modify product: Uncharacterized protein YgaM => putative protein YgaM\n",
      "[18:09:37] Modify product: Probable transporter SCO4007 => putative transporter\n",
      "[18:09:37] Modify product: Uncharacterized protein YjgL => putative protein YjgL\n",
      "[18:09:37] Modify product: Uncharacterized protein YjgL => putative protein YjgL\n",
      "[18:09:37] Modify product: UPF0307 protein YjgA => hypothetical protein\n",
      "[18:09:37] Modify product: UPF0053 inner membrane protein YtfL => hypothetical protein\n",
      "[18:09:37] Modify product: Uncharacterized protein YtfJ => putative protein YtfJ\n",
      "[18:09:37] Modify product: UPF0126 inner membrane protein YicG => hypothetical protein\n",
      "[18:09:37] Modify product: UPF0701 protein YicC => hypothetical protein\n",
      "[18:09:37] Modify product: UPF0758 protein SP_1088 => hypothetical protein\n",
      "[18:09:37] Modify product: Uncharacterized protein HI_0755 => putative protein\n",
      "[18:09:37] Modify product: Uncharacterized protein YibN => putative protein YibN\n",
      "[18:09:37] Modify product: Uncharacterized protein YibL => putative protein YibL\n",
      "[18:09:37] Modify product: Probable UbiX-like flavin prenyltransferase => putative UbiX-like flavin prenyltransferase\n",
      "[18:09:37] Modify product: Probable L-ascorbate-6-phosphate lactonase UlaG => putative L-ascorbate-6-phosphate lactonase UlaG\n",
      "[18:09:37] Modify product: UPF0719 inner membrane protein YjfL => hypothetical protein\n",
      "[18:09:37] Modify product: UPF0716 protein FxsA => hypothetical protein\n",
      "[18:09:37] Modify product: Probable L,D-transpeptidase ErfK/SrfK => putative L,D-transpeptidase ErfK/SrfK\n",
      "[18:09:37] Modify product: Probable FMN/FAD exporter YeeO => putative FMN/FAD exporter YeeO\n",
      "[18:09:37] Modify product: Probable transcriptional regulatory protein YeeN => putative transcriptional regulatory protein YeeN\n",
      "[18:09:37] Modify product: Putative multidrug export ATP-binding/permease protein SA1683 => Putative multidrug export ATP-binding/permease protein\n",
      "[18:09:37] Modify product: Probable transcriptional regulatory protein YedW => putative transcriptional regulatory protein YedW\n",
      "[18:09:37] Modify product: Probable sensor-like histidine kinase YedV => putative sensor-like histidine kinase YedV\n",
      "[18:09:37] Cleaned 399 /product names\n",
      "[18:09:37] Deleting unwanted file: PROKKA_10142018/sprot.faa\n",
      "[18:09:37] Deleting unwanted file: PROKKA_10142018/sprot.blast\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[18:09:38] There are still 1400 unannotated CDS left (started with 4998)\n",
      "[18:09:38] Will use hmmer3 to search against /home/analytics/anaconda3/db/hmm/HAMAP.hmm with 8 CPUs\n",
      "[18:09:38] Running: cat PROKKA_10142018\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 8 --block 21462 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-09 --cpu 1 /home/analytics/anaconda3/db/hmm/HAMAP.hmm /dev/stdin > PROKKA_10142018\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[18:10:11] Modify product: Uncharacterized Nudix hydrolase NudL => putative Nudix hydrolase NudL\n",
      "[18:10:11] Cleaned 1 /product names\n",
      "[18:10:11] Deleting unwanted file: PROKKA_10142018/HAMAP.hmm.faa\n",
      "[18:10:11] Deleting unwanted file: PROKKA_10142018/HAMAP.hmm.hmmer3\n",
      "[18:10:11] Labelling remaining 1312 proteins as 'hypothetical protein'\n",
      "[18:10:11] Possible /pseudo 'D-tagatose-1,6-bisphosphate aldolase subunit GatZ' at NODE_1_length_391920_cov_72.181440 position 93731\n",
      "[18:10:11] Possible /pseudo 'Protease ElaD' at NODE_1_length_391920_cov_72.181440 position 289994\n",
      "[18:10:11] Possible /pseudo 'putative lipoprotein YajI' at NODE_2_length_376590_cov_67.773500 position 142462\n",
      "[18:10:11] Possible /pseudo 'Cold shock-like protein CspG' at NODE_5_length_271618_cov_74.120598 position 16479\n",
      "[18:10:11] Possible /pseudo 'Formate dehydrogenase H' at NODE_8_length_180977_cov_80.739927 position 56696\n",
      "[18:10:11] Possible /pseudo 'Methylmalonyl-CoA mutase' at NODE_9_length_176064_cov_73.410274 position 121102\n",
      "[18:10:11] Possible /pseudo 'Energy-coupling factor transporter ATP-binding protein EcfA1' at NODE_9_length_176064_cov_73.410274 position 135862\n",
      "[18:10:11] Possible /pseudo 'Small toxic polypeptide LdrA' at NODE_10_length_157322_cov_71.698049 position 734\n",
      "[18:10:11] Possible /pseudo 'Hemolysin E, chromosomal' at NODE_10_length_157322_cov_71.698049 position 39823\n",
      "[18:10:11] Possible /pseudo 'Hemolysin E, chromosomal' at NODE_10_length_157322_cov_71.698049 position 40266\n",
      "[18:10:11] Possible /pseudo 'Cryptic beta-glucoside bgl operon antiterminator' at NODE_14_length_117508_cov_91.829149 position 45832\n",
      "[18:10:11] Possible /pseudo '6-phospho-alpha-glucosidase' at NODE_14_length_117508_cov_91.829149 position 90552\n",
      "[18:10:11] Possible /pseudo 'PTS system mannose-specific EIIBCA component' at NODE_19_length_84802_cov_71.598074 position 34878\n",
      "[18:10:11] Possible /pseudo 'Formate dehydrogenase-O major subunit' at NODE_19_length_84802_cov_71.598074 position 42043\n",
      "[18:10:11] Possible /pseudo 'putative protein YjgL' at NODE_22_length_75968_cov_80.340798 position 25599\n",
      "[18:10:11] Possible /pseudo 'Glucarate dehydratase-related protein' at NODE_27_length_60643_cov_69.781079 position 33739\n",
      "[18:10:11] Possible /pseudo 'tRNA pseudouridine synthase C' at NODE_27_length_60643_cov_69.781079 position 36804\n",
      "[18:10:11] Possible /pseudo 'Periplasmic alpha-amylase' at NODE_32_length_45646_cov_76.520432 position 2697\n",
      "[18:10:11] Possible /pseudo 'Small toxic polypeptide LdrD' at NODE_32_length_45646_cov_76.520432 position 45492\n",
      "[18:10:11] Possible /pseudo 'IS3 family transposase IS3' at NODE_33_length_41494_cov_73.803760 position 12237\n",
      "[18:10:11] Possible /pseudo 'Formate dehydrogenase, nitrate-inducible, major subunit' at NODE_35_length_38178_cov_73.280749 position 15855\n",
      "[18:10:11] Possible /pseudo 'Outer membrane protein YedS' at NODE_41_length_26624_cov_69.396515 position 2206\n",
      "[18:10:11] Possible /pseudo 'putative diguanylate cyclase DgcQ' at NODE_41_length_26624_cov_69.396515 position 9569\n",
      "[18:10:11] Possible /pseudo 'General stress protein 16U' at NODE_45_length_20943_cov_78.653485 position 2619\n",
      "[18:10:11] Possible /pseudo 'Multifunctional conjugation protein TraI' at NODE_57_length_7007_cov_106.954689 position 5510\n",
      "[18:10:11] Found 3153 unique /gene codes.\n",
      "[18:10:11] Fixed 2 duplicate /gene - dinI_1 dinI_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - kduD_1 kduD_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - bioF_1 bioF_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - gltC_1 gltC_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - dnaC_1 dnaC_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - rfaH_1 rfaH_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yggR_1 yggR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - murR_1 murR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - spaS_1 spaS_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - aaeA_1 aaeA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yrdA_1 yrdA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - rep_1 rep_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - pdhR_1 pdhR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - proA_1 proA_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - proP_1 proP_2 proP_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - yedS_1 yedS_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - gadX_1 gadX_2\n",
      "[18:10:11] Fixed 4 duplicate /gene - rhaR_1 rhaR_2 rhaR_3 rhaR_4\n",
      "[18:10:11] Fixed 2 duplicate /gene - fimA_1 fimA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yphB_1 yphB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - tmk_1 tmk_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - acrB_1 acrB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ccmF_1 ccmF_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - fes_1 fes_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - alx_1 alx_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - rapA_1 rapA_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - cysE_1 cysE_2 cysE_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - lptB_1 lptB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - lgoT_1 lgoT_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - mshA_1 mshA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - higA_1 higA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ybdO_1 ybdO_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - mepS_1 mepS_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - gatZ_1 gatZ_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yiaW_1 yiaW_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - acrR_1 acrR_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - fixX_1 fixX_2 fixX_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - lptG_1 lptG_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - bhsA_1 bhsA_2 bhsA_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - gtf1_1 gtf1_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - aspT_1 aspT_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - fic_1 fic_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - fbaA_1 fbaA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ftsA_1 ftsA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - srmB_1 srmB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - fdnH_1 fdnH_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - dcuD_1 dcuD_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yajI_1 yajI_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - purR_1 purR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - caiT_1 caiT_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ybgD_1 ybgD_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - mglA_1 mglA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - msbA_1 msbA_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - glnQ_1 glnQ_2 glnQ_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - argE_1 argE_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - php_1 php_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - flu_1 flu_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - chbC_1 chbC_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - rbsK_1 rbsK_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - cusR_1 cusR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - mepM_1 mepM_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - hin_1 hin_2\n",
      "[18:10:11] Fixed 5 duplicate /gene - yceD_1 yceD_2 yceD_3 yceD_4 yceD_5\n",
      "[18:10:11] Fixed 2 duplicate /gene - fruK_1 fruK_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ftsH_1 ftsH_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - flhA_1 flhA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - repA_1 repA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ulaB_1 ulaB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - hemN_1 hemN_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - focC_1 focC_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - acs_1 acs_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - azoR_1 azoR_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - hcaR_1 hcaR_2 hcaR_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - rob_1 rob_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - xdhA_1 xdhA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yagU_1 yagU_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - cusA_1 cusA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yhaJ_1 yhaJ_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - deoB_1 deoB_2\n",
      "[18:10:11] Fixed 4 duplicate /gene - rrrD_1 rrrD_2 rrrD_3 rrrD_4\n",
      "[18:10:11] Fixed 2 duplicate /gene - fdnG_1 fdnG_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - gntK_1 gntK_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ttdT_1 ttdT_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - klcA_1 klcA_2 klcA_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - psuK_1 psuK_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - folP_1 folP_2 folP_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - rhtC_1 rhtC_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ccmH_1 ccmH_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - bioD1_1 bioD1_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ydgA_1 ydgA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - noc_1 noc_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - chbA_1 chbA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - mlc_1 mlc_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - deoR_1 deoR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ygiW_1 ygiW_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - manX_1 manX_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - ompX_1 ompX_2 ompX_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - epsF_1 epsF_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - yhdJ_1 yhdJ_2 yhdJ_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - fdrA_1 fdrA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - cusB_1 cusB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - proQ_1 proQ_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - caiA_1 caiA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - hcpA_1 hcpA_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - vgrG1_1 vgrG1_2 vgrG1_3\n",
      "[18:10:11] Fixed 3 duplicate /gene - umuC_1 umuC_2 umuC_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - narL_1 narL_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yjaB_1 yjaB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - bglG_1 bglG_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - emrE_1 emrE_2 emrE_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - glpR_1 glpR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yhjE_1 yhjE_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yahB_1 yahB_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - gntR_1 gntR_2 gntR_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - ytfJ_1 ytfJ_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - mtlA_1 mtlA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - dgcQ_1 dgcQ_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - umuD_1 umuD_2 umuD_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - ompD_1 ompD_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - cusF_1 cusF_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - osmY_1 osmY_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yhcH_1 yhcH_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - elaD_1 elaD_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - gfcB_1 gfcB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - fucP_1 fucP_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - fixA_1 fixA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - leuC_1 leuC_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - vapC_1 vapC_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - tal_1 tal_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - scpA_1 scpA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yjgL_1 yjgL_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - nlpD_1 nlpD_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - cspG_1 cspG_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - cusC_1 cusC_2 cusC_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - yhhT_1 yhhT_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - oxyR_1 oxyR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - pspB_1 pspB_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - cynR_1 cynR_2 cynR_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - copA_1 copA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - psiB_1 psiB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - lsrK_1 lsrK_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - lolD_1 lolD_2\n",
      "[18:10:11] Fixed 4 duplicate /gene - lexA_1 lexA_2 lexA_3 lexA_4\n",
      "[18:10:11] Fixed 2 duplicate /gene - ftnA_1 ftnA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - dnaK_1 dnaK_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - clpV1_1 clpV1_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - malP_1 malP_2 malP_3\n",
      "[18:10:11] Fixed 4 duplicate /gene - hyfA_1 hyfA_2 hyfA_3 hyfA_4\n",
      "[18:10:11] Fixed 2 duplicate /gene - hha_1 hha_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - malS_1 malS_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - hycG_1 hycG_2\n",
      "[18:10:11] Fixed 5 duplicate /gene - srlR_1 srlR_2 srlR_3 srlR_4 srlR_5\n",
      "[18:10:11] Fixed 3 duplicate /gene - era_1 era_2 era_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - glpG_1 glpG_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ulaC_1 ulaC_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - manP_1 manP_2 manP_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - dcuA_1 dcuA_2\n",
      "[18:10:11] Fixed 5 duplicate /gene - xerD_1 xerD_2 xerD_3 xerD_4 xerD_5\n",
      "[18:10:11] Fixed 2 duplicate /gene - nanE_1 nanE_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ldrD_1 ldrD_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - tdh_1 tdh_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - rtcB_1 rtcB_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - ndhB_1 ndhB_2 ndhB_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - ybcO_1 ybcO_2\n",
      "[18:10:11] Fixed 4 duplicate /gene - btuD_1 btuD_2 btuD_3 btuD_4\n",
      "[18:10:11] Fixed 3 duplicate /gene - mngR_1 mngR_2 mngR_3\n",
      "[18:10:11] Fixed 4 duplicate /gene - intS_1 intS_2 intS_3 intS_4\n",
      "[18:10:11] Fixed 2 duplicate /gene - cdgI_1 cdgI_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - intQ_1 intQ_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ydhX_1 ydhX_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - fdoG_1 fdoG_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - tfaE_1 tfaE_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - rhsD_1 rhsD_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - epsE_1 epsE_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - pitA_1 pitA_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - dmsB_1 dmsB_2 dmsB_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - ulaE_1 ulaE_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - elfG_1 elfG_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - lsrB_1 lsrB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - bglF_1 bglF_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - fixB_1 fixB_2 fixB_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - ptsG_1 ptsG_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - mdh_1 mdh_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - epsM_1 epsM_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - bglA_1 bglA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - higB-2_1 higB-2_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - prfB_1 prfB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - sgrR_1 sgrR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - hdfR_1 hdfR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ecfA1_1 ecfA1_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - arnC_1 arnC_2 arnC_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - truC_1 truC_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - parM_1 parM_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - manZ_1 manZ_2 manZ_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - sbmC_1 sbmC_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - por_1 por_2 por_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - gstB_1 gstB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ydjE_1 ydjE_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ndhI_1 ndhI_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - pduV_1 pduV_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ampG_1 ampG_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - artM_1 artM_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - bvgA_1 bvgA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - papC_1 papC_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ydfO_1 ydfO_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - potA_1 potA_2 potA_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - lapB_1 lapB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yjiE_1 yjiE_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - nagB_1 nagB_2 nagB_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - ydcP_1 ydcP_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - gudX_1 gudX_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ssuC_1 ssuC_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ydjF_1 ydjF_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - lacA_1 lacA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - elfC_1 elfC_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - ydcV_1 ydcV_2 ydcV_3\n",
      "[18:10:11] Fixed 3 duplicate /gene - hlyE_1 hlyE_2 hlyE_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - nanR_1 nanR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - gspA_1 gspA_2\n",
      "[18:10:11] Fixed 4 duplicate /gene - rusA_1 rusA_2 rusA_3 rusA_4\n",
      "[18:10:11] Fixed 2 duplicate /gene - chuR_1 chuR_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - ssb_1 ssb_2 ssb_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - slp_1 slp_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - cirA_1 cirA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - yehB_1 yehB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - cusS_1 cusS_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - dinG_1 dinG_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - ais_1 ais_2 ais_3\n",
      "[18:10:11] Fixed 3 duplicate /gene - norR_1 norR_2 norR_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - ygdR_1 ygdR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ulaA_1 ulaA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - btuB_1 btuB_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - intA_1 intA_2 intA_3\n",
      "[18:10:11] Fixed 5 duplicate /gene - dmlR_1 dmlR_2 dmlR_3 dmlR_4 dmlR_5\n",
      "[18:10:11] Fixed 4 duplicate /gene - xerC_1 xerC_2 xerC_3 xerC_4\n",
      "[18:10:11] Fixed 2 duplicate /gene - kdgK_1 kdgK_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - pgrR_1 pgrR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - cysH_1 cysH_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - tnpR_1 tnpR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - aaeB_1 aaeB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - idi_1 idi_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - agaC_1 agaC_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - lutA_1 lutA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - recD_1 recD_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - ygcS_1 ygcS_2 ygcS_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - fryA_1 fryA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - mipA_1 mipA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ldrA_1 ldrA_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - pncC_1 pncC_2 pncC_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - lsrD_1 lsrD_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ubiE_1 ubiE_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - lldR_1 lldR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - rspA_1 rspA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - arcC1_1 arcC1_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - lon_1 lon_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - fdhF_1 fdhF_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - silE_1 silE_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - dmsC_1 dmsC_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - traI_1 traI_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - acpP_1 acpP_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - tsaR_1 tsaR_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - aglB_1 aglB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - coaA_1 coaA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - sapB_1 sapB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - fryB_1 fryB_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - bla_1 bla_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - cutC_1 cutC_2 cutC_3\n",
      "[18:10:11] Fixed 2 duplicate /gene - ugpC_1 ugpC_2\n",
      "[18:10:11] Fixed 3 duplicate /gene - bglH_1 bglH_2 bglH_3\n",
      "[18:10:11] Fixed 5 duplicate /gene - rhaS_1 rhaS_2 rhaS_3 rhaS_4 rhaS_5\n",
      "[18:10:11] Fixed 2 duplicate /gene - kefC_1 kefC_2\n",
      "[18:10:11] Fixed 4 duplicate /gene - rbsC_1 rbsC_2 rbsC_3 rbsC_4\n",
      "[18:10:11] Fixed 2 duplicate /gene - topA_1 topA_2\n",
      "[18:10:11] Fixed 2 duplicate /gene - ydjZ_1 ydjZ_2\n",
      "[18:10:11] Fixed 274 colliding /gene names.\n",
      "[18:10:11] Adding /locus_tag identifiers\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[18:10:12] Assigned 5090 locus_tags to CDS and RNA features.\n",
      "[18:10:12] Writing outputs to PROKKA_10142018/\n",
      "[18:10:14] Generating annotation statistics file\n",
      "[18:10:14] Generating Genbank and Sequin files\n",
      "[18:10:14] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13.3 from https://github.com/tseemann/prokka' -Z PROKKA_10142018\\/PROKKA_10142018\\.err -i PROKKA_10142018\\/PROKKA_10142018\\.fsa 2> /dev/null\n",
      "[18:10:37] Deleting unwanted file: PROKKA_10142018/errorsummary.val\n",
      "[18:10:37] Deleting unwanted file: PROKKA_10142018/PROKKA_10142018.dr\n",
      "[18:10:37] Deleting unwanted file: PROKKA_10142018/PROKKA_10142018.fixedproducts\n",
      "[18:10:37] Deleting unwanted file: PROKKA_10142018/PROKKA_10142018.ecn\n",
      "[18:10:37] Deleting unwanted file: PROKKA_10142018/PROKKA_10142018.val\n",
      "[18:10:37] Repairing broken .GBK output that tbl2asn produces...\n",
      "[18:10:37] Running: sed 's/COORDINATES: profile/COORDINATES:profile/' < PROKKA_10142018\\/PROKKA_10142018\\.gbf > PROKKA_10142018\\/PROKKA_10142018\\.gbk\n",
      "[18:10:37] Deleting unwanted file: PROKKA_10142018/PROKKA_10142018.gbf\n",
      "[18:10:37] Output files:\n",
      "[18:10:37] PROKKA_10142018/PROKKA_10142018.log\n",
      "[18:10:37] PROKKA_10142018/PROKKA_10142018.fna\n",
      "[18:10:37] PROKKA_10142018/PROKKA_10142018.gff\n",
      "[18:10:37] PROKKA_10142018/PROKKA_10142018.faa\n",
      "[18:10:37] PROKKA_10142018/PROKKA_10142018.ffn\n",
      "[18:10:37] PROKKA_10142018/PROKKA_10142018.tbl\n",
      "[18:10:37] PROKKA_10142018/PROKKA_10142018.fsa\n",
      "[18:10:37] PROKKA_10142018/PROKKA_10142018.tsv\n",
      "[18:10:37] PROKKA_10142018/PROKKA_10142018.txt\n",
      "[18:10:37] PROKKA_10142018/PROKKA_10142018.sqn\n",
      "[18:10:37] PROKKA_10142018/PROKKA_10142018.err\n",
      "[18:10:37] PROKKA_10142018/PROKKA_10142018.gbk\n",
      "[18:10:37] Annotation finished successfully.\n",
      "[18:10:37] Walltime used: 2.72 minutes\n",
      "[18:10:37] If you use this result please cite the Prokka paper:\n",
      "[18:10:37] Seemann T (2014) Prokka: rapid prokaryotic genome annotation. Bioinformatics. 30(14):2068-9.\n",
      "[18:10:37] Type 'prokka --citation' for more details.\n",
      "[18:10:37] Share and enjoy!\n"
     ]
    }
   ],
   "source": [
    "!$PROKKA ../data/week3/assembly_all/contigs.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /home/analytics/anaconda3/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /home/analytics/anaconda3/bin/nhmmer\n",
      "[barrnap] Found bedtools - /home/analytics/anaconda3/bin/bedtools\n",
      "[barrnap] Will use 50 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /home/analytics/anaconda3/lib/barrnap/bin/../db/bac.hmm\n",
      "[barrnap] Scanning ../data/week3/assembly_all/scaffolds.fasta for bac rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 50 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/home/analytics/anaconda3/lib/barrnap/bin/../db/bac.hmm' '../data/week3/assembly_all/scaffolds.fasta'\n",
      "[barrnap] Found: 16S_rRNA NODE_1_length_2374157_cov_75.937583 L=1538/1585 1986881..1988418 + 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA NODE_1_length_2374157_cov_75.937583 L=1538/1585 526768..528305 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA NODE_1_length_2374157_cov_75.937583 L=1538/1585 1129403..1130940 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA NODE_1_length_2374157_cov_75.937583 L=1538/1585 1170919..1172456 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA NODE_1_length_2374157_cov_75.937583 L=1538/1585 1310618..1312155 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA NODE_1_length_2374157_cov_75.937583 L=1538/1585 1408017..1409554 - 16S ribosomal RNA\n",
      "[barrnap] Found: 16S_rRNA NODE_2_length_1046850_cov_73.627294 L=1538/1585 644690..646227 - 16S ribosomal RNA\n",
      "[barrnap] Rejecting short 285 nt predicted 16S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 16S_rRNA NODE_79_length_720_cov_1.094737 L=406/1585 314..719 + 16S ribosomal RNA (partial)\n",
      "[barrnap] Rejecting short 180 nt predicted 16S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Rejecting short 204 nt predicted 16S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 23S_rRNA NODE_1_length_2374157_cov_75.937583 L=2901/3232 1307268..1310168 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA NODE_1_length_2374157_cov_75.937583 L=2901/3232 1126145..1129045 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA NODE_1_length_2374157_cov_75.937583 L=2901/3232 1404759..1407659 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA NODE_1_length_2374157_cov_75.937583 L=2905/3232 1167657..1170561 - 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA NODE_1_length_2374157_cov_75.937583 L=2758/3232 1988919..1991676 + 23S ribosomal RNA\n",
      "[barrnap] Found: 23S_rRNA NODE_2_length_1046850_cov_73.627294 L=2758/3232 641432..644189 - 23S ribosomal RNA\n",
      "[barrnap] Rejecting short 340 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Rejecting short 250 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Rejecting short 365 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Rejecting short 102 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Rejecting short 363 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Rejecting short 101 nt predicted 23S_rRNA. Adjust via --reject option.\n",
      "[barrnap] Found: 5S_rRNA NODE_286_length_223_cov_0.720238 L=111/119 19..129 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA NODE_1_length_2374157_cov_75.937583 L=111/119 1991775..1991885 + 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA NODE_1_length_2374157_cov_75.937583 L=111/119 1125936..1126046 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA NODE_2_length_1046850_cov_73.627294 L=111/119 641223..641333 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA NODE_1_length_2374157_cov_75.937583 L=102/119 1167383..1167484 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA NODE_1_length_2374157_cov_75.937583 L=102/119 1306994..1307095 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA NODE_1_length_2374157_cov_75.937583 L=102/119 1404485..1404586 - 5S ribosomal RNA\n",
      "[barrnap] Found 21 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "##gff-version 3\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t526768\t528305\t0\t-\t.\tName=16S_rRNA;product=16S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1125936\t1126046\t2.2e-11\t-\t.\tName=5S_rRNA;product=5S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1126145\t1129045\t0\t-\t.\tName=23S_rRNA;product=23S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1129403\t1130940\t0\t-\t.\tName=16S_rRNA;product=16S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1167383\t1167484\t4.9e-10\t-\t.\tName=5S_rRNA;product=5S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1167657\t1170561\t0\t-\t.\tName=23S_rRNA;product=23S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1170919\t1172456\t0\t-\t.\tName=16S_rRNA;product=16S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1306994\t1307095\t4.9e-10\t-\t.\tName=5S_rRNA;product=5S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1307268\t1310168\t0\t-\t.\tName=23S_rRNA;product=23S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1310618\t1312155\t0\t-\t.\tName=16S_rRNA;product=16S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1404485\t1404586\t4.9e-10\t-\t.\tName=5S_rRNA;product=5S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1404759\t1407659\t0\t-\t.\tName=23S_rRNA;product=23S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1408017\t1409554\t0\t-\t.\tName=16S_rRNA;product=16S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1986881\t1988418\t0\t+\t.\tName=16S_rRNA;product=16S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1988919\t1991676\t0\t+\t.\tName=23S_rRNA;product=23S ribosomal RNA\n",
      "NODE_1_length_2374157_cov_75.937583\tbarrnap:0.9\trRNA\t1991775\t1991885\t2.2e-11\t+\t.\tName=5S_rRNA;product=5S ribosomal RNA\n",
      "NODE_286_length_223_cov_0.720238\tbarrnap:0.9\trRNA\t19\t129\t5.6e-18\t+\t.\tName=5S_rRNA;product=5S ribosomal RNA\n",
      "NODE_2_length_1046850_cov_73.627294\tbarrnap:0.9\trRNA\t641223\t641333\t2.2e-11\t-\t.\tName=5S_rRNA;product=5S ribosomal RNA\n",
      "NODE_2_length_1046850_cov_73.627294\tbarrnap:0.9\trRNA\t641432\t644189\t0\t-\t.\tName=23S_rRNA;product=23S ribosomal RNA\n",
      "NODE_2_length_1046850_cov_73.627294\tbarrnap:0.9\trRNA\t644690\t646227\t0\t-\t.\tName=16S_rRNA;product=16S ribosomal RNA\n",
      "NODE_79_length_720_cov_1.094737\tbarrnap:0.9\trRNA\t314\t719\t9.9e-23\t+\t.\tName=16S_rRNA;product=16S ribosomal RNA (partial);note=aligned only 25 percent of the 16S ribosomal RNA\n",
      "[barrnap] Writing hit sequences to: ../data/week3/rrna_seq.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo '../data/week3/rrna_seq.fa' -fi '../data/week3/assembly_all/scaffolds.fasta' -bed '/tmp/gOtSSQYlTc'\n",
      "index file ../data/week3/assembly_all/scaffolds.fasta.fai not found, generating...\n",
      "[barrnap] Done.\n"
     ]
    }
   ],
   "source": [
    "!$BARRNAP --threads 50 --outseq ../data/week3/rrna_seq.fa ../data/week3/assembly_all/scaffolds.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">16S_rRNA::NODE_1_length_2374157_cov_75.937583:1986880-1988418(+)\r\n",
      "TTGAAGAGTTTGATCATGGCTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAACGGTAACAGGAAACAGCTTGCTGTTTCGCTGACGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGACCAAAGAGGGGGACCTTCGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTAGCTTGTTGGTGGGGTAACGGCTCACCAAGGCGACGATCCCTAGCTGGTCTGAGAGGATGACCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGCGCAAGCCTGATGCAGCCATGCCGCGTGTATGAAGAAGGCCTTCGGGTTGTAAAGTACTTTCAGCGGGGAGGAAGGGAGTAAAGTTAATACCTTTGCTCATTGACGTTACCCGCAGAAGAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTTTGTTAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATCTGATACTGGCAAGCTTGAGTCTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACGAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGTCGACTTGGAGGTTGTGCCCTTGAGGCGTGGCTTCCGGAGCTAACGCGTTAAGTCGACCGCCTGGGGAGTACGGCCGCAAGGTTAAAACTCAAATGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAAGTTTTCAGAGATGAGAATGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTTGTGAAATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTATCCTTTGTTGCCAGCGGTCCGGCCGGGAACTCAAAGGAGACTGCCAGTGATAAACTGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTGCTACAATGGCGCATACAAAGAGAAGCGACCTCGCGAGAGCAAGCGGACCTCATAAAGTGCGTCGTAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGTGGATCAGAATGCCACGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTGGGTTGCAAAAGAAGTAGGTAGCTTAACCTTCGGGAGGGCGCTTACCACTTTGTGATTCATGACTGGGGTGAAGTCGTAACAAGGTAACCGTAGGGGAACCTGCGGTTGGATCACCTCCTT\r\n",
      ">16S_rRNA::NODE_1_length_2374157_cov_75.937583:526767-528305(-)\r\n",
      "TTGAAGAGTTTGATCATGGCTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAACGGTAACAGGAAACAGCTTGCTGTTTCGCTGACGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGACCAAAGAGGGGGACCTTCGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTAGCTTGTTGGTGGGGTAACGGCTCACCAAGGCGACGATCCCTAGCTGGTCTGAGAGGATGACCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGCGCAAGCCTGATGCAGCCATGCCGCGTGTATGAAGAAGGCCTTCGGGTTGTAAAGTACTTTCAGCGGGGAGGAAGGGAGTAAAGTTAATACCTTTGCTCATTGACGTTACCCGCAGAAGAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTTTGTTAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATCTGATACTGGCAAGCTTGAGTCTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACGAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGTCGACTTGGAGGTTGTGCCCTTGAGGCGTGGCTTCCGGAGCTAACGCGTTAAGTCGACCGCCTGGGGAGTACGGCCGCAAGGTTAAAACTCAAATGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAAGTTTTCAGAGATGAGAATGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTTGTGAAATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTATCCTTTGTTGCCAGCGGTCCGGCCGGGAACTCAAAGGAGACTGCCAGTGATAAACTGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTGCTACAATGGCGCATACAAAGAGAAGCGACCTCGCGAGAGCAAGCGGACCTCATAAAGTGCGTCGTAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGTGGATCAGAATGCCACGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTGGGTTGCAAAAGAAGTAGGTAGCTTAACCTTCGGGAGGGCGCTTACCACTTTGTGATTCATGACTGGGGTGAAGTCGTAACAAGGTAACCGTAGGGGAACCTGCGGTTGGATCACCTCCTT\r\n",
      ">16S_rRNA::NODE_1_length_2374157_cov_75.937583:1129402-1130940(-)\r\n",
      "TTGAAGAGTTTGATCATGGCTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAACGGTAACAGGAAACAGCTTGCTGTTTCGCTGACGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGACCAAAGAGGGGGACCTTCGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTAGCTTGTTGGTGGGGTAACGGCTCACCAAGGCGACGATCCCTAGCTGGTCTGAGAGGATGACCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGCGCAAGCCTGATGCAGCCATGCCGCGTGTATGAAGAAGGCCTTCGGGTTGTAAAGTACTTTCAGCGGGGAGGAAGGGAGTAAAGTTAATACCTTTGCTCATTGACGTTACCCGCAGAAGAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTTTGTTAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATCTGATACTGGCAAGCTTGAGTCTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACGAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGTCGACTTGGAGGTTGTGCCCTTGAGGCGTGGCTTCCGGAGCTAACGCGTTAAGTCGACCGCCTGGGGAGTACGGCCGCAAGGTTAAAACTCAAATGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAAGTTTTCAGAGATGAGAATGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTTGTGAAATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTATCCTTTGTTGCCAGCGGTCCGGCCGGGAACTCAAAGGAGACTGCCAGTGATAAACTGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTGCTACAATGGCGCATACAAAGAGAAGCGACCTCGCGAGAGCAAGCGGACCTCATAAAGTGCGTCGTAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGTGGATCAGAATGCCACGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTGGGTTGCAAAAGAAGTAGGTAGCTTAACCTTCGGGAGGGCGCTTACCACTTTGTGATTCATGACTGGGGTGAAGTCGTAACAAGGTAACCGTAGGGGAACCTGCGGTTGGATCACCTCCTT\r\n",
      ">16S_rRNA::NODE_1_length_2374157_cov_75.937583:1170918-1172456(-)\r\n",
      "TTGAAGAGTTTGATCATGGCTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAACGGTAACAGGAAACAGCTTGCTGTTTCGCTGACGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGACCAAAGAGGGGGACCTTCGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTAGCTTGTTGGTGGGGTAACGGCTCACCAAGGCGACGATCCCTAGCTGGTCTGAGAGGATGACCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGCGCAAGCCTGATGCAGCCATGCCGCGTGTATGAAGAAGGCCTTCGGGTTGTAAAGTACTTTCAGCGGGGAGGAAGGGAGTAAAGTTAATACCTTTGCTCATTGACGTTACCCGCAGAAGAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTTTGTTAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATCTGATACTGGCAAGCTTGAGTCTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACGAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGTCGACTTGGAGGTTGTGCCCTTGAGGCGTGGCTTCCGGAGCTAACGCGTTAAGTCGACCGCCTGGGGAGTACGGCCGCAAGGTTAAAACTCAAATGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAAGTTTTCAGAGATGAGAATGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTTGTGAAATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTATCCTTTGTTGCCAGCGGTCCGGCCGGGAACTCAAAGGAGACTGCCAGTGATAAACTGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTGCTACAATGGCGCATACAAAGAGAAGCGACCTCGCGAGAGCAAGCGGACCTCATAAAGTGCGTCGTAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGTGGATCAGAATGCCACGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTGGGTTGCAAAAGAAGTAGGTAGCTTAACCTTCGGGAGGGCGCTTACCACTTTGTGATTCATGACTGGGGTGAAGTCGTAACAAGGTAACCGTAGGGGAACCTGCGGTTGGATCACCTCCTT\r\n",
      ">16S_rRNA::NODE_1_length_2374157_cov_75.937583:1310617-1312155(-)\r\n",
      "TTGAAGAGTTTGATCATGGCTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAACGGTAACAGGAAACAGCTTGCTGTTTCGCTGACGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGACCAAAGAGGGGGACCTTCGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTAGCTTGTTGGTGGGGTAACGGCTCACCAAGGCGACGATCCCTAGCTGGTCTGAGAGGATGACCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGCGCAAGCCTGATGCAGCCATGCCGCGTGTATGAAGAAGGCCTTCGGGTTGTAAAGTACTTTCAGCGGGGAGGAAGGGAGTAAAGTTAATACCTTTGCTCATTGACGTTACCCGCAGAAGAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTTTGTTAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATCTGATACTGGCAAGCTTGAGTCTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACGAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGTCGACTTGGAGGTTGTGCCCTTGAGGCGTGGCTTCCGGAGCTAACGCGTTAAGTCGACCGCCTGGGGAGTACGGCCGCAAGGTTAAAACTCAAATGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAAGTTTTCAGAGATGAGAATGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTTGTGAAATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTATCCTTTGTTGCCAGCGGTCCGGCCGGGAACTCAAAGGAGACTGCCAGTGATAAACTGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTGCTACAATGGCGCATACAAAGAGAAGCGACCTCGCGAGAGCAAGCGGACCTCATAAAGTGCGTCGTAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGTGGATCAGAATGCCACGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTGGGTTGCAAAAGAAGTAGGTAGCTTAACCTTCGGGAGGGCGCTTACCACTTTGTGATTCATGACTGGGGTGAAGTCGTAACAAGGTAACCGTAGGGGAACCTGCGGTTGGATCACCTCCTT\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 10 ../data/week3/rrna_seq.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 32M\r\n",
      "-rw-rw-r--. 1 analytics analytics  774 Oct 12 11:53 input_dataset.yaml\r\n",
      "-rw-rw-r--. 1 analytics analytics 2.8K Oct 12 11:53 params.txt\r\n",
      "drwxrwxr-x. 3 analytics analytics 4.0K Oct 12 12:07 corrected\r\n",
      "-rw-rw-r--. 1 analytics analytics   96 Oct 12 12:07 dataset.info\r\n",
      "drwxrwxr-x. 3 analytics analytics  194 Oct 12 12:09 K21\r\n",
      "drwxrwxr-x. 3 analytics analytics  194 Oct 12 12:10 K33\r\n",
      "drwxrwxr-x. 4 analytics analytics 4.0K Oct 12 12:13 K55\r\n",
      "-rw-rw-r--. 1 analytics analytics 5.3M Oct 12 12:13 before_rr.fasta\r\n",
      "-rw-rw-r--. 1 analytics analytics 5.3M Oct 12 12:13 contigs.fasta\r\n",
      "-rw-rw-r--. 1 analytics analytics  59K Oct 12 12:13 scaffolds.paths\r\n",
      "-rw-rw-r--. 1 analytics analytics 5.4M Oct 12 12:13 scaffolds.fasta\r\n",
      "-rw-rw-r--. 1 analytics analytics 5.2M Oct 12 12:13 assembly_graph_with_scaffolds.gfa\r\n",
      "-rw-rw-r--. 1 analytics analytics  64K Oct 12 12:13 contigs.paths\r\n",
      "-rw-rw-r--. 1 analytics analytics  11M Oct 12 12:13 assembly_graph.fastg\r\n",
      "drwxrwxr-x. 2 analytics analytics    6 Oct 12 12:13 tmp\r\n",
      "drwxrwxr-x. 2 analytics analytics   36 Oct 12 12:13 misc\r\n",
      "-rw-rw-r--. 1 analytics analytics 154K Oct 12 12:13 spades.log\r\n"
     ]
    }
   ],
   "source": [
    "!ls -lhrt ../data/week3/assembly_all/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Analysis complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Analysis complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Analysis complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Analysis complete for SRR292770_S1_L001_R2_001.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!$JELLYFISH histo mer_counts_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!$JELLYFISH histo mer_counts_27"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!$JELLYFISH count -m 31 -s 25000000 ../data/week3/SRR292678sub_S1_L001_R1_001.fastq\n",
    "\n",
    "!$JELLYFISH histo mer_counts_0 > ../data/week3/histo\n",
    "\n",
    "from matplotlib import pyplot as plt\n",
    "import pandas as pd\n",
    "\n",
    "%matplotlib inline\n",
    "\n",
    "df = pd.read_csv(\"../data/week3/histo\", sep=\" \", header=None)\n",
    "\n",
    "fig = plt.figure(figsize=(12, 4))\n",
    "plt.plot(df.loc[:, 1])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "$N = \\frac{M\\ L}{L-K+1}$  \n",
    "Genome_size = $\\frac{T}{N}$"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Reads occurred ony once  errors. The peak is at 31"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda env:anaconda3]",
   "language": "python",
   "name": "conda-env-anaconda3-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
