{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "HAPLOGREP = \"/home/analytics/distr/haplogrep-cmd/haplogrep-2.1.16.jar\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p ../data/week7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2018-11-14 17:57:59--  http://public.dobzhanskycenter.ru/mrayko/BIMM185/SNP_raw_v4_Full_20170514175358.txt\n",
      "Resolving public.dobzhanskycenter.ru (public.dobzhanskycenter.ru)... 195.70.204.3\n",
      "Connecting to public.dobzhanskycenter.ru (public.dobzhanskycenter.ru)|195.70.204.3|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 15683002 (15M) [text/plain]\n",
      "Saving to: ‘../data/week7/SNP_raw_v4_Full_20170514175358.txt’\n",
      "\n",
      "100%[======================================>] 15,683,002  28.9MB/s   in 0.5s   \n",
      "\n",
      "2018-11-14 17:57:59 (28.9 MB/s) - ‘../data/week7/SNP_raw_v4_Full_20170514175358.txt’ saved [15683002/15683002]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget -P ../data/week7 http://public.dobzhanskycenter.ru/mrayko/BIMM185/SNP_raw_v4_Full_20170514175358.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mv ../data/week7/SNP_raw_v4_Full_20170514175358.txt ../data/week7/SNP.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# This data file generated by 23andMe at: Sun May 14 17:53:58 2017\r",
      "\r\n",
      "#\r",
      "\r\n",
      "# This file contains raw genotype data, including data that is not used in 23andMe reports.\r",
      "\r\n",
      "# This data has undergone a general quality review however only a subset of markers have been \r",
      "\r\n",
      "# individually validated for accuracy. As such, this data is suitable only for research, \r",
      "\r\n",
      "# educational, and informational use and not for medical or other use.\r",
      "\r\n",
      "# \r",
      "\r\n",
      "# Below is a text version of your data.  Fields are TAB-separated\r",
      "\r\n",
      "# Each line corresponds to a single SNP.  For each SNP, we provide its identifier \r",
      "\r\n",
      "# (an rsid or an internal id), its location on the reference human genome, and the \r",
      "\r\n",
      "# genotype call oriented with respect to the plus strand on the human reference sequence.\r",
      "\r\n",
      "# We are using reference human assembly build 37 (also known as Annotation Release 104).\r",
      "\r\n",
      "# Note that it is possible that data downloaded at different times may be different due to ongoing \r",
      "\r\n",
      "# improvements in our ability to call genotypes. More information about these changes can be found at:\r",
      "\r\n",
      "# https://www.23andme.com/you/download/revisions/\r",
      "\r\n",
      "# \r",
      "\r\n",
      "# More information on reference human assembly build 37 (aka Annotation Release 104):\r",
      "\r\n",
      "# http://www.ncbi.nlm.nih.gov/mapview/map_search.cgi?taxid=9606\r",
      "\r\n",
      "#\r",
      "\r\n",
      "# rsid\tchromosome\tposition\tgenotype\r",
      "\r\n",
      "rs12564807\t1\t734462\tAA\r",
      "\r\n",
      "rs3131972\t1\t752721\tAG\r",
      "\r\n",
      "rs148828841\t1\t760998\tCC\r",
      "\r\n",
      "rs12124819\t1\t776546\tAA\r",
      "\r\n",
      "rs115093905\t1\t787173\t--\r",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 25 ../data/week7/SNP.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "BCFTOOLS = \"/home/analytics/distr/samtools/bcftools/bcftools\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2018-11-14 18:44:47--  http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz\n",
      "Resolving hgdownload.cse.ucsc.edu (hgdownload.cse.ucsc.edu)... 128.114.119.163\n",
      "Connecting to hgdownload.cse.ucsc.edu (hgdownload.cse.ucsc.edu)|128.114.119.163|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 983659424 (938M) [application/x-gzip]\n",
      "Saving to: ‘../data/week7/hg38.fa.gz’\n",
      "\n",
      "100%[======================================>] 983,659,424 16.1MB/s   in 60s    \n",
      "\n",
      "2018-11-14 18:45:48 (15.5 MB/s) - ‘../data/week7/hg38.fa.gz’ saved [983659424/983659424]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget -P ../data/week7/ http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "!gunzip ../data/week7/hg38.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat ../data/week7/SNP.txt | tail -n +21 > ../data/week7/SNP_.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "TOVCF = \"/home/analytics/anaconda3/envs/bioinf/bin/23andme-to-vcf\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2018-11-14 18:55:12--  ftp://ftp.ensembl.org/pub/release-94/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna_sm.toplevel.fa.gz\n",
      "           => ‘../data/week7/Homo_sapiens.GRCh38.dna_sm.toplevel.fa.gz’\n",
      "Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.8\n",
      "Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.193.8|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /pub/release-94/fasta/homo_sapiens/dna ... done.\n",
      "==> SIZE Homo_sapiens.GRCh38.dna_sm.toplevel.fa.gz ... 1147601786\n",
      "==> PASV ... done.    ==> RETR Homo_sapiens.GRCh38.dna_sm.toplevel.fa.gz ... done.\n",
      "Length: 1147601786 (1.1G) (unauthoritative)\n",
      "\n",
      "100%[====================================>] 1,147,601,786 25.8MB/s   in 39s    \n",
      "\n",
      "2018-11-14 18:55:52 (27.9 MB/s) - ‘../data/week7/Homo_sapiens.GRCh38.dna_sm.toplevel.fa.gz’ saved [1147601786]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget -P ../data/week7/ ftp://ftp.ensembl.org/pub/release-94/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna_sm.toplevel.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "!$TOVCF --input ../data/week7/SNP.txt --fasta /data/analytics/bioinf_workshop/data/week6/hg19_.fa --fai /data/analytics/bioinf_workshop/data/week6/hg19_.fa.fai --output ../data/week7/prof_SNP.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##fileformat=VCFv4.2\r\n",
      "##source=23andme_to_vcf\r\n",
      "##reference=GRCh37\r\n",
      "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">\r\n",
      "#CHROM POS ID REF ALT QUAL FILTER INFO FORMAT SAMPLE\r\n",
      "chr1\t734462\trs12564807\tG\tA\t.\t.\t.\tGT\t1/1\r\n",
      "chr1\t752721\trs3131972\tA\tG\t.\t.\t.\tGT\t0/1\r\n",
      "chr1\t798959\trs11240777\tg\tA\t.\t.\t.\tGT\t1/2\r\n",
      "chr1\t798959\trs11240777\tg\tG\t.\t.\t.\tGT\t2/1\r\n",
      "chr1\t838555\trs4970383\tc\tC\t.\t.\t.\tGT\t1/1\r\n",
      "chr1\t889159\trs13302945\tA\tC\t.\t.\t.\tGT\t1/1\r\n",
      "chr1\t891945\trs13303106\tA\tG\t.\t.\t.\tGT\t0/1\r\n",
      "chr1\t894573\trs13303010\tG\tA\t.\t.\t.\tGT\t0/1\r\n",
      "chr1\t903104\trs6696281\tC\tT\t.\t.\t.\tGT\t0/1\r\n",
      "chr1\t909238\ti6060381\tG\tC\t.\t.\t.\tGT\t1/1\r\n",
      "chr1\t910935\trs2340592\tG\tA\t.\t.\t.\tGT\t0/1\r\n",
      "chr1\t918384\trs13303118\tG\tT\t.\t.\t.\tGT\t0/1\r\n",
      "chr1\t921071\trs78164078\tg\tG\t.\t.\t.\tGT\t1/1\r\n",
      "chr1\t924898\trs6665000\tc\tA\t.\t.\t.\tGT\t1/1\r\n",
      "chr1\t927309\trs2341362\tT\tC\t.\t.\t.\tGT\t1/1\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 20 ../data/week7/prof_SNP.vcf"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python (bioinf)",
   "language": "python",
   "name": "myenv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
